// @alicloud/dingtalk/dist/attendance_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class AddLeaveTypeHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.AddLeaveTypeHeaders = AddLeaveTypeHeaders;
class AddLeaveTypeRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            bizType: 'bizType',
            extras: 'extras',
            hoursInPerDay: 'hoursInPerDay',
            leaveCertificate: 'leaveCertificate',
            leaveName: 'leaveName',
            leaveViewUnit: 'leaveViewUnit',
            naturalDayLeave: 'naturalDayLeave',
            submitTimeRule: 'submitTimeRule',
            visibilityRules: 'visibilityRules',
            opUserId: 'opUserId',
        };
    }
    static types() {
        return {
            bizType: 'string',
            extras: 'string',
            hoursInPerDay: 'number',
            leaveCertificate: AddLeaveTypeRequestLeaveCertificate,
            leaveName: 'string',
            leaveViewUnit: 'string',
            naturalDayLeave: 'boolean',
            submitTimeRule: AddLeaveTypeRequestSubmitTimeRule,
            visibilityRules: { 'type': 'array', 'itemType': AddLeaveTypeRequestVisibilityRules },
            opUserId: 'string',
        };
    }
}
exports.AddLeaveTypeRequest = AddLeaveTypeRequest;
class AddLeaveTypeResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: AddLeaveTypeResponseBodyResult,
        };
    }
}
exports.AddLeaveTypeResponseBody = AddLeaveTypeResponseBody;
class AddLeaveTypeResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: AddLeaveTypeResponseBody,
        };
    }
}
exports.AddLeaveTypeResponse = AddLeaveTypeResponse;
class AttendanceBleDevicesAddHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.AttendanceBleDevicesAddHeaders = AttendanceBleDevicesAddHeaders;
class AttendanceBleDevicesAddRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deviceIdList: 'deviceIdList',
            groupKey: 'groupKey',
            opUserId: 'opUserId',
        };
    }
    static types() {
        return {
            deviceIdList: { 'type': 'array', 'itemType': 'number' },
            groupKey: 'string',
            opUserId: 'string',
        };
    }
}
exports.AttendanceBleDevicesAddRequest = AttendanceBleDevicesAddRequest;
class AttendanceBleDevicesAddResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            errorList: 'errorList',
            successList: 'successList',
        };
    }
    static types() {
        return {
            errorList: { 'type': 'array', 'itemType': AttendanceBleDevicesAddResponseBodyErrorList },
            successList: { 'type': 'array', 'itemType': AttendanceBleDevicesAddResponseBodySuccessList },
        };
    }
}
exports.AttendanceBleDevicesAddResponseBody = AttendanceBleDevicesAddResponseBody;
class AttendanceBleDevicesAddResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: AttendanceBleDevicesAddResponseBody,
        };
    }
}
exports.AttendanceBleDevicesAddResponse = AttendanceBleDevicesAddResponse;
class AttendanceBleDevicesQueryHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.AttendanceBleDevicesQueryHeaders = AttendanceBleDevicesQueryHeaders;
class AttendanceBleDevicesQueryRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupKey: 'groupKey',
            opUserId: 'opUserId',
        };
    }
    static types() {
        return {
            groupKey: 'string',
            opUserId: 'string',
        };
    }
}
exports.AttendanceBleDevicesQueryRequest = AttendanceBleDevicesQueryRequest;
class AttendanceBleDevicesQueryResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': AttendanceBleDevicesQueryResponseBodyResult },
        };
    }
}
exports.AttendanceBleDevicesQueryResponseBody = AttendanceBleDevicesQueryResponseBody;
class AttendanceBleDevicesQueryResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: AttendanceBleDevicesQueryResponseBody,
        };
    }
}
exports.AttendanceBleDevicesQueryResponse = AttendanceBleDevicesQueryResponse;
class AttendanceBleDevicesRemoveHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.AttendanceBleDevicesRemoveHeaders = AttendanceBleDevicesRemoveHeaders;
class AttendanceBleDevicesRemoveRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deviceIdList: 'deviceIdList',
            groupKey: 'groupKey',
            opUserId: 'opUserId',
        };
    }
    static types() {
        return {
            deviceIdList: { 'type': 'array', 'itemType': 'number' },
            groupKey: 'string',
            opUserId: 'string',
        };
    }
}
exports.AttendanceBleDevicesRemoveRequest = AttendanceBleDevicesRemoveRequest;
class AttendanceBleDevicesRemoveResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            errorList: 'errorList',
            successList: 'successList',
        };
    }
    static types() {
        return {
            errorList: { 'type': 'array', 'itemType': AttendanceBleDevicesRemoveResponseBodyErrorList },
            successList: { 'type': 'array', 'itemType': 'number' },
        };
    }
}
exports.AttendanceBleDevicesRemoveResponseBody = AttendanceBleDevicesRemoveResponseBody;
class AttendanceBleDevicesRemoveResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: AttendanceBleDevicesRemoveResponseBody,
        };
    }
}
exports.AttendanceBleDevicesRemoveResponse = AttendanceBleDevicesRemoveResponse;
class CheckClosingAccountHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CheckClosingAccountHeaders = CheckClosingAccountHeaders;
class CheckClosingAccountRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            bizCode: 'bizCode',
            userIds: 'userIds',
            userTimeRange: 'userTimeRange',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            userIds: { 'type': 'array', 'itemType': 'string' },
            userTimeRange: { 'type': 'array', 'itemType': CheckClosingAccountRequestUserTimeRange },
        };
    }
}
exports.CheckClosingAccountRequest = CheckClosingAccountRequest;
class CheckClosingAccountResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            code: 'code',
            mesage: 'mesage',
            pass: 'pass',
        };
    }
    static types() {
        return {
            code: 'string',
            mesage: 'string',
            pass: 'boolean',
        };
    }
}
exports.CheckClosingAccountResponseBody = CheckClosingAccountResponseBody;
class CheckClosingAccountResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: CheckClosingAccountResponseBody,
        };
    }
}
exports.CheckClosingAccountResponse = CheckClosingAccountResponse;
class CheckWritePermissionHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CheckWritePermissionHeaders = CheckWritePermissionHeaders;
class CheckWritePermissionRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            category: 'category',
            entityIds: 'entityIds',
            opUserId: 'opUserId',
            resourceKey: 'resourceKey',
        };
    }
    static types() {
        return {
            category: 'string',
            entityIds: { 'type': 'array', 'itemType': 'number' },
            opUserId: 'string',
            resourceKey: 'string',
        };
    }
}
exports.CheckWritePermissionRequest = CheckWritePermissionRequest;
class CheckWritePermissionResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            entityPermissionMap: 'entityPermissionMap',
        };
    }
    static types() {
        return {
            entityPermissionMap: { 'type': 'map', 'keyType': 'string', 'valueType': 'boolean' },
        };
    }
}
exports.CheckWritePermissionResponseBody = CheckWritePermissionResponseBody;
class CheckWritePermissionResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: CheckWritePermissionResponseBody,
        };
    }
}
exports.CheckWritePermissionResponse = CheckWritePermissionResponse;
class CreateApproveHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CreateApproveHeaders = CreateApproveHeaders;
class CreateApproveRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            approveId: 'approveId',
            opUserid: 'opUserid',
            punchParam: 'punchParam',
            subType: 'subType',
            tagName: 'tagName',
            userid: 'userid',
        };
    }
    static types() {
        return {
            approveId: 'string',
            opUserid: 'string',
            punchParam: CreateApproveRequestPunchParam,
            subType: 'string',
            tagName: 'string',
            userid: 'string',
        };
    }
}
exports.CreateApproveRequest = CreateApproveRequest;
class CreateApproveResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            dingtalkApproveId: 'dingtalkApproveId',
        };
    }
    static types() {
        return {
            dingtalkApproveId: 'string',
        };
    }
}
exports.CreateApproveResponseBody = CreateApproveResponseBody;
class CreateApproveResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: CreateApproveResponseBody,
        };
    }
}
exports.CreateApproveResponse = CreateApproveResponse;
class DeleteWaterMarkTemplateHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.DeleteWaterMarkTemplateHeaders = DeleteWaterMarkTemplateHeaders;
class DeleteWaterMarkTemplateRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            formCode: 'formCode',
            formContent: 'formContent',
            openConversationId: 'openConversationId',
            systemTemplate: 'systemTemplate',
            userId: 'userId',
        };
    }
    static types() {
        return {
            formCode: 'string',
            formContent: 'string',
            openConversationId: 'string',
            systemTemplate: 'boolean',
            userId: 'string',
        };
    }
}
exports.DeleteWaterMarkTemplateRequest = DeleteWaterMarkTemplateRequest;
class DeleteWaterMarkTemplateResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
}
exports.DeleteWaterMarkTemplateResponseBody = DeleteWaterMarkTemplateResponseBody;
class DeleteWaterMarkTemplateResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: DeleteWaterMarkTemplateResponseBody,
        };
    }
}
exports.DeleteWaterMarkTemplateResponse = DeleteWaterMarkTemplateResponse;
class DingTalkSecurityCheckHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.DingTalkSecurityCheckHeaders = DingTalkSecurityCheckHeaders;
class DingTalkSecurityCheckRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            clientVer: 'clientVer',
            platform: 'platform',
            platformVer: 'platformVer',
            sec: 'sec',
            userId: 'userId',
        };
    }
    static types() {
        return {
            clientVer: 'string',
            platform: 'string',
            platformVer: 'string',
            sec: 'string',
            userId: 'string',
        };
    }
}
exports.DingTalkSecurityCheckRequest = DingTalkSecurityCheckRequest;
class DingTalkSecurityCheckResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: DingTalkSecurityCheckResponseBodyResult,
        };
    }
}
exports.DingTalkSecurityCheckResponseBody = DingTalkSecurityCheckResponseBody;
class DingTalkSecurityCheckResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: DingTalkSecurityCheckResponseBody,
        };
    }
}
exports.DingTalkSecurityCheckResponse = DingTalkSecurityCheckResponse;
class GetAdjustmentsHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetAdjustmentsHeaders = GetAdjustmentsHeaders;
class GetAdjustmentsRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            pageNumber: 'pageNumber',
            pageSize: 'pageSize',
        };
    }
    static types() {
        return {
            pageNumber: 'number',
            pageSize: 'number',
        };
    }
}
exports.GetAdjustmentsRequest = GetAdjustmentsRequest;
class GetAdjustmentsResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': GetAdjustmentsResponseBodyResult },
        };
    }
}
exports.GetAdjustmentsResponseBody = GetAdjustmentsResponseBody;
class GetAdjustmentsResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetAdjustmentsResponseBody,
        };
    }
}
exports.GetAdjustmentsResponse = GetAdjustmentsResponse;
class GetCheckInSchemaTemplateHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetCheckInSchemaTemplateHeaders = GetCheckInSchemaTemplateHeaders;
class GetCheckInSchemaTemplateRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            bizCode: 'bizCode',
            openConversationId: 'openConversationId',
            sceneCode: 'sceneCode',
            userId: 'userId',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            openConversationId: 'string',
            sceneCode: 'string',
            userId: 'string',
        };
    }
}
exports.GetCheckInSchemaTemplateRequest = GetCheckInSchemaTemplateRequest;
class GetCheckInSchemaTemplateResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: GetCheckInSchemaTemplateResponseBodyResult,
        };
    }
}
exports.GetCheckInSchemaTemplateResponseBody = GetCheckInSchemaTemplateResponseBody;
class GetCheckInSchemaTemplateResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetCheckInSchemaTemplateResponseBody,
        };
    }
}
exports.GetCheckInSchemaTemplateResponse = GetCheckInSchemaTemplateResponse;
class GetClosingAccountsHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetClosingAccountsHeaders = GetClosingAccountsHeaders;
class GetClosingAccountsRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            userIds: 'userIds',
        };
    }
    static types() {
        return {
            userIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.GetClosingAccountsRequest = GetClosingAccountsRequest;
class GetClosingAccountsResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': GetClosingAccountsResponseBodyResult },
        };
    }
}
exports.GetClosingAccountsResponseBody = GetClosingAccountsResponseBody;
class GetClosingAccountsResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetClosingAccountsResponseBody,
        };
    }
}
exports.GetClosingAccountsResponse = GetClosingAccountsResponse;
class GetLeaveTypeHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetLeaveTypeHeaders = GetLeaveTypeHeaders;
class GetLeaveTypeRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            opUserId: 'opUserId',
            vacationSource: 'vacationSource',
        };
    }
    static types() {
        return {
            opUserId: 'string',
            vacationSource: 'string',
        };
    }
}
exports.GetLeaveTypeRequest = GetLeaveTypeRequest;
class GetLeaveTypeResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': GetLeaveTypeResponseBodyResult },
        };
    }
}
exports.GetLeaveTypeResponseBody = GetLeaveTypeResponseBody;
class GetLeaveTypeResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetLeaveTypeResponseBody,
        };
    }
}
exports.GetLeaveTypeResponse = GetLeaveTypeResponse;
class GetMachineHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetMachineHeaders = GetMachineHeaders;
class GetMachineResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: GetMachineResponseBodyResult,
        };
    }
}
exports.GetMachineResponseBody = GetMachineResponseBody;
class GetMachineResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetMachineResponseBody,
        };
    }
}
exports.GetMachineResponse = GetMachineResponse;
class GetMachineUserHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetMachineUserHeaders = GetMachineUserHeaders;
class GetMachineUserRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            maxResults: 'maxResults',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            maxResults: 'number',
            nextToken: 'string',
        };
    }
}
exports.GetMachineUserRequest = GetMachineUserRequest;
class GetMachineUserResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: GetMachineUserResponseBodyResult,
        };
    }
}
exports.GetMachineUserResponseBody = GetMachineUserResponseBody;
class GetMachineUserResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetMachineUserResponseBody,
        };
    }
}
exports.GetMachineUserResponse = GetMachineUserResponse;
class GetOvertimeSettingHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetOvertimeSettingHeaders = GetOvertimeSettingHeaders;
class GetOvertimeSettingRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            overtimeSettingIds: 'overtimeSettingIds',
        };
    }
    static types() {
        return {
            overtimeSettingIds: { 'type': 'array', 'itemType': 'number' },
        };
    }
}
exports.GetOvertimeSettingRequest = GetOvertimeSettingRequest;
class GetOvertimeSettingResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': GetOvertimeSettingResponseBodyResult },
        };
    }
}
exports.GetOvertimeSettingResponseBody = GetOvertimeSettingResponseBody;
class GetOvertimeSettingResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetOvertimeSettingResponseBody,
        };
    }
}
exports.GetOvertimeSettingResponse = GetOvertimeSettingResponse;
class GetSimpleOvertimeSettingHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetSimpleOvertimeSettingHeaders = GetSimpleOvertimeSettingHeaders;
class GetSimpleOvertimeSettingRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            pageNumber: 'pageNumber',
            pageSize: 'pageSize',
        };
    }
    static types() {
        return {
            pageNumber: 'number',
            pageSize: 'number',
        };
    }
}
exports.GetSimpleOvertimeSettingRequest = GetSimpleOvertimeSettingRequest;
class GetSimpleOvertimeSettingResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': GetSimpleOvertimeSettingResponseBodyResult },
        };
    }
}
exports.GetSimpleOvertimeSettingResponseBody = GetSimpleOvertimeSettingResponseBody;
class GetSimpleOvertimeSettingResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetSimpleOvertimeSettingResponseBody,
        };
    }
}
exports.GetSimpleOvertimeSettingResponse = GetSimpleOvertimeSettingResponse;
class GetUserHolidaysHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetUserHolidaysHeaders = GetUserHolidaysHeaders;
class GetUserHolidaysRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            userIds: 'userIds',
            workDateFrom: 'workDateFrom',
            workDateTo: 'workDateTo',
        };
    }
    static types() {
        return {
            userIds: { 'type': 'array', 'itemType': 'string' },
            workDateFrom: 'number',
            workDateTo: 'number',
        };
    }
}
exports.GetUserHolidaysRequest = GetUserHolidaysRequest;
class GetUserHolidaysResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': GetUserHolidaysResponseBodyResult },
        };
    }
}
exports.GetUserHolidaysResponseBody = GetUserHolidaysResponseBody;
class GetUserHolidaysResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetUserHolidaysResponseBody,
        };
    }
}
exports.GetUserHolidaysResponse = GetUserHolidaysResponse;
class InitAndGetLeaveALlocationQuotasHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.InitAndGetLeaveALlocationQuotasHeaders = InitAndGetLeaveALlocationQuotasHeaders;
class InitAndGetLeaveALlocationQuotasRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            leaveCode: 'leaveCode',
            opUserId: 'opUserId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            leaveCode: 'string',
            opUserId: 'string',
            userId: 'string',
        };
    }
}
exports.InitAndGetLeaveALlocationQuotasRequest = InitAndGetLeaveALlocationQuotasRequest;
class InitAndGetLeaveALlocationQuotasResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': InitAndGetLeaveALlocationQuotasResponseBodyResult },
        };
    }
}
exports.InitAndGetLeaveALlocationQuotasResponseBody = InitAndGetLeaveALlocationQuotasResponseBody;
class InitAndGetLeaveALlocationQuotasResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: InitAndGetLeaveALlocationQuotasResponseBody,
        };
    }
}
exports.InitAndGetLeaveALlocationQuotasResponse = InitAndGetLeaveALlocationQuotasResponse;
class ModifyWaterMarkTemplateHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ModifyWaterMarkTemplateHeaders = ModifyWaterMarkTemplateHeaders;
class ModifyWaterMarkTemplateRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            formCode: 'formCode',
            icon: 'icon',
            layoutDesignId: 'layoutDesignId',
            schemaContent: 'schemaContent',
            title: 'title',
            waterMarkId: 'waterMarkId',
            openConversationId: 'openConversationId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            formCode: 'string',
            icon: 'string',
            layoutDesignId: 'string',
            schemaContent: 'string',
            title: 'string',
            waterMarkId: 'string',
            openConversationId: 'string',
            userId: 'string',
        };
    }
}
exports.ModifyWaterMarkTemplateRequest = ModifyWaterMarkTemplateRequest;
class ModifyWaterMarkTemplateResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
}
exports.ModifyWaterMarkTemplateResponseBody = ModifyWaterMarkTemplateResponseBody;
class ModifyWaterMarkTemplateResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: ModifyWaterMarkTemplateResponseBody,
        };
    }
}
exports.ModifyWaterMarkTemplateResponse = ModifyWaterMarkTemplateResponse;
class ProcessApproveCreateHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ProcessApproveCreateHeaders = ProcessApproveCreateHeaders;
class ProcessApproveCreateRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            approveId: 'approveId',
            opUserId: 'opUserId',
            punchParam: 'punchParam',
            subType: 'subType',
            tagName: 'tagName',
            userId: 'userId',
        };
    }
    static types() {
        return {
            approveId: 'string',
            opUserId: 'string',
            punchParam: ProcessApproveCreateRequestPunchParam,
            subType: 'string',
            tagName: 'string',
            userId: 'string',
        };
    }
}
exports.ProcessApproveCreateRequest = ProcessApproveCreateRequest;
class ProcessApproveCreateResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: ProcessApproveCreateResponseBodyResult,
        };
    }
}
exports.ProcessApproveCreateResponseBody = ProcessApproveCreateResponseBody;
class ProcessApproveCreateResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: ProcessApproveCreateResponseBody,
        };
    }
}
exports.ProcessApproveCreateResponse = ProcessApproveCreateResponse;
class SaveCustomWaterMarkTemplateHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SaveCustomWaterMarkTemplateHeaders = SaveCustomWaterMarkTemplateHeaders;
class SaveCustomWaterMarkTemplateRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            bizCode: 'bizCode',
            icon: 'icon',
            layoutDesignId: 'layoutDesignId',
            sceneCode: 'sceneCode',
            schemaContent: 'schemaContent',
            title: 'title',
            openConversationId: 'openConversationId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            icon: 'string',
            layoutDesignId: 'string',
            sceneCode: 'string',
            schemaContent: 'string',
            title: 'string',
            openConversationId: 'string',
            userId: 'string',
        };
    }
}
exports.SaveCustomWaterMarkTemplateRequest = SaveCustomWaterMarkTemplateRequest;
class SaveCustomWaterMarkTemplateResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: SaveCustomWaterMarkTemplateResponseBodyResult,
        };
    }
}
exports.SaveCustomWaterMarkTemplateResponseBody = SaveCustomWaterMarkTemplateResponseBody;
class SaveCustomWaterMarkTemplateResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: SaveCustomWaterMarkTemplateResponseBody,
        };
    }
}
exports.SaveCustomWaterMarkTemplateResponse = SaveCustomWaterMarkTemplateResponse;
class SyncScheduleInfoHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SyncScheduleInfoHeaders = SyncScheduleInfoHeaders;
class SyncScheduleInfoRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            opUserId: 'opUserId',
            scheduleInfos: 'scheduleInfos',
        };
    }
    static types() {
        return {
            opUserId: 'string',
            scheduleInfos: { 'type': 'array', 'itemType': SyncScheduleInfoRequestScheduleInfos },
        };
    }
}
exports.SyncScheduleInfoRequest = SyncScheduleInfoRequest;
class SyncScheduleInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.SyncScheduleInfoResponse = SyncScheduleInfoResponse;
class UpdateLeaveTypeHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateLeaveTypeHeaders = UpdateLeaveTypeHeaders;
class UpdateLeaveTypeRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            bizType: 'bizType',
            extras: 'extras',
            hoursInPerDay: 'hoursInPerDay',
            leaveCertificate: 'leaveCertificate',
            leaveCode: 'leaveCode',
            leaveName: 'leaveName',
            leaveViewUnit: 'leaveViewUnit',
            naturalDayLeave: 'naturalDayLeave',
            submitTimeRule: 'submitTimeRule',
            visibilityRules: 'visibilityRules',
            opUserId: 'opUserId',
        };
    }
    static types() {
        return {
            bizType: 'string',
            extras: 'string',
            hoursInPerDay: 'number',
            leaveCertificate: UpdateLeaveTypeRequestLeaveCertificate,
            leaveCode: 'string',
            leaveName: 'string',
            leaveViewUnit: 'string',
            naturalDayLeave: 'boolean',
            submitTimeRule: UpdateLeaveTypeRequestSubmitTimeRule,
            visibilityRules: { 'type': 'array', 'itemType': UpdateLeaveTypeRequestVisibilityRules },
            opUserId: 'string',
        };
    }
}
exports.UpdateLeaveTypeRequest = UpdateLeaveTypeRequest;
class UpdateLeaveTypeResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: UpdateLeaveTypeResponseBodyResult,
        };
    }
}
exports.UpdateLeaveTypeResponseBody = UpdateLeaveTypeResponseBody;
class UpdateLeaveTypeResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: UpdateLeaveTypeResponseBody,
        };
    }
}
exports.UpdateLeaveTypeResponse = UpdateLeaveTypeResponse;
class ResultDurationSettingsValue extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            calcType: 'calcType',
            durationType: 'durationType',
            overtimeRedress: 'overtimeRedress',
            settings: 'settings',
            overtimeRedressBy: 'overtimeRedressBy',
            vacationRate: 'vacationRate',
            skipTime: 'skipTime',
            skipTimeByFrames: 'skipTimeByFrames',
            skipTimeByDurations: 'skipTimeByDurations',
            holidayPlanOvertimeRedress: 'holidayPlanOvertimeRedress',
            holidayPlanOvertimeRedressBy: 'holidayPlanOvertimeRedressBy',
            holidayPlanVacationRate: 'holidayPlanVacationRate',
        };
    }
    static types() {
        return {
            calcType: 'number',
            durationType: 'number',
            overtimeRedress: 'boolean',
            settings: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            overtimeRedressBy: 'string',
            vacationRate: 'number',
            skipTime: 'string',
            skipTimeByFrames: { 'type': 'array', 'itemType': ResultDurationSettingsValueSkipTimeByFrames },
            skipTimeByDurations: { 'type': 'array', 'itemType': ResultDurationSettingsValueSkipTimeByDurations },
            holidayPlanOvertimeRedress: 'boolean',
            holidayPlanOvertimeRedressBy: 'string',
            holidayPlanVacationRate: 'number',
        };
    }
}
exports.ResultDurationSettingsValue = ResultDurationSettingsValue;
class AddLeaveTypeRequestLeaveCertificate extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            duration: 'duration',
            enable: 'enable',
            promptInformation: 'promptInformation',
            unit: 'unit',
        };
    }
    static types() {
        return {
            duration: 'number',
            enable: 'boolean',
            promptInformation: 'string',
            unit: 'string',
        };
    }
}
exports.AddLeaveTypeRequestLeaveCertificate = AddLeaveTypeRequestLeaveCertificate;
class AddLeaveTypeRequestSubmitTimeRule extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            enableTimeLimit: 'enableTimeLimit',
            timeType: 'timeType',
            timeUnit: 'timeUnit',
            timeValue: 'timeValue',
        };
    }
    static types() {
        return {
            enableTimeLimit: 'boolean',
            timeType: 'string',
            timeUnit: 'string',
            timeValue: 'number',
        };
    }
}
exports.AddLeaveTypeRequestSubmitTimeRule = AddLeaveTypeRequestSubmitTimeRule;
class AddLeaveTypeRequestVisibilityRules extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            type: 'type',
            visible: 'visible',
        };
    }
    static types() {
        return {
            type: 'string',
            visible: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.AddLeaveTypeRequestVisibilityRules = AddLeaveTypeRequestVisibilityRules;
class AddLeaveTypeResponseBodyResultLeaveCertificate extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            duration: 'duration',
            enable: 'enable',
            promptInformation: 'promptInformation',
            unit: 'unit',
        };
    }
    static types() {
        return {
            duration: 'number',
            enable: 'boolean',
            promptInformation: 'string',
            unit: 'string',
        };
    }
}
exports.AddLeaveTypeResponseBodyResultLeaveCertificate = AddLeaveTypeResponseBodyResultLeaveCertificate;
class AddLeaveTypeResponseBodyResultSubmitTimeRule extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            enableTimeLimit: 'enableTimeLimit',
            timeType: 'timeType',
            timeUnit: 'timeUnit',
            timeValue: 'timeValue',
        };
    }
    static types() {
        return {
            enableTimeLimit: 'boolean',
            timeType: 'string',
            timeUnit: 'string',
            timeValue: 'number',
        };
    }
}
exports.AddLeaveTypeResponseBodyResultSubmitTimeRule = AddLeaveTypeResponseBodyResultSubmitTimeRule;
class AddLeaveTypeResponseBodyResultVisibilityRules extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            type: 'type',
            visible: 'visible',
        };
    }
    static types() {
        return {
            type: 'string',
            visible: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.AddLeaveTypeResponseBodyResultVisibilityRules = AddLeaveTypeResponseBodyResultVisibilityRules;
class AddLeaveTypeResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            bizType: 'bizType',
            hoursInPerDay: 'hoursInPerDay',
            leaveCertificate: 'leaveCertificate',
            leaveCode: 'leaveCode',
            leaveName: 'leaveName',
            leaveViewUnit: 'leaveViewUnit',
            naturalDayLeave: 'naturalDayLeave',
            submitTimeRule: 'submitTimeRule',
            visibilityRules: 'visibilityRules',
        };
    }
    static types() {
        return {
            bizType: 'string',
            hoursInPerDay: 'number',
            leaveCertificate: AddLeaveTypeResponseBodyResultLeaveCertificate,
            leaveCode: 'string',
            leaveName: 'string',
            leaveViewUnit: 'string',
            naturalDayLeave: 'boolean',
            submitTimeRule: AddLeaveTypeResponseBodyResultSubmitTimeRule,
            visibilityRules: { 'type': 'array', 'itemType': AddLeaveTypeResponseBodyResultVisibilityRules },
        };
    }
}
exports.AddLeaveTypeResponseBodyResult = AddLeaveTypeResponseBodyResult;
class AttendanceBleDevicesAddResponseBodyErrorListFailureList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deviceId: 'deviceId',
            deviceName: 'deviceName',
            sn: 'sn',
        };
    }
    static types() {
        return {
            deviceId: 'number',
            deviceName: 'string',
            sn: 'string',
        };
    }
}
exports.AttendanceBleDevicesAddResponseBodyErrorListFailureList = AttendanceBleDevicesAddResponseBodyErrorListFailureList;
class AttendanceBleDevicesAddResponseBodyErrorList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            code: 'code',
            failureList: 'failureList',
            msg: 'msg',
        };
    }
    static types() {
        return {
            code: 'string',
            failureList: { 'type': 'array', 'itemType': AttendanceBleDevicesAddResponseBodyErrorListFailureList },
            msg: 'string',
        };
    }
}
exports.AttendanceBleDevicesAddResponseBodyErrorList = AttendanceBleDevicesAddResponseBodyErrorList;
class AttendanceBleDevicesAddResponseBodySuccessList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deviceId: 'deviceId',
            deviceName: 'deviceName',
            sn: 'sn',
        };
    }
    static types() {
        return {
            deviceId: 'number',
            deviceName: 'string',
            sn: 'string',
        };
    }
}
exports.AttendanceBleDevicesAddResponseBodySuccessList = AttendanceBleDevicesAddResponseBodySuccessList;
class AttendanceBleDevicesQueryResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deviceId: 'deviceId',
            deviceName: 'deviceName',
            sn: 'sn',
        };
    }
    static types() {
        return {
            deviceId: 'number',
            deviceName: 'string',
            sn: 'string',
        };
    }
}
exports.AttendanceBleDevicesQueryResponseBodyResult = AttendanceBleDevicesQueryResponseBodyResult;
class AttendanceBleDevicesRemoveResponseBodyErrorList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            code: 'code',
            failureList: 'failureList',
            msg: 'msg',
        };
    }
    static types() {
        return {
            code: 'string',
            failureList: { 'type': 'array', 'itemType': 'number' },
            msg: 'string',
        };
    }
}
exports.AttendanceBleDevicesRemoveResponseBodyErrorList = AttendanceBleDevicesRemoveResponseBodyErrorList;
class CheckClosingAccountRequestUserTimeRange extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            endTime: 'endTime',
            startTime: 'startTime',
        };
    }
    static types() {
        return {
            endTime: 'number',
            startTime: 'number',
        };
    }
}
exports.CheckClosingAccountRequestUserTimeRange = CheckClosingAccountRequestUserTimeRange;
class CreateApproveRequestPunchParam extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            positionId: 'positionId',
            positionName: 'positionName',
            positionType: 'positionType',
            punchTime: 'punchTime',
        };
    }
    static types() {
        return {
            positionId: 'string',
            positionName: 'string',
            positionType: 'string',
            punchTime: 'number',
        };
    }
}
exports.CreateApproveRequestPunchParam = CreateApproveRequestPunchParam;
class DingTalkSecurityCheckResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            hasRisk: 'hasRisk',
            riskInfo: 'riskInfo',
        };
    }
    static types() {
        return {
            hasRisk: 'boolean',
            riskInfo: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.DingTalkSecurityCheckResponseBodyResult = DingTalkSecurityCheckResponseBodyResult;
class GetAdjustmentsResponseBodyResultItems extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            id: 'id',
            name: 'name',
        };
    }
    static types() {
        return {
            id: 'number',
            name: 'string',
        };
    }
}
exports.GetAdjustmentsResponseBodyResultItems = GetAdjustmentsResponseBodyResultItems;
class GetAdjustmentsResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            items: 'items',
            pageNumber: 'pageNumber',
            totalPage: 'totalPage',
        };
    }
    static types() {
        return {
            items: { 'type': 'array', 'itemType': GetAdjustmentsResponseBodyResultItems },
            pageNumber: 'number',
            totalPage: 'number',
        };
    }
}
exports.GetAdjustmentsResponseBodyResult = GetAdjustmentsResponseBodyResult;
class GetCheckInSchemaTemplateResponseBodyResultWaterMarkTemplateModels extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            canModify: 'canModify',
            formCode: 'formCode',
            icon: 'icon',
            layoutDesign: 'layoutDesign',
            sceneCode: 'sceneCode',
            schemaContent: 'schemaContent',
            suiteKey: 'suiteKey',
            systemTemplate: 'systemTemplate',
            title: 'title',
            waterMarkId: 'waterMarkId',
        };
    }
    static types() {
        return {
            canModify: 'boolean',
            formCode: 'string',
            icon: 'string',
            layoutDesign: 'string',
            sceneCode: 'string',
            schemaContent: 'string',
            suiteKey: 'string',
            systemTemplate: 'boolean',
            title: 'string',
            waterMarkId: 'string',
        };
    }
}
exports.GetCheckInSchemaTemplateResponseBodyResultWaterMarkTemplateModels = GetCheckInSchemaTemplateResponseBodyResultWaterMarkTemplateModels;
class GetCheckInSchemaTemplateResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            bizCode: 'bizCode',
            canModifyAndAddTemplate: 'canModifyAndAddTemplate',
            conversationAdmin: 'conversationAdmin',
            customTemplateMaxSize: 'customTemplateMaxSize',
            openConversationId: 'openConversationId',
            showStat: 'showStat',
            templateDegrade: 'templateDegrade',
            waterMarkTemplateModels: 'waterMarkTemplateModels',
        };
    }
    static types() {
        return {
            bizCode: 'string',
            canModifyAndAddTemplate: 'boolean',
            conversationAdmin: 'boolean',
            customTemplateMaxSize: 'number',
            openConversationId: 'string',
            showStat: 'boolean',
            templateDegrade: 'boolean',
            waterMarkTemplateModels: { 'type': 'array', 'itemType': GetCheckInSchemaTemplateResponseBodyResultWaterMarkTemplateModels },
        };
    }
}
exports.GetCheckInSchemaTemplateResponseBodyResult = GetCheckInSchemaTemplateResponseBodyResult;
class GetClosingAccountsResponseBodyResultClosingAccountModel extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            closingDay: 'closingDay',
            closingHourMinutes: 'closingHourMinutes',
            endDay: 'endDay',
            endMonth: 'endMonth',
            startDay: 'startDay',
            startMonth: 'startMonth',
        };
    }
    static types() {
        return {
            closingDay: 'number',
            closingHourMinutes: 'number',
            endDay: 'number',
            endMonth: 'number',
            startDay: 'number',
            startMonth: 'number',
        };
    }
}
exports.GetClosingAccountsResponseBodyResultClosingAccountModel = GetClosingAccountsResponseBodyResultClosingAccountModel;
class GetClosingAccountsResponseBodyResultUnsealClosingAccountModel extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            invalidTimeStamp: 'invalidTimeStamp',
        };
    }
    static types() {
        return {
            invalidTimeStamp: 'number',
        };
    }
}
exports.GetClosingAccountsResponseBodyResultUnsealClosingAccountModel = GetClosingAccountsResponseBodyResultUnsealClosingAccountModel;
class GetClosingAccountsResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            closingAccountModel: 'closingAccountModel',
            switchOn: 'switchOn',
            unsealClosingAccountModel: 'unsealClosingAccountModel',
            userId: 'userId',
        };
    }
    static types() {
        return {
            closingAccountModel: GetClosingAccountsResponseBodyResultClosingAccountModel,
            switchOn: 'boolean',
            unsealClosingAccountModel: GetClosingAccountsResponseBodyResultUnsealClosingAccountModel,
            userId: 'string',
        };
    }
}
exports.GetClosingAccountsResponseBodyResult = GetClosingAccountsResponseBodyResult;
class GetLeaveTypeResponseBodyResultLeaveCertificate extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            duration: 'duration',
            enable: 'enable',
            promptInformation: 'promptInformation',
            unit: 'unit',
        };
    }
    static types() {
        return {
            duration: 'number',
            enable: 'boolean',
            promptInformation: 'string',
            unit: 'string',
        };
    }
}
exports.GetLeaveTypeResponseBodyResultLeaveCertificate = GetLeaveTypeResponseBodyResultLeaveCertificate;
class GetLeaveTypeResponseBodyResultSubmitTimeRule extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            enableTimeLimit: 'enableTimeLimit',
            timeType: 'timeType',
            timeUnit: 'timeUnit',
            timeValue: 'timeValue',
        };
    }
    static types() {
        return {
            enableTimeLimit: 'boolean',
            timeType: 'string',
            timeUnit: 'string',
            timeValue: 'number',
        };
    }
}
exports.GetLeaveTypeResponseBodyResultSubmitTimeRule = GetLeaveTypeResponseBodyResultSubmitTimeRule;
class GetLeaveTypeResponseBodyResultVisibilityRules extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            type: 'type',
            visible: 'visible',
        };
    }
    static types() {
        return {
            type: 'string',
            visible: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.GetLeaveTypeResponseBodyResultVisibilityRules = GetLeaveTypeResponseBodyResultVisibilityRules;
class GetLeaveTypeResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            bizType: 'bizType',
            hoursInPerDay: 'hoursInPerDay',
            leaveCertificate: 'leaveCertificate',
            leaveCode: 'leaveCode',
            leaveName: 'leaveName',
            leaveViewUnit: 'leaveViewUnit',
            naturalDayLeave: 'naturalDayLeave',
            source: 'source',
            submitTimeRule: 'submitTimeRule',
            validityType: 'validityType',
            validityValue: 'validityValue',
            visibilityRules: 'visibilityRules',
        };
    }
    static types() {
        return {
            bizType: 'string',
            hoursInPerDay: 'number',
            leaveCertificate: GetLeaveTypeResponseBodyResultLeaveCertificate,
            leaveCode: 'string',
            leaveName: 'string',
            leaveViewUnit: 'string',
            naturalDayLeave: 'boolean',
            source: 'string',
            submitTimeRule: GetLeaveTypeResponseBodyResultSubmitTimeRule,
            validityType: 'string',
            validityValue: 'string',
            visibilityRules: { 'type': 'array', 'itemType': GetLeaveTypeResponseBodyResultVisibilityRules },
        };
    }
}
exports.GetLeaveTypeResponseBodyResult = GetLeaveTypeResponseBodyResult;
class GetMachineResponseBodyResultMachineBluetoothVO extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            address: 'address',
            bluetoothCheckWithFace: 'bluetoothCheckWithFace',
            bluetoothDistanceMode: 'bluetoothDistanceMode',
            bluetoothDistanceModeDesc: 'bluetoothDistanceModeDesc',
            bluetoothValue: 'bluetoothValue',
            latitude: 'latitude',
            limitUserDeviceCount: 'limitUserDeviceCount',
            longitude: 'longitude',
            monitorLocationAbnormal: 'monitorLocationAbnormal',
            userDeviceCount: 'userDeviceCount',
        };
    }
    static types() {
        return {
            address: 'string',
            bluetoothCheckWithFace: 'boolean',
            bluetoothDistanceMode: 'string',
            bluetoothDistanceModeDesc: 'string',
            bluetoothValue: 'boolean',
            latitude: 'number',
            limitUserDeviceCount: 'boolean',
            longitude: 'number',
            monitorLocationAbnormal: 'boolean',
            userDeviceCount: 'number',
        };
    }
}
exports.GetMachineResponseBodyResultMachineBluetoothVO = GetMachineResponseBodyResultMachineBluetoothVO;
class GetMachineResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            atmManagerList: 'atmManagerList',
            devId: 'devId',
            deviceId: 'deviceId',
            deviceName: 'deviceName',
            deviceSn: 'deviceSn',
            machineBluetoothVO: 'machineBluetoothVO',
            maxFace: 'maxFace',
            netStatus: 'netStatus',
            productName: 'productName',
            productVersion: 'productVersion',
            voiceMode: 'voiceMode',
        };
    }
    static types() {
        return {
            atmManagerList: { 'type': 'array', 'itemType': 'string' },
            devId: 'number',
            deviceId: 'string',
            deviceName: 'string',
            deviceSn: 'string',
            machineBluetoothVO: GetMachineResponseBodyResultMachineBluetoothVO,
            maxFace: 'number',
            netStatus: 'string',
            productName: 'string',
            productVersion: 'string',
            voiceMode: 'number',
        };
    }
}
exports.GetMachineResponseBodyResult = GetMachineResponseBodyResult;
class GetMachineUserResponseBodyResultUserList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            hasFace: 'hasFace',
            name: 'name',
            userId: 'userId',
        };
    }
    static types() {
        return {
            hasFace: 'boolean',
            name: 'string',
            userId: 'string',
        };
    }
}
exports.GetMachineUserResponseBodyResultUserList = GetMachineUserResponseBodyResultUserList;
class GetMachineUserResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            hasMore: 'hasMore',
            nextToken: 'nextToken',
            userList: 'userList',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            nextToken: 'string',
            userList: { 'type': 'array', 'itemType': GetMachineUserResponseBodyResultUserList },
        };
    }
}
exports.GetMachineUserResponseBodyResult = GetMachineUserResponseBodyResult;
class GetOvertimeSettingResponseBodyResultOvertimeDivisions extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            nextDayType: 'nextDayType',
            previousDayType: 'previousDayType',
            timeSplitPoint: 'timeSplitPoint',
        };
    }
    static types() {
        return {
            nextDayType: 'string',
            previousDayType: 'string',
            timeSplitPoint: 'string',
        };
    }
}
exports.GetOvertimeSettingResponseBodyResultOvertimeDivisions = GetOvertimeSettingResponseBodyResultOvertimeDivisions;
class GetOvertimeSettingResponseBodyResultWarningSettings extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            action: 'action',
            threshold: 'threshold',
            time: 'time',
        };
    }
    static types() {
        return {
            action: 'string',
            threshold: 'number',
            time: 'string',
        };
    }
}
exports.GetOvertimeSettingResponseBodyResultWarningSettings = GetOvertimeSettingResponseBodyResultWarningSettings;
class GetOvertimeSettingResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            default: 'default',
            durationSettings: 'durationSettings',
            id: 'id',
            name: 'name',
            overtimeDivisions: 'overtimeDivisions',
            settingId: 'settingId',
            stepType: 'stepType',
            stepValue: 'stepValue',
            warningSettings: 'warningSettings',
            workMinutesPerDay: 'workMinutesPerDay',
        };
    }
    static types() {
        return {
            default: 'boolean',
            durationSettings: { 'type': 'map', 'keyType': 'string', 'valueType': ResultDurationSettingsValue },
            id: 'number',
            name: 'string',
            overtimeDivisions: { 'type': 'array', 'itemType': GetOvertimeSettingResponseBodyResultOvertimeDivisions },
            settingId: 'number',
            stepType: 'number',
            stepValue: 'number',
            warningSettings: { 'type': 'array', 'itemType': GetOvertimeSettingResponseBodyResultWarningSettings },
            workMinutesPerDay: 'number',
        };
    }
}
exports.GetOvertimeSettingResponseBodyResult = GetOvertimeSettingResponseBodyResult;
class GetSimpleOvertimeSettingResponseBodyResultItems extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            id: 'id',
            name: 'name',
        };
    }
    static types() {
        return {
            id: 'number',
            name: 'string',
        };
    }
}
exports.GetSimpleOvertimeSettingResponseBodyResultItems = GetSimpleOvertimeSettingResponseBodyResultItems;
class GetSimpleOvertimeSettingResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            items: 'items',
            pageNumber: 'pageNumber',
            totalPage: 'totalPage',
        };
    }
    static types() {
        return {
            items: { 'type': 'array', 'itemType': GetSimpleOvertimeSettingResponseBodyResultItems },
            pageNumber: 'number',
            totalPage: 'number',
        };
    }
}
exports.GetSimpleOvertimeSettingResponseBodyResult = GetSimpleOvertimeSettingResponseBodyResult;
class GetUserHolidaysResponseBodyResultHolidays extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            holidayName: 'holidayName',
            holidayType: 'holidayType',
            realWorkDate: 'realWorkDate',
            workDate: 'workDate',
        };
    }
    static types() {
        return {
            holidayName: 'string',
            holidayType: 'string',
            realWorkDate: 'number',
            workDate: 'number',
        };
    }
}
exports.GetUserHolidaysResponseBodyResultHolidays = GetUserHolidaysResponseBodyResultHolidays;
class GetUserHolidaysResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            holidays: 'holidays',
            userId: 'userId',
        };
    }
    static types() {
        return {
            holidays: { 'type': 'array', 'itemType': GetUserHolidaysResponseBodyResultHolidays },
            userId: 'string',
        };
    }
}
exports.GetUserHolidaysResponseBodyResult = GetUserHolidaysResponseBodyResult;
class InitAndGetLeaveALlocationQuotasResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            endTime: 'endTime',
            leaveCode: 'leaveCode',
            quotaCycle: 'quotaCycle',
            quotaId: 'quotaId',
            quotaNumPerDay: 'quotaNumPerDay',
            quotaNumPerHour: 'quotaNumPerHour',
            startTime: 'startTime',
            usedNumPerDay: 'usedNumPerDay',
            usedNumPerHour: 'usedNumPerHour',
            userId: 'userId',
        };
    }
    static types() {
        return {
            endTime: 'number',
            leaveCode: 'string',
            quotaCycle: 'string',
            quotaId: 'string',
            quotaNumPerDay: 'number',
            quotaNumPerHour: 'number',
            startTime: 'number',
            usedNumPerDay: 'number',
            usedNumPerHour: 'number',
            userId: 'string',
        };
    }
}
exports.InitAndGetLeaveALlocationQuotasResponseBodyResult = InitAndGetLeaveALlocationQuotasResponseBodyResult;
class ProcessApproveCreateRequestPunchParam extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            positionId: 'positionId',
            positionName: 'positionName',
            positionType: 'positionType',
            punchTime: 'punchTime',
        };
    }
    static types() {
        return {
            positionId: 'string',
            positionName: 'string',
            positionType: 'string',
            punchTime: 'number',
        };
    }
}
exports.ProcessApproveCreateRequestPunchParam = ProcessApproveCreateRequestPunchParam;
class ProcessApproveCreateResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            dingtalkApproveId: 'dingtalkApproveId',
        };
    }
    static types() {
        return {
            dingtalkApproveId: 'string',
        };
    }
}
exports.ProcessApproveCreateResponseBodyResult = ProcessApproveCreateResponseBodyResult;
class SaveCustomWaterMarkTemplateResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            formCode: 'formCode',
            waterMarkId: 'waterMarkId',
        };
    }
    static types() {
        return {
            formCode: 'string',
            waterMarkId: 'string',
        };
    }
}
exports.SaveCustomWaterMarkTemplateResponseBodyResult = SaveCustomWaterMarkTemplateResponseBodyResult;
class SyncScheduleInfoRequestScheduleInfos extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            planId: 'planId',
            positionKeys: 'positionKeys',
            retainAttendanceCheck: 'retainAttendanceCheck',
            wifiKeys: 'wifiKeys',
        };
    }
    static types() {
        return {
            planId: 'number',
            positionKeys: { 'type': 'array', 'itemType': 'string' },
            retainAttendanceCheck: 'boolean',
            wifiKeys: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.SyncScheduleInfoRequestScheduleInfos = SyncScheduleInfoRequestScheduleInfos;
class UpdateLeaveTypeRequestLeaveCertificate extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            duration: 'duration',
            enable: 'enable',
            promptInformation: 'promptInformation',
            unit: 'unit',
        };
    }
    static types() {
        return {
            duration: 'number',
            enable: 'boolean',
            promptInformation: 'string',
            unit: 'string',
        };
    }
}
exports.UpdateLeaveTypeRequestLeaveCertificate = UpdateLeaveTypeRequestLeaveCertificate;
class UpdateLeaveTypeRequestSubmitTimeRule extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            enableTimeLimit: 'enableTimeLimit',
            timeType: 'timeType',
            timeUnit: 'timeUnit',
            timeValue: 'timeValue',
        };
    }
    static types() {
        return {
            enableTimeLimit: 'boolean',
            timeType: 'string',
            timeUnit: 'string',
            timeValue: 'number',
        };
    }
}
exports.UpdateLeaveTypeRequestSubmitTimeRule = UpdateLeaveTypeRequestSubmitTimeRule;
class UpdateLeaveTypeRequestVisibilityRules extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            type: 'type',
            visible: 'visible',
        };
    }
    static types() {
        return {
            type: 'string',
            visible: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.UpdateLeaveTypeRequestVisibilityRules = UpdateLeaveTypeRequestVisibilityRules;
class UpdateLeaveTypeResponseBodyResultLeaveCertificate extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            duration: 'duration',
            enable: 'enable',
            promptInformation: 'promptInformation',
            unit: 'unit',
        };
    }
    static types() {
        return {
            duration: 'number',
            enable: 'boolean',
            promptInformation: 'string',
            unit: 'string',
        };
    }
}
exports.UpdateLeaveTypeResponseBodyResultLeaveCertificate = UpdateLeaveTypeResponseBodyResultLeaveCertificate;
class UpdateLeaveTypeResponseBodyResultSubmitTimeRule extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            enableTimeLimit: 'enableTimeLimit',
            timeType: 'timeType',
            timeUnit: 'timeUnit',
            timeValue: 'timeValue',
        };
    }
    static types() {
        return {
            enableTimeLimit: 'boolean',
            timeType: 'string',
            timeUnit: 'string',
            timeValue: 'number',
        };
    }
}
exports.UpdateLeaveTypeResponseBodyResultSubmitTimeRule = UpdateLeaveTypeResponseBodyResultSubmitTimeRule;
class UpdateLeaveTypeResponseBodyResultVisibilityRules extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            type: 'type',
            visible: 'visible',
        };
    }
    static types() {
        return {
            type: 'string',
            visible: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.UpdateLeaveTypeResponseBodyResultVisibilityRules = UpdateLeaveTypeResponseBodyResultVisibilityRules;
class UpdateLeaveTypeResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            bizType: 'bizType',
            hoursInPerDay: 'hoursInPerDay',
            leaveCertificate: 'leaveCertificate',
            leaveCode: 'leaveCode',
            leaveName: 'leaveName',
            leaveViewUnit: 'leaveViewUnit',
            naturalDayLeave: 'naturalDayLeave',
            submitTimeRule: 'submitTimeRule',
            visibilityRules: 'visibilityRules',
        };
    }
    static types() {
        return {
            bizType: 'string',
            hoursInPerDay: 'number',
            leaveCertificate: UpdateLeaveTypeResponseBodyResultLeaveCertificate,
            leaveCode: 'string',
            leaveName: 'string',
            leaveViewUnit: 'string',
            naturalDayLeave: 'boolean',
            submitTimeRule: UpdateLeaveTypeResponseBodyResultSubmitTimeRule,
            visibilityRules: { 'type': 'array', 'itemType': UpdateLeaveTypeResponseBodyResultVisibilityRules },
        };
    }
}
exports.UpdateLeaveTypeResponseBodyResult = UpdateLeaveTypeResponseBodyResult;
class ResultDurationSettingsValueSkipTimeByFrames extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            startTime: 'startTime',
            endTime: 'endTime',
            valid: 'valid',
        };
    }
    static types() {
        return {
            startTime: 'string',
            endTime: 'string',
            valid: 'boolean',
        };
    }
}
exports.ResultDurationSettingsValueSkipTimeByFrames = ResultDurationSettingsValueSkipTimeByFrames;
class ResultDurationSettingsValueSkipTimeByDurations extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            duration: 'duration',
            minus: 'minus',
        };
    }
    static types() {
        return {
            duration: 'number',
            minus: 'number',
        };
    }
}
exports.ResultDurationSettingsValueSkipTimeByDurations = ResultDurationSettingsValueSkipTimeByDurations;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async addLeaveType(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddLeaveTypeHeaders({});
        return await this.addLeaveTypeWithOptions(request, headers, runtime);
    }
    async addLeaveTypeWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            query["opUserId"] = request.opUserId;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.bizType)) {
            body["bizType"] = request.bizType;
        }
        if (!tea_util_1.default.isUnset(request.extras)) {
            body["extras"] = request.extras;
        }
        if (!tea_util_1.default.isUnset(request.hoursInPerDay)) {
            body["hoursInPerDay"] = request.hoursInPerDay;
        }
        if (!tea_util_1.default.isUnset($tea.toMap(request.leaveCertificate))) {
            body["leaveCertificate"] = request.leaveCertificate;
        }
        if (!tea_util_1.default.isUnset(request.leaveName)) {
            body["leaveName"] = request.leaveName;
        }
        if (!tea_util_1.default.isUnset(request.leaveViewUnit)) {
            body["leaveViewUnit"] = request.leaveViewUnit;
        }
        if (!tea_util_1.default.isUnset(request.naturalDayLeave)) {
            body["naturalDayLeave"] = request.naturalDayLeave;
        }
        if (!tea_util_1.default.isUnset($tea.toMap(request.submitTimeRule))) {
            body["submitTimeRule"] = request.submitTimeRule;
        }
        if (!tea_util_1.default.isUnset(request.visibilityRules)) {
            body["visibilityRules"] = request.visibilityRules;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("AddLeaveType", "attendance_1.0", "HTTP", "POST", "AK", `/v1.0/attendance/leaves/types`, "json", req, runtime), new AddLeaveTypeResponse({}));
    }
    async attendanceBleDevicesAdd(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AttendanceBleDevicesAddHeaders({});
        return await this.attendanceBleDevicesAddWithOptions(request, headers, runtime);
    }
    async attendanceBleDevicesAddWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.deviceIdList)) {
            body["deviceIdList"] = request.deviceIdList;
        }
        if (!tea_util_1.default.isUnset(request.groupKey)) {
            body["groupKey"] = request.groupKey;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            body["opUserId"] = request.opUserId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("AttendanceBleDevicesAdd", "attendance_1.0", "HTTP", "POST", "AK", `/v1.0/attendance/group/bledevices`, "json", req, runtime), new AttendanceBleDevicesAddResponse({}));
    }
    async attendanceBleDevicesQuery(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AttendanceBleDevicesQueryHeaders({});
        return await this.attendanceBleDevicesQueryWithOptions(request, headers, runtime);
    }
    async attendanceBleDevicesQueryWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.groupKey)) {
            body["groupKey"] = request.groupKey;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            body["opUserId"] = request.opUserId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequestWithForm("AttendanceBleDevicesQuery", "attendance_1.0", "HTTP", "POST", "AK", `/v1.0/attendance/group/bledevices/query`, "json", req, runtime), new AttendanceBleDevicesQueryResponse({}));
    }
    async attendanceBleDevicesRemove(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AttendanceBleDevicesRemoveHeaders({});
        return await this.attendanceBleDevicesRemoveWithOptions(request, headers, runtime);
    }
    async attendanceBleDevicesRemoveWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.deviceIdList)) {
            body["deviceIdList"] = request.deviceIdList;
        }
        if (!tea_util_1.default.isUnset(request.groupKey)) {
            body["groupKey"] = request.groupKey;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            body["opUserId"] = request.opUserId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("AttendanceBleDevicesRemove", "attendance_1.0", "HTTP", "POST", "AK", `/v1.0/attendance/group/bledevices/remove`, "json", req, runtime), new AttendanceBleDevicesRemoveResponse({}));
    }
    async checkClosingAccount(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CheckClosingAccountHeaders({});
        return await this.checkClosingAccountWithOptions(request, headers, runtime);
    }
    async checkClosingAccountWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            body["bizCode"] = request.bizCode;
        }
        if (!tea_util_1.default.isUnset(request.userIds)) {
            body["userIds"] = request.userIds;
        }
        if (!tea_util_1.default.isUnset(request.userTimeRange)) {
            body["userTimeRange"] = request.userTimeRange;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("CheckClosingAccount", "attendance_1.0", "HTTP", "POST", "AK", `/v1.0/attendance/closingAccounts/status/query`, "json", req, runtime), new CheckClosingAccountResponse({}));
    }
    async checkWritePermission(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CheckWritePermissionHeaders({});
        return await this.checkWritePermissionWithOptions(request, headers, runtime);
    }
    async checkWritePermissionWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.category)) {
            body["category"] = request.category;
        }
        if (!tea_util_1.default.isUnset(request.entityIds)) {
            body["entityIds"] = request.entityIds;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            body["opUserId"] = request.opUserId;
        }
        if (!tea_util_1.default.isUnset(request.resourceKey)) {
            body["resourceKey"] = request.resourceKey;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("CheckWritePermission", "attendance_1.0", "HTTP", "POST", "AK", `/v1.0/attendance/writePermissions/query`, "json", req, runtime), new CheckWritePermissionResponse({}));
    }
    async createApprove(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateApproveHeaders({});
        return await this.createApproveWithOptions(request, headers, runtime);
    }
    async createApproveWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.approveId)) {
            body["approveId"] = request.approveId;
        }
        if (!tea_util_1.default.isUnset(request.opUserid)) {
            body["opUserid"] = request.opUserid;
        }
        if (!tea_util_1.default.isUnset($tea.toMap(request.punchParam))) {
            body["punchParam"] = request.punchParam;
        }
        if (!tea_util_1.default.isUnset(request.subType)) {
            body["subType"] = request.subType;
        }
        if (!tea_util_1.default.isUnset(request.tagName)) {
            body["tagName"] = request.tagName;
        }
        if (!tea_util_1.default.isUnset(request.userid)) {
            body["userid"] = request.userid;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("CreateApprove", "attendance_1.0", "HTTP", "POST", "AK", `/v1.0/attendance/approves`, "json", req, runtime), new CreateApproveResponse({}));
    }
    async deleteWaterMarkTemplate(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeleteWaterMarkTemplateHeaders({});
        return await this.deleteWaterMarkTemplateWithOptions(request, headers, runtime);
    }
    async deleteWaterMarkTemplateWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.formCode)) {
            query["formCode"] = request.formCode;
        }
        if (!tea_util_1.default.isUnset(request.formContent)) {
            query["formContent"] = request.formContent;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            query["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.systemTemplate)) {
            query["systemTemplate"] = request.systemTemplate;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("DeleteWaterMarkTemplate", "attendance_1.0", "HTTP", "DELETE", "AK", `/v1.0/attendance/watermarks/templates`, "json", req, runtime), new DeleteWaterMarkTemplateResponse({}));
    }
    async dingTalkSecurityCheck(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DingTalkSecurityCheckHeaders({});
        return await this.dingTalkSecurityCheckWithOptions(request, headers, runtime);
    }
    async dingTalkSecurityCheckWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.clientVer)) {
            body["clientVer"] = request.clientVer;
        }
        if (!tea_util_1.default.isUnset(request.platform)) {
            body["platform"] = request.platform;
        }
        if (!tea_util_1.default.isUnset(request.platformVer)) {
            body["platformVer"] = request.platformVer;
        }
        if (!tea_util_1.default.isUnset(request.sec)) {
            body["sec"] = request.sec;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("DingTalkSecurityCheck", "attendance_1.0", "HTTP", "POST", "AK", `/v1.0/attendance/securities/check`, "json", req, runtime), new DingTalkSecurityCheckResponse({}));
    }
    async getAdjustments(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetAdjustmentsHeaders({});
        return await this.getAdjustmentsWithOptions(request, headers, runtime);
    }
    async getAdjustmentsWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.pageNumber)) {
            query["pageNumber"] = request.pageNumber;
        }
        if (!tea_util_1.default.isUnset(request.pageSize)) {
            query["pageSize"] = request.pageSize;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("GetAdjustments", "attendance_1.0", "HTTP", "GET", "AK", `/v1.0/attendance/adjustments`, "json", req, runtime), new GetAdjustmentsResponse({}));
    }
    async getCheckInSchemaTemplate(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetCheckInSchemaTemplateHeaders({});
        return await this.getCheckInSchemaTemplateWithOptions(request, headers, runtime);
    }
    async getCheckInSchemaTemplateWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            query["bizCode"] = request.bizCode;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            query["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.sceneCode)) {
            query["sceneCode"] = request.sceneCode;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("GetCheckInSchemaTemplate", "attendance_1.0", "HTTP", "GET", "AK", `/v1.0/attendance/watermarks/templates`, "json", req, runtime), new GetCheckInSchemaTemplateResponse({}));
    }
    async getClosingAccounts(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetClosingAccountsHeaders({});
        return await this.getClosingAccountsWithOptions(request, headers, runtime);
    }
    async getClosingAccountsWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.userIds)) {
            body["userIds"] = request.userIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("GetClosingAccounts", "attendance_1.0", "HTTP", "POST", "AK", `/v1.0/attendance/closingAccounts/rules/query`, "json", req, runtime), new GetClosingAccountsResponse({}));
    }
    async getLeaveType(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetLeaveTypeHeaders({});
        return await this.getLeaveTypeWithOptions(request, headers, runtime);
    }
    async getLeaveTypeWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            query["opUserId"] = request.opUserId;
        }
        if (!tea_util_1.default.isUnset(request.vacationSource)) {
            query["vacationSource"] = request.vacationSource;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("GetLeaveType", "attendance_1.0", "HTTP", "GET", "AK", `/v1.0/attendance/leaves/types`, "json", req, runtime), new GetLeaveTypeResponse({}));
    }
    async getMachine(devId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetMachineHeaders({});
        return await this.getMachineWithOptions(devId, headers, runtime);
    }
    async getMachineWithOptions(devId, headers, runtime) {
        devId = openapi_util_1.default.getEncodeParam(devId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("GetMachine", "attendance_1.0", "HTTP", "GET", "AK", `/v1.0/attendance/machines/${devId}`, "json", req, runtime), new GetMachineResponse({}));
    }
    async getMachineUser(devId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetMachineUserHeaders({});
        return await this.getMachineUserWithOptions(devId, request, headers, runtime);
    }
    async getMachineUserWithOptions(devId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        devId = openapi_util_1.default.getEncodeParam(devId);
        let query = {};
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("GetMachineUser", "attendance_1.0", "HTTP", "GET", "AK", `/v1.0/attendance/machines/getUser/${devId}`, "json", req, runtime), new GetMachineUserResponse({}));
    }
    async getOvertimeSetting(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetOvertimeSettingHeaders({});
        return await this.getOvertimeSettingWithOptions(request, headers, runtime);
    }
    async getOvertimeSettingWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.overtimeSettingIds)) {
            body["overtimeSettingIds"] = request.overtimeSettingIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("GetOvertimeSetting", "attendance_1.0", "HTTP", "POST", "AK", `/v1.0/attendance/overtimeSettings/query`, "json", req, runtime), new GetOvertimeSettingResponse({}));
    }
    async getSimpleOvertimeSetting(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetSimpleOvertimeSettingHeaders({});
        return await this.getSimpleOvertimeSettingWithOptions(request, headers, runtime);
    }
    async getSimpleOvertimeSettingWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.pageNumber)) {
            query["pageNumber"] = request.pageNumber;
        }
        if (!tea_util_1.default.isUnset(request.pageSize)) {
            query["pageSize"] = request.pageSize;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("GetSimpleOvertimeSetting", "attendance_1.0", "HTTP", "GET", "AK", `/v1.0/attendance/overtimeSettings`, "json", req, runtime), new GetSimpleOvertimeSettingResponse({}));
    }
    async getUserHolidays(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetUserHolidaysHeaders({});
        return await this.getUserHolidaysWithOptions(request, headers, runtime);
    }
    async getUserHolidaysWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.userIds)) {
            body["userIds"] = request.userIds;
        }
        if (!tea_util_1.default.isUnset(request.workDateFrom)) {
            body["workDateFrom"] = request.workDateFrom;
        }
        if (!tea_util_1.default.isUnset(request.workDateTo)) {
            body["workDateTo"] = request.workDateTo;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("GetUserHolidays", "attendance_1.0", "HTTP", "POST", "AK", `/v1.0/attendance/holidays`, "json", req, runtime), new GetUserHolidaysResponse({}));
    }
    async initAndGetLeaveALlocationQuotas(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new InitAndGetLeaveALlocationQuotasHeaders({});
        return await this.initAndGetLeaveALlocationQuotasWithOptions(request, headers, runtime);
    }
    async initAndGetLeaveALlocationQuotasWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.leaveCode)) {
            query["leaveCode"] = request.leaveCode;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            query["opUserId"] = request.opUserId;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("InitAndGetLeaveALlocationQuotas", "attendance_1.0", "HTTP", "GET", "AK", `/v1.0/attendance/leaves/initializations/balances`, "json", req, runtime), new InitAndGetLeaveALlocationQuotasResponse({}));
    }
    async modifyWaterMarkTemplate(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ModifyWaterMarkTemplateHeaders({});
        return await this.modifyWaterMarkTemplateWithOptions(request, headers, runtime);
    }
    async modifyWaterMarkTemplateWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            query["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.formCode)) {
            body["formCode"] = request.formCode;
        }
        if (!tea_util_1.default.isUnset(request.icon)) {
            body["icon"] = request.icon;
        }
        if (!tea_util_1.default.isUnset(request.layoutDesignId)) {
            body["layoutDesignId"] = request.layoutDesignId;
        }
        if (!tea_util_1.default.isUnset(request.schemaContent)) {
            body["schemaContent"] = request.schemaContent;
        }
        if (!tea_util_1.default.isUnset(request.title)) {
            body["title"] = request.title;
        }
        if (!tea_util_1.default.isUnset(request.waterMarkId)) {
            body["waterMarkId"] = request.waterMarkId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("ModifyWaterMarkTemplate", "attendance_1.0", "HTTP", "PUT", "AK", `/v1.0/attendance/watermarks/templates`, "json", req, runtime), new ModifyWaterMarkTemplateResponse({}));
    }
    async processApproveCreate(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ProcessApproveCreateHeaders({});
        return await this.processApproveCreateWithOptions(request, headers, runtime);
    }
    async processApproveCreateWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.approveId)) {
            body["approveId"] = request.approveId;
        }
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            body["opUserId"] = request.opUserId;
        }
        if (!tea_util_1.default.isUnset($tea.toMap(request.punchParam))) {
            body["punchParam"] = request.punchParam;
        }
        if (!tea_util_1.default.isUnset(request.subType)) {
            body["subType"] = request.subType;
        }
        if (!tea_util_1.default.isUnset(request.tagName)) {
            body["tagName"] = request.tagName;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("ProcessApproveCreate", "attendance_1.0", "HTTP", "POST", "AK", `/v1.0/attendance/workflows/checkInForms`, "json", req, runtime), new ProcessApproveCreateResponse({}));
    }
    async saveCustomWaterMarkTemplate(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SaveCustomWaterMarkTemplateHeaders({});
        return await this.saveCustomWaterMarkTemplateWithOptions(request, headers, runtime);
    }
    async saveCustomWaterMarkTemplateWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            query["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.bizCode)) {
            body["bizCode"] = request.bizCode;
        }
        if (!tea_util_1.default.isUnset(request.icon)) {
            body["icon"] = request.icon;
        }
        if (!tea_util_1.default.isUnset(request.layoutDesignId)) {
            body["layoutDesignId"] = request.layoutDesignId;
        }
        if (!tea_util_1.default.isUnset(request.sceneCode)) {
            body["sceneCode"] = request.sceneCode;
        }
        if (!tea_util_1.default.isUnset(request.schemaContent)) {
            body["schemaContent"] = request.schemaContent;
        }
        if (!tea_util_1.default.isUnset(request.title)) {
            body["title"] = request.title;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("SaveCustomWaterMarkTemplate", "attendance_1.0", "HTTP", "POST", "AK", `/v1.0/attendance/watermarks/templates`, "json", req, runtime), new SaveCustomWaterMarkTemplateResponse({}));
    }
    async syncScheduleInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SyncScheduleInfoHeaders({});
        return await this.syncScheduleInfoWithOptions(request, headers, runtime);
    }
    async syncScheduleInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            body["opUserId"] = request.opUserId;
        }
        if (!tea_util_1.default.isUnset(request.scheduleInfos)) {
            body["scheduleInfos"] = request.scheduleInfos;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("SyncScheduleInfo", "attendance_1.0", "HTTP", "PUT", "AK", `/v1.0/attendance/schedules/additionalInfo`, "none", req, runtime), new SyncScheduleInfoResponse({}));
    }
    async updateLeaveType(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateLeaveTypeHeaders({});
        return await this.updateLeaveTypeWithOptions(request, headers, runtime);
    }
    async updateLeaveTypeWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.opUserId)) {
            query["opUserId"] = request.opUserId;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.bizType)) {
            body["bizType"] = request.bizType;
        }
        if (!tea_util_1.default.isUnset(request.extras)) {
            body["extras"] = request.extras;
        }
        if (!tea_util_1.default.isUnset(request.hoursInPerDay)) {
            body["hoursInPerDay"] = request.hoursInPerDay;
        }
        if (!tea_util_1.default.isUnset($tea.toMap(request.leaveCertificate))) {
            body["leaveCertificate"] = request.leaveCertificate;
        }
        if (!tea_util_1.default.isUnset(request.leaveCode)) {
            body["leaveCode"] = request.leaveCode;
        }
        if (!tea_util_1.default.isUnset(request.leaveName)) {
            body["leaveName"] = request.leaveName;
        }
        if (!tea_util_1.default.isUnset(request.leaveViewUnit)) {
            body["leaveViewUnit"] = request.leaveViewUnit;
        }
        if (!tea_util_1.default.isUnset(request.naturalDayLeave)) {
            body["naturalDayLeave"] = request.naturalDayLeave;
        }
        if (!tea_util_1.default.isUnset($tea.toMap(request.submitTimeRule))) {
            body["submitTimeRule"] = request.submitTimeRule;
        }
        if (!tea_util_1.default.isUnset(request.visibilityRules)) {
            body["visibilityRules"] = request.visibilityRules;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateLeaveType", "attendance_1.0", "HTTP", "PUT", "AK", `/v1.0/attendance/leaves/types`, "json", req, runtime), new UpdateLeaveTypeResponse({}));
    }
}
exports.default = Client;
