// @alicloud/dingtalk/dist/project_integration_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class AddAttendeeToEventGroupHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.AddAttendeeToEventGroupHeaders = AddAttendeeToEventGroupHeaders;
class AddAttendeeToEventGroupResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
        };
    }
}
exports.AddAttendeeToEventGroupResponse = AddAttendeeToEventGroupResponse;
class CreateEventGroupHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CreateEventGroupHeaders = CreateEventGroupHeaders;
class CreateEventGroupResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
        };
    }
}
exports.CreateEventGroupResponse = CreateEventGroupResponse;
class SendInteractiveCardHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SendInteractiveCardHeaders = SendInteractiveCardHeaders;
class SendInteractiveCardResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
        };
    }
}
exports.SendInteractiveCardResponse = SendInteractiveCardResponse;
class SendSingleInteractiveCardHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SendSingleInteractiveCardHeaders = SendSingleInteractiveCardHeaders;
class SendSingleInteractiveCardResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
        };
    }
}
exports.SendSingleInteractiveCardResponse = SendSingleInteractiveCardResponse;
class UpdateInteractiveCardHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateInteractiveCardHeaders = UpdateInteractiveCardHeaders;
class UpdateInteractiveCardResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
        };
    }
}
exports.UpdateInteractiveCardResponse = UpdateInteractiveCardResponse;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async addAttendeeToEventGroup(userId, groupId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddAttendeeToEventGroupHeaders({});
        return await this.addAttendeeToEventGroupWithOptions(userId, groupId, headers, runtime);
    }
    async addAttendeeToEventGroupWithOptions(userId, groupId, headers, runtime) {
        userId = openapi_util_1.default.getEncodeParam(userId);
        groupId = openapi_util_1.default.getEncodeParam(groupId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("AddAttendeeToEventGroup", "projectIntegration_1.0", "HTTP", "POST", "AK", `/v1.0/projectIntegration/users/${userId}/eventGroups/${groupId}/members`, "json", req, runtime), new AddAttendeeToEventGroupResponse({}));
    }
    async createEventGroup(userId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateEventGroupHeaders({});
        return await this.createEventGroupWithOptions(userId, headers, runtime);
    }
    async createEventGroupWithOptions(userId, headers, runtime) {
        userId = openapi_util_1.default.getEncodeParam(userId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("CreateEventGroup", "projectIntegration_1.0", "HTTP", "POST", "AK", `/v1.0/projectIntegration/users/${userId}/eventGroups`, "json", req, runtime), new CreateEventGroupResponse({}));
    }
    async sendInteractiveCard(userId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendInteractiveCardHeaders({});
        return await this.sendInteractiveCardWithOptions(userId, headers, runtime);
    }
    async sendInteractiveCardWithOptions(userId, headers, runtime) {
        userId = openapi_util_1.default.getEncodeParam(userId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("SendInteractiveCard", "projectIntegration_1.0", "HTTP", "POST", "AK", `/v1.0/projectIntegration/users/${userId}/groupChatCardMessages`, "json", req, runtime), new SendInteractiveCardResponse({}));
    }
    async sendSingleInteractiveCard(userId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendSingleInteractiveCardHeaders({});
        return await this.sendSingleInteractiveCardWithOptions(userId, headers, runtime);
    }
    async sendSingleInteractiveCardWithOptions(userId, headers, runtime) {
        userId = openapi_util_1.default.getEncodeParam(userId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("SendSingleInteractiveCard", "projectIntegration_1.0", "HTTP", "POST", "AK", `/v1.0/projectIntegration/users/${userId}/singleChatCardMessages`, "json", req, runtime), new SendSingleInteractiveCardResponse({}));
    }
    async updateInteractiveCard(userId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateInteractiveCardHeaders({});
        return await this.updateInteractiveCardWithOptions(userId, headers, runtime);
    }
    async updateInteractiveCardWithOptions(userId, headers, runtime) {
        userId = openapi_util_1.default.getEncodeParam(userId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("UpdateInteractiveCard", "projectIntegration_1.0", "HTTP", "PUT", "AK", `/v1.0/projectIntegration/users/${userId}/cardMessages`, "json", req, runtime), new UpdateInteractiveCardResponse({}));
    }
}
exports.default = Client;
