// @alicloud/dingtalk/dist/im_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class AutoOpenDingTalkConnectHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.AutoOpenDingTalkConnectHeaders = AutoOpenDingTalkConnectHeaders;
class AutoOpenDingTalkConnectResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            message: 'message',
        };
    }
    static types() {
        return {
            message: 'string',
        };
    }
}
exports.AutoOpenDingTalkConnectResponseBody = AutoOpenDingTalkConnectResponseBody;
class AutoOpenDingTalkConnectResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: AutoOpenDingTalkConnectResponseBody,
        };
    }
}
exports.AutoOpenDingTalkConnectResponse = AutoOpenDingTalkConnectResponse;
class BatchQueryGroupMemberHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.BatchQueryGroupMemberHeaders = BatchQueryGroupMemberHeaders;
class BatchQueryGroupMemberRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            coolAppCode: 'coolAppCode',
            maxResults: 'maxResults',
            nextToken: 'nextToken',
            openConversationId: 'openConversationId',
        };
    }
    static types() {
        return {
            coolAppCode: 'string',
            maxResults: 'number',
            nextToken: 'string',
            openConversationId: 'string',
        };
    }
}
exports.BatchQueryGroupMemberRequest = BatchQueryGroupMemberRequest;
class BatchQueryGroupMemberResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            hasMore: 'hasMore',
            memberUserIds: 'memberUserIds',
            nextToken: 'nextToken',
            success: 'success',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            memberUserIds: { 'type': 'array', 'itemType': 'string' },
            nextToken: 'string',
            success: 'boolean',
        };
    }
}
exports.BatchQueryGroupMemberResponseBody = BatchQueryGroupMemberResponseBody;
class BatchQueryGroupMemberResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: BatchQueryGroupMemberResponseBody,
        };
    }
}
exports.BatchQueryGroupMemberResponse = BatchQueryGroupMemberResponse;
class CardTemplateBuildActionHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CardTemplateBuildActionHeaders = CardTemplateBuildActionHeaders;
class CardTemplateBuildActionRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            action: 'action',
            cardTemplateJson: 'cardTemplateJson',
        };
    }
    static types() {
        return {
            action: 'string',
            cardTemplateJson: 'string',
        };
    }
}
exports.CardTemplateBuildActionRequest = CardTemplateBuildActionRequest;
class CardTemplateBuildActionResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            cardTemplateJson: 'cardTemplateJson',
        };
    }
    static types() {
        return {
            cardTemplateJson: 'string',
        };
    }
}
exports.CardTemplateBuildActionResponseBody = CardTemplateBuildActionResponseBody;
class CardTemplateBuildActionResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: CardTemplateBuildActionResponseBody,
        };
    }
}
exports.CardTemplateBuildActionResponse = CardTemplateBuildActionResponse;
class ChatIdToOpenConversationIdHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ChatIdToOpenConversationIdHeaders = ChatIdToOpenConversationIdHeaders;
class ChatIdToOpenConversationIdResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            openConversationId: 'openConversationId',
        };
    }
    static types() {
        return {
            openConversationId: 'string',
        };
    }
}
exports.ChatIdToOpenConversationIdResponseBody = ChatIdToOpenConversationIdResponseBody;
class ChatIdToOpenConversationIdResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: ChatIdToOpenConversationIdResponseBody,
        };
    }
}
exports.ChatIdToOpenConversationIdResponse = ChatIdToOpenConversationIdResponse;
class ChatSubAdminUpdateHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ChatSubAdminUpdateHeaders = ChatSubAdminUpdateHeaders;
class ChatSubAdminUpdateRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            openConversationId: 'openConversationId',
            role: 'role',
            userIds: 'userIds',
        };
    }
    static types() {
        return {
            openConversationId: 'string',
            role: 'number',
            userIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.ChatSubAdminUpdateRequest = ChatSubAdminUpdateRequest;
class ChatSubAdminUpdateResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'string',
        };
    }
}
exports.ChatSubAdminUpdateResponseBody = ChatSubAdminUpdateResponseBody;
class ChatSubAdminUpdateResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: ChatSubAdminUpdateResponseBody,
        };
    }
}
exports.ChatSubAdminUpdateResponse = ChatSubAdminUpdateResponse;
class CreateCoupleGroupConversationHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CreateCoupleGroupConversationHeaders = CreateCoupleGroupConversationHeaders;
class CreateCoupleGroupConversationRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appUserId: 'appUserId',
            groupAvatar: 'groupAvatar',
            groupName: 'groupName',
            groupOwnerId: 'groupOwnerId',
            groupTemplateId: 'groupTemplateId',
            operatorId: 'operatorId',
        };
    }
    static types() {
        return {
            appUserId: 'string',
            groupAvatar: 'string',
            groupName: 'string',
            groupOwnerId: 'string',
            groupTemplateId: 'string',
            operatorId: 'string',
        };
    }
}
exports.CreateCoupleGroupConversationRequest = CreateCoupleGroupConversationRequest;
class CreateCoupleGroupConversationResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            chatId: 'chatId',
            openConversationId: 'openConversationId',
        };
    }
    static types() {
        return {
            chatId: 'string',
            openConversationId: 'string',
        };
    }
}
exports.CreateCoupleGroupConversationResponseBody = CreateCoupleGroupConversationResponseBody;
class CreateCoupleGroupConversationResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: CreateCoupleGroupConversationResponseBody,
        };
    }
}
exports.CreateCoupleGroupConversationResponse = CreateCoupleGroupConversationResponse;
class CreateGroupConversationHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CreateGroupConversationHeaders = CreateGroupConversationHeaders;
class CreateGroupConversationRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appUserIds: 'appUserIds',
            groupAvatar: 'groupAvatar',
            groupName: 'groupName',
            groupOwnerId: 'groupOwnerId',
            groupTemplateId: 'groupTemplateId',
            operatorId: 'operatorId',
            userIds: 'userIds',
        };
    }
    static types() {
        return {
            appUserIds: { 'type': 'array', 'itemType': 'string' },
            groupAvatar: 'string',
            groupName: 'string',
            groupOwnerId: 'string',
            groupTemplateId: 'string',
            operatorId: 'string',
            userIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.CreateGroupConversationRequest = CreateGroupConversationRequest;
class CreateGroupConversationResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appUserIds: 'appUserIds',
            chatId: 'chatId',
            openConversationId: 'openConversationId',
            userIds: 'userIds',
        };
    }
    static types() {
        return {
            appUserIds: { 'type': 'array', 'itemType': 'string' },
            chatId: 'string',
            openConversationId: 'string',
            userIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.CreateGroupConversationResponseBody = CreateGroupConversationResponseBody;
class CreateGroupConversationResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: CreateGroupConversationResponseBody,
        };
    }
}
exports.CreateGroupConversationResponse = CreateGroupConversationResponse;
class CreateInterconnectionHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CreateInterconnectionHeaders = CreateInterconnectionHeaders;
class CreateInterconnectionRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            interconnections: 'interconnections',
        };
    }
    static types() {
        return {
            interconnections: { 'type': 'array', 'itemType': CreateInterconnectionRequestInterconnections },
        };
    }
}
exports.CreateInterconnectionRequest = CreateInterconnectionRequest;
class CreateInterconnectionResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            results: 'results',
        };
    }
    static types() {
        return {
            results: { 'type': 'array', 'itemType': CreateInterconnectionResponseBodyResults },
        };
    }
}
exports.CreateInterconnectionResponseBody = CreateInterconnectionResponseBody;
class CreateInterconnectionResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: CreateInterconnectionResponseBody,
        };
    }
}
exports.CreateInterconnectionResponse = CreateInterconnectionResponse;
class CreateStoreGroupConversationHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CreateStoreGroupConversationHeaders = CreateStoreGroupConversationHeaders;
class CreateStoreGroupConversationRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appUserId: 'appUserId',
            businessUniqueKey: 'businessUniqueKey',
            groupAvatar: 'groupAvatar',
            groupName: 'groupName',
            groupTemplateId: 'groupTemplateId',
            operatorId: 'operatorId',
            userIds: 'userIds',
        };
    }
    static types() {
        return {
            appUserId: 'string',
            businessUniqueKey: 'string',
            groupAvatar: 'string',
            groupName: 'string',
            groupTemplateId: 'string',
            operatorId: 'string',
            userIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.CreateStoreGroupConversationRequest = CreateStoreGroupConversationRequest;
class CreateStoreGroupConversationResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            chatId: 'chatId',
            openConversationId: 'openConversationId',
        };
    }
    static types() {
        return {
            chatId: 'string',
            openConversationId: 'string',
        };
    }
}
exports.CreateStoreGroupConversationResponseBody = CreateStoreGroupConversationResponseBody;
class CreateStoreGroupConversationResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: CreateStoreGroupConversationResponseBody,
        };
    }
}
exports.CreateStoreGroupConversationResponse = CreateStoreGroupConversationResponse;
class GetConversationUrlHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetConversationUrlHeaders = GetConversationUrlHeaders;
class GetConversationUrlRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appUserId: 'appUserId',
            channelCode: 'channelCode',
            openConversationId: 'openConversationId',
            sourceCode: 'sourceCode',
            userId: 'userId',
        };
    }
    static types() {
        return {
            appUserId: 'string',
            channelCode: 'string',
            openConversationId: 'string',
            sourceCode: 'string',
            userId: 'string',
        };
    }
}
exports.GetConversationUrlRequest = GetConversationUrlRequest;
class GetConversationUrlResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            url: 'url',
        };
    }
    static types() {
        return {
            url: 'string',
        };
    }
}
exports.GetConversationUrlResponseBody = GetConversationUrlResponseBody;
class GetConversationUrlResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetConversationUrlResponseBody,
        };
    }
}
exports.GetConversationUrlResponse = GetConversationUrlResponse;
class GetInterconnectionUrlHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetInterconnectionUrlHeaders = GetInterconnectionUrlHeaders;
class GetInterconnectionUrlRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appUserAvatar: 'appUserAvatar',
            appUserAvatarType: 'appUserAvatarType',
            appUserId: 'appUserId',
            appUserMobileNumber: 'appUserMobileNumber',
            appUserName: 'appUserName',
            msgPageType: 'msgPageType',
            qrCode: 'qrCode',
            signature: 'signature',
            sourceCode: 'sourceCode',
            sourceType: 'sourceType',
            userId: 'userId',
        };
    }
    static types() {
        return {
            appUserAvatar: 'string',
            appUserAvatarType: 'number',
            appUserId: 'string',
            appUserMobileNumber: 'string',
            appUserName: 'string',
            msgPageType: 'number',
            qrCode: 'string',
            signature: 'string',
            sourceCode: 'string',
            sourceType: 'number',
            userId: 'string',
        };
    }
}
exports.GetInterconnectionUrlRequest = GetInterconnectionUrlRequest;
class GetInterconnectionUrlResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            url: 'url',
        };
    }
    static types() {
        return {
            url: 'string',
        };
    }
}
exports.GetInterconnectionUrlResponseBody = GetInterconnectionUrlResponseBody;
class GetInterconnectionUrlResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetInterconnectionUrlResponseBody,
        };
    }
}
exports.GetInterconnectionUrlResponse = GetInterconnectionUrlResponse;
class GetSceneGroupInfoHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetSceneGroupInfoHeaders = GetSceneGroupInfoHeaders;
class GetSceneGroupInfoRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            coolAppCode: 'coolAppCode',
            openConversationId: 'openConversationId',
        };
    }
    static types() {
        return {
            coolAppCode: 'string',
            openConversationId: 'string',
        };
    }
}
exports.GetSceneGroupInfoRequest = GetSceneGroupInfoRequest;
class GetSceneGroupInfoResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupUrl: 'groupUrl',
            icon: 'icon',
            openConversationId: 'openConversationId',
            ownerUserId: 'ownerUserId',
            success: 'success',
            templateId: 'templateId',
            title: 'title',
        };
    }
    static types() {
        return {
            groupUrl: 'string',
            icon: 'string',
            openConversationId: 'string',
            ownerUserId: 'string',
            success: 'boolean',
            templateId: 'string',
            title: 'string',
        };
    }
}
exports.GetSceneGroupInfoResponseBody = GetSceneGroupInfoResponseBody;
class GetSceneGroupInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetSceneGroupInfoResponseBody,
        };
    }
}
exports.GetSceneGroupInfoResponse = GetSceneGroupInfoResponse;
class GetSceneGroupMembersHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetSceneGroupMembersHeaders = GetSceneGroupMembersHeaders;
class GetSceneGroupMembersRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            coolAppCode: 'coolAppCode',
            cursor: 'cursor',
            openConversationId: 'openConversationId',
            size: 'size',
        };
    }
    static types() {
        return {
            coolAppCode: 'string',
            cursor: 'string',
            openConversationId: 'string',
            size: 'number',
        };
    }
}
exports.GetSceneGroupMembersRequest = GetSceneGroupMembersRequest;
class GetSceneGroupMembersResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            hasMore: 'hasMore',
            memberUserIds: 'memberUserIds',
            nextCursor: 'nextCursor',
            success: 'success',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            memberUserIds: { 'type': 'array', 'itemType': 'string' },
            nextCursor: 'string',
            success: 'boolean',
        };
    }
}
exports.GetSceneGroupMembersResponseBody = GetSceneGroupMembersResponseBody;
class GetSceneGroupMembersResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetSceneGroupMembersResponseBody,
        };
    }
}
exports.GetSceneGroupMembersResponse = GetSceneGroupMembersResponse;
class GroupBanWordsHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GroupBanWordsHeaders = GroupBanWordsHeaders;
class GroupBanWordsRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            banWordsMode: 'banWordsMode',
            openConversationId: 'openConversationId',
            options: 'options',
        };
    }
    static types() {
        return {
            banWordsMode: 'number',
            openConversationId: 'string',
            options: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
        };
    }
}
exports.GroupBanWordsRequest = GroupBanWordsRequest;
class GroupBanWordsResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.GroupBanWordsResponse = GroupBanWordsResponse;
class GroupCapacityInquiryHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GroupCapacityInquiryHeaders = GroupCapacityInquiryHeaders;
class GroupCapacityInquiryRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            effectiveDuration: 'effectiveDuration',
            openConversationId: 'openConversationId',
            operator: 'operator',
            options: 'options',
            targetCapacity: 'targetCapacity',
        };
    }
    static types() {
        return {
            effectiveDuration: 'string',
            openConversationId: 'string',
            operator: 'string',
            options: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            targetCapacity: 'number',
        };
    }
}
exports.GroupCapacityInquiryRequest = GroupCapacityInquiryRequest;
class GroupCapacityInquiryResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            actualPrice: 'actualPrice',
            createdAt: 'createdAt',
            currentCapacity: 'currentCapacity',
            currentEffectUntil: 'currentEffectUntil',
            discount: 'discount',
            extInfo: 'extInfo',
            groupOwner: 'groupOwner',
            groupTitle: 'groupTitle',
            markedPrice: 'markedPrice',
            memberCount: 'memberCount',
            openConversationId: 'openConversationId',
            operator: 'operator',
            targetCapacity: 'targetCapacity',
            targetEffectUntil: 'targetEffectUntil',
            token: 'token',
        };
    }
    static types() {
        return {
            actualPrice: 'number',
            createdAt: 'number',
            currentCapacity: 'number',
            currentEffectUntil: 'number',
            discount: 'number',
            extInfo: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            groupOwner: 'string',
            groupTitle: 'string',
            markedPrice: 'number',
            memberCount: 'number',
            openConversationId: 'string',
            operator: 'string',
            targetCapacity: 'number',
            targetEffectUntil: 'number',
            token: 'string',
        };
    }
}
exports.GroupCapacityInquiryResponseBody = GroupCapacityInquiryResponseBody;
class GroupCapacityInquiryResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GroupCapacityInquiryResponseBody,
        };
    }
}
exports.GroupCapacityInquiryResponse = GroupCapacityInquiryResponse;
class GroupCapacityOrderConfirmHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GroupCapacityOrderConfirmHeaders = GroupCapacityOrderConfirmHeaders;
class GroupCapacityOrderConfirmRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            operator: 'operator',
            orderId: 'orderId',
        };
    }
    static types() {
        return {
            operator: 'string',
            orderId: 'string',
        };
    }
}
exports.GroupCapacityOrderConfirmRequest = GroupCapacityOrderConfirmRequest;
class GroupCapacityOrderConfirmResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.GroupCapacityOrderConfirmResponse = GroupCapacityOrderConfirmResponse;
class GroupCapacityOrderPlaceHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GroupCapacityOrderPlaceHeaders = GroupCapacityOrderPlaceHeaders;
class GroupCapacityOrderPlaceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            actualPrice: 'actualPrice',
            currentCapacity: 'currentCapacity',
            currentEffectUntil: 'currentEffectUntil',
            discount: 'discount',
            extInfo: 'extInfo',
            markedPrice: 'markedPrice',
            openConversationId: 'openConversationId',
            operator: 'operator',
            targetCapacity: 'targetCapacity',
            targetEffectUntil: 'targetEffectUntil',
            token: 'token',
        };
    }
    static types() {
        return {
            actualPrice: 'number',
            currentCapacity: 'number',
            currentEffectUntil: 'number',
            discount: 'number',
            extInfo: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            markedPrice: 'number',
            openConversationId: 'string',
            operator: 'string',
            targetCapacity: 'number',
            targetEffectUntil: 'number',
            token: 'string',
        };
    }
}
exports.GroupCapacityOrderPlaceRequest = GroupCapacityOrderPlaceRequest;
class GroupCapacityOrderPlaceResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            actualPrice: 'actualPrice',
            currentCapacity: 'currentCapacity',
            currentEffectUntil: 'currentEffectUntil',
            discount: 'discount',
            extInfo: 'extInfo',
            markedPrice: 'markedPrice',
            openConversationId: 'openConversationId',
            operator: 'operator',
            orderId: 'orderId',
            targetCapacity: 'targetCapacity',
            targetEffectUntil: 'targetEffectUntil',
            token: 'token',
        };
    }
    static types() {
        return {
            actualPrice: 'number',
            currentCapacity: 'number',
            currentEffectUntil: 'number',
            discount: 'number',
            extInfo: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            markedPrice: 'number',
            openConversationId: 'string',
            operator: 'string',
            orderId: 'string',
            targetCapacity: 'number',
            targetEffectUntil: 'number',
            token: 'string',
        };
    }
}
exports.GroupCapacityOrderPlaceResponseBody = GroupCapacityOrderPlaceResponseBody;
class GroupCapacityOrderPlaceResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GroupCapacityOrderPlaceResponseBody,
        };
    }
}
exports.GroupCapacityOrderPlaceResponse = GroupCapacityOrderPlaceResponse;
class GroupManageQueryHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GroupManageQueryHeaders = GroupManageQueryHeaders;
class GroupManageQueryRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            createdAfter: 'createdAfter',
            groupId: 'groupId',
            groupMemberSamples: 'groupMemberSamples',
            groupOwner: 'groupOwner',
            groupTitleKeywords: 'groupTitleKeywords',
            groupUrl: 'groupUrl',
            maxResults: 'maxResults',
            membersOver: 'membersOver',
            nextToken: 'nextToken',
            openConversationId: 'openConversationId',
        };
    }
    static types() {
        return {
            createdAfter: 'number',
            groupId: 'string',
            groupMemberSamples: { 'type': 'array', 'itemType': 'string' },
            groupOwner: 'string',
            groupTitleKeywords: { 'type': 'array', 'itemType': 'string' },
            groupUrl: 'string',
            maxResults: 'number',
            membersOver: 'number',
            nextToken: 'string',
            openConversationId: 'string',
        };
    }
}
exports.GroupManageQueryRequest = GroupManageQueryRequest;
class GroupManageQueryResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupInfoList: 'groupInfoList',
            hasMore: 'hasMore',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            groupInfoList: { 'type': 'array', 'itemType': GroupManageQueryResponseBodyGroupInfoList },
            hasMore: 'boolean',
            nextToken: 'string',
        };
    }
}
exports.GroupManageQueryResponseBody = GroupManageQueryResponseBody;
class GroupManageQueryResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GroupManageQueryResponseBody,
        };
    }
}
exports.GroupManageQueryResponse = GroupManageQueryResponse;
class GroupManageReduceHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GroupManageReduceHeaders = GroupManageReduceHeaders;
class GroupManageReduceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            capacityLimit: 'capacityLimit',
            openConversationId: 'openConversationId',
            options: 'options',
        };
    }
    static types() {
        return {
            capacityLimit: 'number',
            openConversationId: 'string',
            options: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
        };
    }
}
exports.GroupManageReduceRequest = GroupManageReduceRequest;
class GroupManageReduceResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.GroupManageReduceResponse = GroupManageReduceResponse;
class InteractiveCardCreateInstanceHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.InteractiveCardCreateInstanceHeaders = InteractiveCardCreateInstanceHeaders;
class InteractiveCardCreateInstanceRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            callbackRouteKey: 'callbackRouteKey',
            cardData: 'cardData',
            cardTemplateId: 'cardTemplateId',
            chatBotId: 'chatBotId',
            conversationType: 'conversationType',
            openConversationId: 'openConversationId',
            outTrackId: 'outTrackId',
            privateData: 'privateData',
            pullStrategy: 'pullStrategy',
            receiverUserIdList: 'receiverUserIdList',
            robotCode: 'robotCode',
            userIdType: 'userIdType',
        };
    }
    static types() {
        return {
            callbackRouteKey: 'string',
            cardData: InteractiveCardCreateInstanceRequestCardData,
            cardTemplateId: 'string',
            chatBotId: 'string',
            conversationType: 'number',
            openConversationId: 'string',
            outTrackId: 'string',
            privateData: { 'type': 'map', 'keyType': 'string', 'valueType': PrivateDataValue },
            pullStrategy: 'boolean',
            receiverUserIdList: { 'type': 'array', 'itemType': 'string' },
            robotCode: 'string',
            userIdType: 'number',
        };
    }
}
exports.InteractiveCardCreateInstanceRequest = InteractiveCardCreateInstanceRequest;
class InteractiveCardCreateInstanceResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            processQueryKey: 'processQueryKey',
        };
    }
    static types() {
        return {
            processQueryKey: 'string',
        };
    }
}
exports.InteractiveCardCreateInstanceResponseBody = InteractiveCardCreateInstanceResponseBody;
class InteractiveCardCreateInstanceResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: InteractiveCardCreateInstanceResponseBody,
        };
    }
}
exports.InteractiveCardCreateInstanceResponse = InteractiveCardCreateInstanceResponse;
class QueryGroupInfoByMemberAuthHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryGroupInfoByMemberAuthHeaders = QueryGroupInfoByMemberAuthHeaders;
class QueryGroupInfoByMemberAuthRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            coolAppCode: 'coolAppCode',
            openConversationId: 'openConversationId',
        };
    }
    static types() {
        return {
            coolAppCode: 'string',
            openConversationId: 'string',
        };
    }
}
exports.QueryGroupInfoByMemberAuthRequest = QueryGroupInfoByMemberAuthRequest;
class QueryGroupInfoByMemberAuthResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            memberCount: 'memberCount',
        };
    }
    static types() {
        return {
            memberCount: 'number',
        };
    }
}
exports.QueryGroupInfoByMemberAuthResponseBody = QueryGroupInfoByMemberAuthResponseBody;
class QueryGroupInfoByMemberAuthResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: QueryGroupInfoByMemberAuthResponseBody,
        };
    }
}
exports.QueryGroupInfoByMemberAuthResponse = QueryGroupInfoByMemberAuthResponse;
class QueryGroupMemberHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryGroupMemberHeaders = QueryGroupMemberHeaders;
class QueryGroupMemberRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            openConversationId: 'openConversationId',
        };
    }
    static types() {
        return {
            openConversationId: 'string',
        };
    }
}
exports.QueryGroupMemberRequest = QueryGroupMemberRequest;
class QueryGroupMemberResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupMembers: 'groupMembers',
            openConversationId: 'openConversationId',
        };
    }
    static types() {
        return {
            groupMembers: { 'type': 'array', 'itemType': QueryGroupMemberResponseBodyGroupMembers },
            openConversationId: 'string',
        };
    }
}
exports.QueryGroupMemberResponseBody = QueryGroupMemberResponseBody;
class QueryGroupMemberResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: QueryGroupMemberResponseBody,
        };
    }
}
exports.QueryGroupMemberResponse = QueryGroupMemberResponse;
class QueryGroupMemberByMemberAuthHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryGroupMemberByMemberAuthHeaders = QueryGroupMemberByMemberAuthHeaders;
class QueryGroupMemberByMemberAuthRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            coolAppCode: 'coolAppCode',
            openConversationId: 'openConversationId',
        };
    }
    static types() {
        return {
            coolAppCode: 'string',
            openConversationId: 'string',
        };
    }
}
exports.QueryGroupMemberByMemberAuthRequest = QueryGroupMemberByMemberAuthRequest;
class QueryGroupMemberByMemberAuthResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupMemberList: 'groupMemberList',
        };
    }
    static types() {
        return {
            groupMemberList: { 'type': 'array', 'itemType': QueryGroupMemberByMemberAuthResponseBodyGroupMemberList },
        };
    }
}
exports.QueryGroupMemberByMemberAuthResponseBody = QueryGroupMemberByMemberAuthResponseBody;
class QueryGroupMemberByMemberAuthResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: QueryGroupMemberByMemberAuthResponseBody,
        };
    }
}
exports.QueryGroupMemberByMemberAuthResponse = QueryGroupMemberByMemberAuthResponse;
class QueryGroupMuteStatusHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryGroupMuteStatusHeaders = QueryGroupMuteStatusHeaders;
class QueryGroupMuteStatusRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            openConversationId: 'openConversationId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            openConversationId: 'string',
            userId: 'string',
        };
    }
}
exports.QueryGroupMuteStatusRequest = QueryGroupMuteStatusRequest;
class QueryGroupMuteStatusResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupMuteMode: 'groupMuteMode',
            userMuteResult: 'userMuteResult',
        };
    }
    static types() {
        return {
            groupMuteMode: 'boolean',
            userMuteResult: QueryGroupMuteStatusResponseBodyUserMuteResult,
        };
    }
}
exports.QueryGroupMuteStatusResponseBody = QueryGroupMuteStatusResponseBody;
class QueryGroupMuteStatusResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: QueryGroupMuteStatusResponseBody,
        };
    }
}
exports.QueryGroupMuteStatusResponse = QueryGroupMuteStatusResponse;
class QueryMembersOfGroupRoleHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryMembersOfGroupRoleHeaders = QueryMembersOfGroupRoleHeaders;
class QueryMembersOfGroupRoleRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            openConversationId: 'openConversationId',
            openRoleId: 'openRoleId',
            timestamp: 'timestamp',
        };
    }
    static types() {
        return {
            openConversationId: 'string',
            openRoleId: 'string',
            timestamp: 'number',
        };
    }
}
exports.QueryMembersOfGroupRoleRequest = QueryMembersOfGroupRoleRequest;
class QueryMembersOfGroupRoleResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            userIds: 'userIds',
        };
    }
    static types() {
        return {
            userIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.QueryMembersOfGroupRoleResponseBody = QueryMembersOfGroupRoleResponseBody;
class QueryMembersOfGroupRoleResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: QueryMembersOfGroupRoleResponseBody,
        };
    }
}
exports.QueryMembersOfGroupRoleResponse = QueryMembersOfGroupRoleResponse;
class QuerySingleGroupHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QuerySingleGroupHeaders = QuerySingleGroupHeaders;
class QuerySingleGroupRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupMembers: 'groupMembers',
            groupTemplateId: 'groupTemplateId',
        };
    }
    static types() {
        return {
            groupMembers: { 'type': 'array', 'itemType': QuerySingleGroupRequestGroupMembers },
            groupTemplateId: 'string',
        };
    }
}
exports.QuerySingleGroupRequest = QuerySingleGroupRequest;
class QuerySingleGroupResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            openConversations: 'openConversations',
        };
    }
    static types() {
        return {
            openConversations: { 'type': 'array', 'itemType': QuerySingleGroupResponseBodyOpenConversations },
        };
    }
}
exports.QuerySingleGroupResponseBody = QuerySingleGroupResponseBody;
class QuerySingleGroupResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: QuerySingleGroupResponseBody,
        };
    }
}
exports.QuerySingleGroupResponse = QuerySingleGroupResponse;
class QueryUnReadMessageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryUnReadMessageHeaders = QueryUnReadMessageHeaders;
class QueryUnReadMessageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appUserId: 'appUserId',
            openConversationIds: 'openConversationIds',
        };
    }
    static types() {
        return {
            appUserId: 'string',
            openConversationIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.QueryUnReadMessageRequest = QueryUnReadMessageRequest;
class QueryUnReadMessageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            unReadCount: 'unReadCount',
            unReadItems: 'unReadItems',
        };
    }
    static types() {
        return {
            unReadCount: 'number',
            unReadItems: { 'type': 'array', 'itemType': QueryUnReadMessageResponseBodyUnReadItems },
        };
    }
}
exports.QueryUnReadMessageResponseBody = QueryUnReadMessageResponseBody;
class QueryUnReadMessageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: QueryUnReadMessageResponseBody,
        };
    }
}
exports.QueryUnReadMessageResponse = QueryUnReadMessageResponse;
class SendInteractiveCardHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SendInteractiveCardHeaders = SendInteractiveCardHeaders;
class SendInteractiveCardRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            atOpenIds: 'atOpenIds',
            callbackRouteKey: 'callbackRouteKey',
            cardData: 'cardData',
            cardOptions: 'cardOptions',
            cardTemplateId: 'cardTemplateId',
            chatBotId: 'chatBotId',
            conversationType: 'conversationType',
            openConversationId: 'openConversationId',
            outTrackId: 'outTrackId',
            privateData: 'privateData',
            pullStrategy: 'pullStrategy',
            receiverUserIdList: 'receiverUserIdList',
            robotCode: 'robotCode',
            userIdType: 'userIdType',
        };
    }
    static types() {
        return {
            atOpenIds: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            callbackRouteKey: 'string',
            cardData: SendInteractiveCardRequestCardData,
            cardOptions: SendInteractiveCardRequestCardOptions,
            cardTemplateId: 'string',
            chatBotId: 'string',
            conversationType: 'number',
            openConversationId: 'string',
            outTrackId: 'string',
            privateData: { 'type': 'map', 'keyType': 'string', 'valueType': PrivateDataValue },
            pullStrategy: 'boolean',
            receiverUserIdList: { 'type': 'array', 'itemType': 'string' },
            robotCode: 'string',
            userIdType: 'number',
        };
    }
}
exports.SendInteractiveCardRequest = SendInteractiveCardRequest;
class SendInteractiveCardResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
            success: 'success',
        };
    }
    static types() {
        return {
            result: SendInteractiveCardResponseBodyResult,
            success: 'boolean',
        };
    }
}
exports.SendInteractiveCardResponseBody = SendInteractiveCardResponseBody;
class SendInteractiveCardResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: SendInteractiveCardResponseBody,
        };
    }
}
exports.SendInteractiveCardResponse = SendInteractiveCardResponse;
class SendRobotInteractiveCardHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SendRobotInteractiveCardHeaders = SendRobotInteractiveCardHeaders;
class SendRobotInteractiveCardRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            callbackUrl: 'callbackUrl',
            cardBizId: 'cardBizId',
            cardData: 'cardData',
            cardTemplateId: 'cardTemplateId',
            openConversationId: 'openConversationId',
            pullStrategy: 'pullStrategy',
            robotCode: 'robotCode',
            sendOptions: 'sendOptions',
            singleChatReceiver: 'singleChatReceiver',
            unionIdPrivateDataMap: 'unionIdPrivateDataMap',
            userIdPrivateDataMap: 'userIdPrivateDataMap',
        };
    }
    static types() {
        return {
            callbackUrl: 'string',
            cardBizId: 'string',
            cardData: 'string',
            cardTemplateId: 'string',
            openConversationId: 'string',
            pullStrategy: 'boolean',
            robotCode: 'string',
            sendOptions: SendRobotInteractiveCardRequestSendOptions,
            singleChatReceiver: 'string',
            unionIdPrivateDataMap: 'string',
            userIdPrivateDataMap: 'string',
        };
    }
}
exports.SendRobotInteractiveCardRequest = SendRobotInteractiveCardRequest;
class SendRobotInteractiveCardResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            processQueryKey: 'processQueryKey',
        };
    }
    static types() {
        return {
            processQueryKey: 'string',
        };
    }
}
exports.SendRobotInteractiveCardResponseBody = SendRobotInteractiveCardResponseBody;
class SendRobotInteractiveCardResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: SendRobotInteractiveCardResponseBody,
        };
    }
}
exports.SendRobotInteractiveCardResponse = SendRobotInteractiveCardResponse;
class SendRobotMessageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SendRobotMessageHeaders = SendRobotMessageHeaders;
class SendRobotMessageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            atAll: 'atAll',
            atAppUserId: 'atAppUserId',
            atDingUserId: 'atDingUserId',
            msgContent: 'msgContent',
            msgType: 'msgType',
            openConversationIds: 'openConversationIds',
        };
    }
    static types() {
        return {
            atAll: 'boolean',
            atAppUserId: 'string',
            atDingUserId: 'string',
            msgContent: 'string',
            msgType: 'string',
            openConversationIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.SendRobotMessageRequest = SendRobotMessageRequest;
class SendRobotMessageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
}
exports.SendRobotMessageResponseBody = SendRobotMessageResponseBody;
class SendRobotMessageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: SendRobotMessageResponseBody,
        };
    }
}
exports.SendRobotMessageResponse = SendRobotMessageResponse;
class SendTemplateInteractiveCardHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SendTemplateInteractiveCardHeaders = SendTemplateInteractiveCardHeaders;
class SendTemplateInteractiveCardRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            callbackUrl: 'callbackUrl',
            cardData: 'cardData',
            cardTemplateId: 'cardTemplateId',
            openConversationId: 'openConversationId',
            outTrackId: 'outTrackId',
            robotCode: 'robotCode',
            sendOptions: 'sendOptions',
            singleChatReceiver: 'singleChatReceiver',
        };
    }
    static types() {
        return {
            callbackUrl: 'string',
            cardData: 'string',
            cardTemplateId: 'string',
            openConversationId: 'string',
            outTrackId: 'string',
            robotCode: 'string',
            sendOptions: SendTemplateInteractiveCardRequestSendOptions,
            singleChatReceiver: 'string',
        };
    }
}
exports.SendTemplateInteractiveCardRequest = SendTemplateInteractiveCardRequest;
class SendTemplateInteractiveCardResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            processQueryKey: 'processQueryKey',
        };
    }
    static types() {
        return {
            processQueryKey: 'string',
        };
    }
}
exports.SendTemplateInteractiveCardResponseBody = SendTemplateInteractiveCardResponseBody;
class SendTemplateInteractiveCardResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: SendTemplateInteractiveCardResponseBody,
        };
    }
}
exports.SendTemplateInteractiveCardResponse = SendTemplateInteractiveCardResponse;
class TopboxCloseHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.TopboxCloseHeaders = TopboxCloseHeaders;
class TopboxCloseRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            conversationType: 'conversationType',
            coolAppCode: 'coolAppCode',
            openConversationId: 'openConversationId',
            outTrackId: 'outTrackId',
            receiverUserIdList: 'receiverUserIdList',
            robotCode: 'robotCode',
        };
    }
    static types() {
        return {
            conversationType: 'number',
            coolAppCode: 'string',
            openConversationId: 'string',
            outTrackId: 'string',
            receiverUserIdList: { 'type': 'array', 'itemType': 'string' },
            robotCode: 'string',
        };
    }
}
exports.TopboxCloseRequest = TopboxCloseRequest;
class TopboxCloseResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.TopboxCloseResponse = TopboxCloseResponse;
class TopboxOpenHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.TopboxOpenHeaders = TopboxOpenHeaders;
class TopboxOpenRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            conversationType: 'conversationType',
            coolAppCode: 'coolAppCode',
            expiredTime: 'expiredTime',
            openConversationId: 'openConversationId',
            outTrackId: 'outTrackId',
            platforms: 'platforms',
            receiverUserIdList: 'receiverUserIdList',
            robotCode: 'robotCode',
        };
    }
    static types() {
        return {
            conversationType: 'number',
            coolAppCode: 'string',
            expiredTime: 'number',
            openConversationId: 'string',
            outTrackId: 'string',
            platforms: 'string',
            receiverUserIdList: { 'type': 'array', 'itemType': 'string' },
            robotCode: 'string',
        };
    }
}
exports.TopboxOpenRequest = TopboxOpenRequest;
class TopboxOpenResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.TopboxOpenResponse = TopboxOpenResponse;
class UpdateGroupAvatarHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateGroupAvatarHeaders = UpdateGroupAvatarHeaders;
class UpdateGroupAvatarRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupAvatar: 'groupAvatar',
            openConversationId: 'openConversationId',
        };
    }
    static types() {
        return {
            groupAvatar: 'string',
            openConversationId: 'string',
        };
    }
}
exports.UpdateGroupAvatarRequest = UpdateGroupAvatarRequest;
class UpdateGroupAvatarResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            newGroupAvatar: 'newGroupAvatar',
        };
    }
    static types() {
        return {
            newGroupAvatar: 'string',
        };
    }
}
exports.UpdateGroupAvatarResponseBody = UpdateGroupAvatarResponseBody;
class UpdateGroupAvatarResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: UpdateGroupAvatarResponseBody,
        };
    }
}
exports.UpdateGroupAvatarResponse = UpdateGroupAvatarResponse;
class UpdateGroupNameHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateGroupNameHeaders = UpdateGroupNameHeaders;
class UpdateGroupNameRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupName: 'groupName',
            openConversationId: 'openConversationId',
        };
    }
    static types() {
        return {
            groupName: 'string',
            openConversationId: 'string',
        };
    }
}
exports.UpdateGroupNameRequest = UpdateGroupNameRequest;
class UpdateGroupNameResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            newGroupName: 'newGroupName',
        };
    }
    static types() {
        return {
            newGroupName: 'string',
        };
    }
}
exports.UpdateGroupNameResponseBody = UpdateGroupNameResponseBody;
class UpdateGroupNameResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: UpdateGroupNameResponseBody,
        };
    }
}
exports.UpdateGroupNameResponse = UpdateGroupNameResponse;
class UpdateGroupPermissionHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateGroupPermissionHeaders = UpdateGroupPermissionHeaders;
class UpdateGroupPermissionRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            openConversationId: 'openConversationId',
            permissionGroup: 'permissionGroup',
            status: 'status',
        };
    }
    static types() {
        return {
            openConversationId: 'string',
            permissionGroup: 'string',
            status: 'string',
        };
    }
}
exports.UpdateGroupPermissionRequest = UpdateGroupPermissionRequest;
class UpdateGroupPermissionResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
}
exports.UpdateGroupPermissionResponseBody = UpdateGroupPermissionResponseBody;
class UpdateGroupPermissionResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: UpdateGroupPermissionResponseBody,
        };
    }
}
exports.UpdateGroupPermissionResponse = UpdateGroupPermissionResponse;
class UpdateGroupSubAdminHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateGroupSubAdminHeaders = UpdateGroupSubAdminHeaders;
class UpdateGroupSubAdminRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            openConversationId: 'openConversationId',
            role: 'role',
            userIds: 'userIds',
        };
    }
    static types() {
        return {
            openConversationId: 'string',
            role: 'number',
            userIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.UpdateGroupSubAdminRequest = UpdateGroupSubAdminRequest;
class UpdateGroupSubAdminResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
}
exports.UpdateGroupSubAdminResponseBody = UpdateGroupSubAdminResponseBody;
class UpdateGroupSubAdminResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: UpdateGroupSubAdminResponseBody,
        };
    }
}
exports.UpdateGroupSubAdminResponse = UpdateGroupSubAdminResponse;
class UpdateInteractiveCardHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateInteractiveCardHeaders = UpdateInteractiveCardHeaders;
class UpdateInteractiveCardRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            cardData: 'cardData',
            cardOptions: 'cardOptions',
            outTrackId: 'outTrackId',
            privateData: 'privateData',
            userIdType: 'userIdType',
        };
    }
    static types() {
        return {
            cardData: UpdateInteractiveCardRequestCardData,
            cardOptions: UpdateInteractiveCardRequestCardOptions,
            outTrackId: 'string',
            privateData: { 'type': 'map', 'keyType': 'string', 'valueType': PrivateDataValue },
            userIdType: 'number',
        };
    }
}
exports.UpdateInteractiveCardRequest = UpdateInteractiveCardRequest;
class UpdateInteractiveCardResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'string',
        };
    }
}
exports.UpdateInteractiveCardResponseBody = UpdateInteractiveCardResponseBody;
class UpdateInteractiveCardResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: UpdateInteractiveCardResponseBody,
        };
    }
}
exports.UpdateInteractiveCardResponse = UpdateInteractiveCardResponse;
class UpdateMemberBanWordsHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateMemberBanWordsHeaders = UpdateMemberBanWordsHeaders;
class UpdateMemberBanWordsRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            muteDuration: 'muteDuration',
            muteStatus: 'muteStatus',
            openConversationId: 'openConversationId',
            userIdList: 'userIdList',
        };
    }
    static types() {
        return {
            muteDuration: 'number',
            muteStatus: 'number',
            openConversationId: 'string',
            userIdList: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.UpdateMemberBanWordsRequest = UpdateMemberBanWordsRequest;
class UpdateMemberBanWordsResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.UpdateMemberBanWordsResponse = UpdateMemberBanWordsResponse;
class UpdateMemberGroupNickHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateMemberGroupNickHeaders = UpdateMemberGroupNickHeaders;
class UpdateMemberGroupNickRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupNick: 'groupNick',
            openConversationId: 'openConversationId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            groupNick: 'string',
            openConversationId: 'string',
            userId: 'string',
        };
    }
}
exports.UpdateMemberGroupNickRequest = UpdateMemberGroupNickRequest;
class UpdateMemberGroupNickResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
}
exports.UpdateMemberGroupNickResponseBody = UpdateMemberGroupNickResponseBody;
class UpdateMemberGroupNickResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: UpdateMemberGroupNickResponseBody,
        };
    }
}
exports.UpdateMemberGroupNickResponse = UpdateMemberGroupNickResponse;
class UpdateRobotInteractiveCardHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateRobotInteractiveCardHeaders = UpdateRobotInteractiveCardHeaders;
class UpdateRobotInteractiveCardRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            cardBizId: 'cardBizId',
            cardData: 'cardData',
            unionIdPrivateDataMap: 'unionIdPrivateDataMap',
            updateOptions: 'updateOptions',
            userIdPrivateDataMap: 'userIdPrivateDataMap',
        };
    }
    static types() {
        return {
            cardBizId: 'string',
            cardData: 'string',
            unionIdPrivateDataMap: 'string',
            updateOptions: UpdateRobotInteractiveCardRequestUpdateOptions,
            userIdPrivateDataMap: 'string',
        };
    }
}
exports.UpdateRobotInteractiveCardRequest = UpdateRobotInteractiveCardRequest;
class UpdateRobotInteractiveCardResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            processQueryKey: 'processQueryKey',
        };
    }
    static types() {
        return {
            processQueryKey: 'string',
        };
    }
}
exports.UpdateRobotInteractiveCardResponseBody = UpdateRobotInteractiveCardResponseBody;
class UpdateRobotInteractiveCardResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: UpdateRobotInteractiveCardResponseBody,
        };
    }
}
exports.UpdateRobotInteractiveCardResponse = UpdateRobotInteractiveCardResponse;
class UpdateTheGroupRolesOfGroupMemberHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateTheGroupRolesOfGroupMemberHeaders = UpdateTheGroupRolesOfGroupMemberHeaders;
class UpdateTheGroupRolesOfGroupMemberRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            openConversationId: 'openConversationId',
            openRoleIds: 'openRoleIds',
            userId: 'userId',
        };
    }
    static types() {
        return {
            openConversationId: 'string',
            openRoleIds: { 'type': 'array', 'itemType': 'string' },
            userId: 'string',
        };
    }
}
exports.UpdateTheGroupRolesOfGroupMemberRequest = UpdateTheGroupRolesOfGroupMemberRequest;
class UpdateTheGroupRolesOfGroupMemberResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
}
exports.UpdateTheGroupRolesOfGroupMemberResponseBody = UpdateTheGroupRolesOfGroupMemberResponseBody;
class UpdateTheGroupRolesOfGroupMemberResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: UpdateTheGroupRolesOfGroupMemberResponseBody,
        };
    }
}
exports.UpdateTheGroupRolesOfGroupMemberResponse = UpdateTheGroupRolesOfGroupMemberResponse;
class AddGroupMemberHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.AddGroupMemberHeaders = AddGroupMemberHeaders;
class AddGroupMemberRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appUserIds: 'appUserIds',
            openConversationId: 'openConversationId',
            operatorId: 'operatorId',
            userIds: 'userIds',
        };
    }
    static types() {
        return {
            appUserIds: { 'type': 'array', 'itemType': 'string' },
            openConversationId: 'string',
            operatorId: 'string',
            userIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.AddGroupMemberRequest = AddGroupMemberRequest;
class AddGroupMemberResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appUserIds: 'appUserIds',
            userIds: 'userIds',
        };
    }
    static types() {
        return {
            appUserIds: { 'type': 'array', 'itemType': 'string' },
            userIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.AddGroupMemberResponseBody = AddGroupMemberResponseBody;
class AddGroupMemberResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: AddGroupMemberResponseBody,
        };
    }
}
exports.AddGroupMemberResponse = AddGroupMemberResponse;
class RemoveGroupMemberHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.RemoveGroupMemberHeaders = RemoveGroupMemberHeaders;
class RemoveGroupMemberRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appUserIds: 'appUserIds',
            openConversationId: 'openConversationId',
            operatorId: 'operatorId',
            userIds: 'userIds',
        };
    }
    static types() {
        return {
            appUserIds: { 'type': 'array', 'itemType': 'string' },
            openConversationId: 'string',
            operatorId: 'string',
            userIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.RemoveGroupMemberRequest = RemoveGroupMemberRequest;
class RemoveGroupMemberResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.RemoveGroupMemberResponse = RemoveGroupMemberResponse;
class SendDingMessageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SendDingMessageHeaders = SendDingMessageHeaders;
class SendDingMessageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            code: 'code',
            message: 'message',
            messageType: 'messageType',
            openConversationId: 'openConversationId',
            receiverId: 'receiverId',
            senderId: 'senderId',
        };
    }
    static types() {
        return {
            code: 'string',
            message: 'string',
            messageType: 'string',
            openConversationId: 'string',
            receiverId: 'string',
            senderId: 'string',
        };
    }
}
exports.SendDingMessageRequest = SendDingMessageRequest;
class SendDingMessageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
        };
    }
    static types() {
        return {
            requestId: 'string',
        };
    }
}
exports.SendDingMessageResponseBody = SendDingMessageResponseBody;
class SendDingMessageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: SendDingMessageResponseBody,
        };
    }
}
exports.SendDingMessageResponse = SendDingMessageResponse;
class SendMessageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SendMessageHeaders = SendMessageHeaders;
class SendMessageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            message: 'message',
            messageType: 'messageType',
            openConversationId: 'openConversationId',
            receiverId: 'receiverId',
            senderId: 'senderId',
            sourceInfos: 'sourceInfos',
        };
    }
    static types() {
        return {
            message: 'string',
            messageType: 'string',
            openConversationId: 'string',
            receiverId: 'string',
            senderId: 'string',
            sourceInfos: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
        };
    }
}
exports.SendMessageRequest = SendMessageRequest;
class SendMessageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            requestId: 'requestId',
        };
    }
    static types() {
        return {
            requestId: 'string',
        };
    }
}
exports.SendMessageResponseBody = SendMessageResponseBody;
class SendMessageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: SendMessageResponseBody,
        };
    }
}
exports.SendMessageResponse = SendMessageResponse;
class PrivateDataValue extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            cardParamMap: 'cardParamMap',
            cardMediaIdParamMap: 'cardMediaIdParamMap',
        };
    }
    static types() {
        return {
            cardParamMap: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            cardMediaIdParamMap: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.PrivateDataValue = PrivateDataValue;
class CreateInterconnectionRequestInterconnections extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appUserAvatar: 'appUserAvatar',
            appUserAvatarMediaType: 'appUserAvatarMediaType',
            appUserDynamics: 'appUserDynamics',
            appUserId: 'appUserId',
            appUserMobile: 'appUserMobile',
            appUserName: 'appUserName',
            channelCode: 'channelCode',
            userId: 'userId',
        };
    }
    static types() {
        return {
            appUserAvatar: 'string',
            appUserAvatarMediaType: 'number',
            appUserDynamics: 'string',
            appUserId: 'string',
            appUserMobile: 'string',
            appUserName: 'string',
            channelCode: 'string',
            userId: 'string',
        };
    }
}
exports.CreateInterconnectionRequestInterconnections = CreateInterconnectionRequestInterconnections;
class CreateInterconnectionResponseBodyResults extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appUserId: 'appUserId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            appUserId: 'string',
            userId: 'string',
        };
    }
}
exports.CreateInterconnectionResponseBodyResults = CreateInterconnectionResponseBodyResults;
class GroupManageQueryResponseBodyGroupInfoList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            banWordsMode: 'banWordsMode',
            capacity: 'capacity',
            createdAt: 'createdAt',
            extInfo: 'extInfo',
            groupAdminList: 'groupAdminList',
            groupOwner: 'groupOwner',
            groupTitle: 'groupTitle',
            memberCount: 'memberCount',
            openConversationId: 'openConversationId',
            type: 'type',
        };
    }
    static types() {
        return {
            banWordsMode: 'number',
            capacity: 'number',
            createdAt: 'number',
            extInfo: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            groupAdminList: { 'type': 'array', 'itemType': 'string' },
            groupOwner: 'string',
            groupTitle: 'string',
            memberCount: 'number',
            openConversationId: 'string',
            type: 'string',
        };
    }
}
exports.GroupManageQueryResponseBodyGroupInfoList = GroupManageQueryResponseBodyGroupInfoList;
class InteractiveCardCreateInstanceRequestCardData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            cardMediaIdParamMap: 'cardMediaIdParamMap',
            cardParamMap: 'cardParamMap',
        };
    }
    static types() {
        return {
            cardMediaIdParamMap: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            cardParamMap: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.InteractiveCardCreateInstanceRequestCardData = InteractiveCardCreateInstanceRequestCardData;
class QueryGroupMemberResponseBodyGroupMembers extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupMemberAvatar: 'groupMemberAvatar',
            groupMemberDynamics: 'groupMemberDynamics',
            groupMemberId: 'groupMemberId',
            groupMemberName: 'groupMemberName',
            groupMemberType: 'groupMemberType',
        };
    }
    static types() {
        return {
            groupMemberAvatar: 'string',
            groupMemberDynamics: 'string',
            groupMemberId: 'string',
            groupMemberName: 'string',
            groupMemberType: 'number',
        };
    }
}
exports.QueryGroupMemberResponseBodyGroupMembers = QueryGroupMemberResponseBodyGroupMembers;
class QueryGroupMemberByMemberAuthResponseBodyGroupMemberList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupNickName: 'groupNickName',
            orgName: 'orgName',
            profilePhotoUrl: 'profilePhotoUrl',
            userId: 'userId',
        };
    }
    static types() {
        return {
            groupNickName: 'string',
            orgName: 'string',
            profilePhotoUrl: 'string',
            userId: 'string',
        };
    }
}
exports.QueryGroupMemberByMemberAuthResponseBodyGroupMemberList = QueryGroupMemberByMemberAuthResponseBodyGroupMemberList;
class QueryGroupMuteStatusResponseBodyUserMuteResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            muteEndTime: 'muteEndTime',
            muteStartTime: 'muteStartTime',
            userMuteMode: 'userMuteMode',
        };
    }
    static types() {
        return {
            muteEndTime: 'number',
            muteStartTime: 'number',
            userMuteMode: 'boolean',
        };
    }
}
exports.QueryGroupMuteStatusResponseBodyUserMuteResult = QueryGroupMuteStatusResponseBodyUserMuteResult;
class QuerySingleGroupRequestGroupMembers extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appUserId: 'appUserId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            appUserId: 'string',
            userId: 'string',
        };
    }
}
exports.QuerySingleGroupRequestGroupMembers = QuerySingleGroupRequestGroupMembers;
class QuerySingleGroupResponseBodyOpenConversations extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appUserId: 'appUserId',
            openConversationId: 'openConversationId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            appUserId: 'string',
            openConversationId: 'string',
            userId: 'string',
        };
    }
}
exports.QuerySingleGroupResponseBodyOpenConversations = QuerySingleGroupResponseBodyOpenConversations;
class QueryUnReadMessageResponseBodyUnReadItems extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            openConversationId: 'openConversationId',
            unReadCount: 'unReadCount',
        };
    }
    static types() {
        return {
            openConversationId: 'string',
            unReadCount: 'number',
        };
    }
}
exports.QueryUnReadMessageResponseBodyUnReadItems = QueryUnReadMessageResponseBodyUnReadItems;
class SendInteractiveCardRequestCardData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            cardMediaIdParamMap: 'cardMediaIdParamMap',
            cardParamMap: 'cardParamMap',
        };
    }
    static types() {
        return {
            cardMediaIdParamMap: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            cardParamMap: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.SendInteractiveCardRequestCardData = SendInteractiveCardRequestCardData;
class SendInteractiveCardRequestCardOptions extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            supportForward: 'supportForward',
        };
    }
    static types() {
        return {
            supportForward: 'boolean',
        };
    }
}
exports.SendInteractiveCardRequestCardOptions = SendInteractiveCardRequestCardOptions;
class SendInteractiveCardResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            processQueryKey: 'processQueryKey',
        };
    }
    static types() {
        return {
            processQueryKey: 'string',
        };
    }
}
exports.SendInteractiveCardResponseBodyResult = SendInteractiveCardResponseBodyResult;
class SendRobotInteractiveCardRequestSendOptions extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            atAll: 'atAll',
            atUserListJson: 'atUserListJson',
            cardPropertyJson: 'cardPropertyJson',
            receiverListJson: 'receiverListJson',
        };
    }
    static types() {
        return {
            atAll: 'boolean',
            atUserListJson: 'string',
            cardPropertyJson: 'string',
            receiverListJson: 'string',
        };
    }
}
exports.SendRobotInteractiveCardRequestSendOptions = SendRobotInteractiveCardRequestSendOptions;
class SendTemplateInteractiveCardRequestSendOptions extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            atAll: 'atAll',
            atUserListJson: 'atUserListJson',
            cardPropertyJson: 'cardPropertyJson',
            receiverListJson: 'receiverListJson',
        };
    }
    static types() {
        return {
            atAll: 'boolean',
            atUserListJson: 'string',
            cardPropertyJson: 'string',
            receiverListJson: 'string',
        };
    }
}
exports.SendTemplateInteractiveCardRequestSendOptions = SendTemplateInteractiveCardRequestSendOptions;
class UpdateInteractiveCardRequestCardData extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            cardMediaIdParamMap: 'cardMediaIdParamMap',
            cardParamMap: 'cardParamMap',
        };
    }
    static types() {
        return {
            cardMediaIdParamMap: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            cardParamMap: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.UpdateInteractiveCardRequestCardData = UpdateInteractiveCardRequestCardData;
class UpdateInteractiveCardRequestCardOptions extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            updateCardDataByKey: 'updateCardDataByKey',
            updatePrivateDataByKey: 'updatePrivateDataByKey',
        };
    }
    static types() {
        return {
            updateCardDataByKey: 'boolean',
            updatePrivateDataByKey: 'boolean',
        };
    }
}
exports.UpdateInteractiveCardRequestCardOptions = UpdateInteractiveCardRequestCardOptions;
class UpdateRobotInteractiveCardRequestUpdateOptions extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            updateCardDataByKey: 'updateCardDataByKey',
            updatePrivateDataByKey: 'updatePrivateDataByKey',
        };
    }
    static types() {
        return {
            updateCardDataByKey: 'boolean',
            updatePrivateDataByKey: 'boolean',
        };
    }
}
exports.UpdateRobotInteractiveCardRequestUpdateOptions = UpdateRobotInteractiveCardRequestUpdateOptions;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async autoOpenDingTalkConnect() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AutoOpenDingTalkConnectHeaders({});
        return await this.autoOpenDingTalkConnectWithOptions(headers, runtime);
    }
    async autoOpenDingTalkConnectWithOptions(headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("AutoOpenDingTalkConnect", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interconnections/apps/open`, "json", req, runtime), new AutoOpenDingTalkConnectResponse({}));
    }
    async batchQueryGroupMember(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new BatchQueryGroupMemberHeaders({});
        return await this.batchQueryGroupMemberWithOptions(request, headers, runtime);
    }
    async batchQueryGroupMemberWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.coolAppCode)) {
            body["coolAppCode"] = request.coolAppCode;
        }
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            body["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            body["nextToken"] = request.nextToken;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("BatchQueryGroupMember", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/sceneGroups/members/batchQuery`, "json", req, runtime), new BatchQueryGroupMemberResponse({}));
    }
    async cardTemplateBuildAction(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CardTemplateBuildActionHeaders({});
        return await this.cardTemplateBuildActionWithOptions(request, headers, runtime);
    }
    async cardTemplateBuildActionWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.action)) {
            body["action"] = request.action;
        }
        if (!tea_util_1.default.isUnset(request.cardTemplateJson)) {
            body["cardTemplateJson"] = request.cardTemplateJson;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("CardTemplateBuildAction", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interactiveCards/templates/buildAction`, "json", req, runtime), new CardTemplateBuildActionResponse({}));
    }
    async chatIdToOpenConversationId(chatId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ChatIdToOpenConversationIdHeaders({});
        return await this.chatIdToOpenConversationIdWithOptions(chatId, headers, runtime);
    }
    async chatIdToOpenConversationIdWithOptions(chatId, headers, runtime) {
        chatId = openapi_util_1.default.getEncodeParam(chatId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("ChatIdToOpenConversationId", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/chat/${chatId}/convertToOpenConversationId`, "json", req, runtime), new ChatIdToOpenConversationIdResponse({}));
    }
    async chatSubAdminUpdate(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ChatSubAdminUpdateHeaders({});
        return await this.chatSubAdminUpdateWithOptions(request, headers, runtime);
    }
    async chatSubAdminUpdateWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.role)) {
            body["role"] = request.role;
        }
        if (!tea_util_1.default.isUnset(request.userIds)) {
            body["userIds"] = request.userIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("ChatSubAdminUpdate", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/subAdministrators`, "json", req, runtime), new ChatSubAdminUpdateResponse({}));
    }
    async createCoupleGroupConversation(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateCoupleGroupConversationHeaders({});
        return await this.createCoupleGroupConversationWithOptions(request, headers, runtime);
    }
    async createCoupleGroupConversationWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.appUserId)) {
            body["appUserId"] = request.appUserId;
        }
        if (!tea_util_1.default.isUnset(request.groupAvatar)) {
            body["groupAvatar"] = request.groupAvatar;
        }
        if (!tea_util_1.default.isUnset(request.groupName)) {
            body["groupName"] = request.groupName;
        }
        if (!tea_util_1.default.isUnset(request.groupOwnerId)) {
            body["groupOwnerId"] = request.groupOwnerId;
        }
        if (!tea_util_1.default.isUnset(request.groupTemplateId)) {
            body["groupTemplateId"] = request.groupTemplateId;
        }
        if (!tea_util_1.default.isUnset(request.operatorId)) {
            body["operatorId"] = request.operatorId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("CreateCoupleGroupConversation", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interconnections/coupleGroups`, "json", req, runtime), new CreateCoupleGroupConversationResponse({}));
    }
    async createGroupConversation(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateGroupConversationHeaders({});
        return await this.createGroupConversationWithOptions(request, headers, runtime);
    }
    async createGroupConversationWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.appUserIds)) {
            body["appUserIds"] = request.appUserIds;
        }
        if (!tea_util_1.default.isUnset(request.groupAvatar)) {
            body["groupAvatar"] = request.groupAvatar;
        }
        if (!tea_util_1.default.isUnset(request.groupName)) {
            body["groupName"] = request.groupName;
        }
        if (!tea_util_1.default.isUnset(request.groupOwnerId)) {
            body["groupOwnerId"] = request.groupOwnerId;
        }
        if (!tea_util_1.default.isUnset(request.groupTemplateId)) {
            body["groupTemplateId"] = request.groupTemplateId;
        }
        if (!tea_util_1.default.isUnset(request.operatorId)) {
            body["operatorId"] = request.operatorId;
        }
        if (!tea_util_1.default.isUnset(request.userIds)) {
            body["userIds"] = request.userIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("CreateGroupConversation", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interconnections/groups`, "json", req, runtime), new CreateGroupConversationResponse({}));
    }
    async createInterconnection(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateInterconnectionHeaders({});
        return await this.createInterconnectionWithOptions(request, headers, runtime);
    }
    async createInterconnectionWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.interconnections)) {
            body["interconnections"] = request.interconnections;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("CreateInterconnection", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interconnections`, "json", req, runtime), new CreateInterconnectionResponse({}));
    }
    async createStoreGroupConversation(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateStoreGroupConversationHeaders({});
        return await this.createStoreGroupConversationWithOptions(request, headers, runtime);
    }
    async createStoreGroupConversationWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.appUserId)) {
            body["appUserId"] = request.appUserId;
        }
        if (!tea_util_1.default.isUnset(request.businessUniqueKey)) {
            body["businessUniqueKey"] = request.businessUniqueKey;
        }
        if (!tea_util_1.default.isUnset(request.groupAvatar)) {
            body["groupAvatar"] = request.groupAvatar;
        }
        if (!tea_util_1.default.isUnset(request.groupName)) {
            body["groupName"] = request.groupName;
        }
        if (!tea_util_1.default.isUnset(request.groupTemplateId)) {
            body["groupTemplateId"] = request.groupTemplateId;
        }
        if (!tea_util_1.default.isUnset(request.operatorId)) {
            body["operatorId"] = request.operatorId;
        }
        if (!tea_util_1.default.isUnset(request.userIds)) {
            body["userIds"] = request.userIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("CreateStoreGroupConversation", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interconnections/storeGroups`, "json", req, runtime), new CreateStoreGroupConversationResponse({}));
    }
    async getConversationUrl(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetConversationUrlHeaders({});
        return await this.getConversationUrlWithOptions(request, headers, runtime);
    }
    async getConversationUrlWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.appUserId)) {
            body["appUserId"] = request.appUserId;
        }
        if (!tea_util_1.default.isUnset(request.channelCode)) {
            body["channelCode"] = request.channelCode;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.sourceCode)) {
            body["sourceCode"] = request.sourceCode;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("GetConversationUrl", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/conversations/urls`, "json", req, runtime), new GetConversationUrlResponse({}));
    }
    async getInterconnectionUrl(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetInterconnectionUrlHeaders({});
        return await this.getInterconnectionUrlWithOptions(request, headers, runtime);
    }
    async getInterconnectionUrlWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.appUserAvatar)) {
            body["appUserAvatar"] = request.appUserAvatar;
        }
        if (!tea_util_1.default.isUnset(request.appUserAvatarType)) {
            body["appUserAvatarType"] = request.appUserAvatarType;
        }
        if (!tea_util_1.default.isUnset(request.appUserId)) {
            body["appUserId"] = request.appUserId;
        }
        if (!tea_util_1.default.isUnset(request.appUserMobileNumber)) {
            body["appUserMobileNumber"] = request.appUserMobileNumber;
        }
        if (!tea_util_1.default.isUnset(request.appUserName)) {
            body["appUserName"] = request.appUserName;
        }
        if (!tea_util_1.default.isUnset(request.msgPageType)) {
            body["msgPageType"] = request.msgPageType;
        }
        if (!tea_util_1.default.isUnset(request.qrCode)) {
            body["qrCode"] = request.qrCode;
        }
        if (!tea_util_1.default.isUnset(request.signature)) {
            body["signature"] = request.signature;
        }
        if (!tea_util_1.default.isUnset(request.sourceCode)) {
            body["sourceCode"] = request.sourceCode;
        }
        if (!tea_util_1.default.isUnset(request.sourceType)) {
            body["sourceType"] = request.sourceType;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("GetInterconnectionUrl", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interconnections/sessions/urls`, "json", req, runtime), new GetInterconnectionUrlResponse({}));
    }
    async getSceneGroupInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetSceneGroupInfoHeaders({});
        return await this.getSceneGroupInfoWithOptions(request, headers, runtime);
    }
    async getSceneGroupInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.coolAppCode)) {
            body["coolAppCode"] = request.coolAppCode;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("GetSceneGroupInfo", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/sceneGroups/query`, "json", req, runtime), new GetSceneGroupInfoResponse({}));
    }
    async getSceneGroupMembers(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetSceneGroupMembersHeaders({});
        return await this.getSceneGroupMembersWithOptions(request, headers, runtime);
    }
    async getSceneGroupMembersWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.coolAppCode)) {
            body["coolAppCode"] = request.coolAppCode;
        }
        if (!tea_util_1.default.isUnset(request.cursor)) {
            body["cursor"] = request.cursor;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.size)) {
            body["size"] = request.size;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("GetSceneGroupMembers", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/sceneGroups/members/query`, "json", req, runtime), new GetSceneGroupMembersResponse({}));
    }
    async groupBanWords(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GroupBanWordsHeaders({});
        return await this.groupBanWordsWithOptions(request, headers, runtime);
    }
    async groupBanWordsWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.banWordsMode)) {
            body["banWordsMode"] = request.banWordsMode;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.options)) {
            body["options"] = request.options;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("GroupBanWords", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/groups/words/ban`, "none", req, runtime), new GroupBanWordsResponse({}));
    }
    async groupCapacityInquiry(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GroupCapacityInquiryHeaders({});
        return await this.groupCapacityInquiryWithOptions(request, headers, runtime);
    }
    async groupCapacityInquiryWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.effectiveDuration)) {
            body["effectiveDuration"] = request.effectiveDuration;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.operator)) {
            body["operator"] = request.operator;
        }
        if (!tea_util_1.default.isUnset(request.options)) {
            body["options"] = request.options;
        }
        if (!tea_util_1.default.isUnset(request.targetCapacity)) {
            body["targetCapacity"] = request.targetCapacity;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("GroupCapacityInquiry", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/groups/capacities/inquiries/query`, "json", req, runtime), new GroupCapacityInquiryResponse({}));
    }
    async groupCapacityOrderConfirm(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GroupCapacityOrderConfirmHeaders({});
        return await this.groupCapacityOrderConfirmWithOptions(request, headers, runtime);
    }
    async groupCapacityOrderConfirmWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.operator)) {
            body["operator"] = request.operator;
        }
        if (!tea_util_1.default.isUnset(request.orderId)) {
            body["orderId"] = request.orderId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("GroupCapacityOrderConfirm", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/groups/capacities/orders/confirm`, "none", req, runtime), new GroupCapacityOrderConfirmResponse({}));
    }
    async groupCapacityOrderPlace(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GroupCapacityOrderPlaceHeaders({});
        return await this.groupCapacityOrderPlaceWithOptions(request, headers, runtime);
    }
    async groupCapacityOrderPlaceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.actualPrice)) {
            body["actualPrice"] = request.actualPrice;
        }
        if (!tea_util_1.default.isUnset(request.currentCapacity)) {
            body["currentCapacity"] = request.currentCapacity;
        }
        if (!tea_util_1.default.isUnset(request.currentEffectUntil)) {
            body["currentEffectUntil"] = request.currentEffectUntil;
        }
        if (!tea_util_1.default.isUnset(request.discount)) {
            body["discount"] = request.discount;
        }
        if (!tea_util_1.default.isUnset(request.extInfo)) {
            body["extInfo"] = request.extInfo;
        }
        if (!tea_util_1.default.isUnset(request.markedPrice)) {
            body["markedPrice"] = request.markedPrice;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.operator)) {
            body["operator"] = request.operator;
        }
        if (!tea_util_1.default.isUnset(request.targetCapacity)) {
            body["targetCapacity"] = request.targetCapacity;
        }
        if (!tea_util_1.default.isUnset(request.targetEffectUntil)) {
            body["targetEffectUntil"] = request.targetEffectUntil;
        }
        if (!tea_util_1.default.isUnset(request.token)) {
            body["token"] = request.token;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("GroupCapacityOrderPlace", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/groups/capacities/orders/place`, "json", req, runtime), new GroupCapacityOrderPlaceResponse({}));
    }
    async groupManageQuery(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GroupManageQueryHeaders({});
        return await this.groupManageQueryWithOptions(request, headers, runtime);
    }
    async groupManageQueryWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.createdAfter)) {
            body["createdAfter"] = request.createdAfter;
        }
        if (!tea_util_1.default.isUnset(request.groupId)) {
            body["groupId"] = request.groupId;
        }
        if (!tea_util_1.default.isUnset(request.groupMemberSamples)) {
            body["groupMemberSamples"] = request.groupMemberSamples;
        }
        if (!tea_util_1.default.isUnset(request.groupOwner)) {
            body["groupOwner"] = request.groupOwner;
        }
        if (!tea_util_1.default.isUnset(request.groupTitleKeywords)) {
            body["groupTitleKeywords"] = request.groupTitleKeywords;
        }
        if (!tea_util_1.default.isUnset(request.groupUrl)) {
            body["groupUrl"] = request.groupUrl;
        }
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            body["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.membersOver)) {
            body["membersOver"] = request.membersOver;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            body["nextToken"] = request.nextToken;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("GroupManageQuery", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/groups/managements/query`, "json", req, runtime), new GroupManageQueryResponse({}));
    }
    async groupManageReduce(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GroupManageReduceHeaders({});
        return await this.groupManageReduceWithOptions(request, headers, runtime);
    }
    async groupManageReduceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.capacityLimit)) {
            body["capacityLimit"] = request.capacityLimit;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.options)) {
            body["options"] = request.options;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("GroupManageReduce", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/groups/capacities/reduce`, "none", req, runtime), new GroupManageReduceResponse({}));
    }
    async interactiveCardCreateInstance(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new InteractiveCardCreateInstanceHeaders({});
        return await this.interactiveCardCreateInstanceWithOptions(request, headers, runtime);
    }
    async interactiveCardCreateInstanceWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.callbackRouteKey)) {
            body["callbackRouteKey"] = request.callbackRouteKey;
        }
        if (!tea_util_1.default.isUnset($tea.toMap(request.cardData))) {
            body["cardData"] = request.cardData;
        }
        if (!tea_util_1.default.isUnset(request.cardTemplateId)) {
            body["cardTemplateId"] = request.cardTemplateId;
        }
        if (!tea_util_1.default.isUnset(request.chatBotId)) {
            body["chatBotId"] = request.chatBotId;
        }
        if (!tea_util_1.default.isUnset(request.conversationType)) {
            body["conversationType"] = request.conversationType;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.outTrackId)) {
            body["outTrackId"] = request.outTrackId;
        }
        if (!tea_util_1.default.isUnset(request.privateData)) {
            body["privateData"] = request.privateData;
        }
        if (!tea_util_1.default.isUnset(request.pullStrategy)) {
            body["pullStrategy"] = request.pullStrategy;
        }
        if (!tea_util_1.default.isUnset(request.receiverUserIdList)) {
            body["receiverUserIdList"] = request.receiverUserIdList;
        }
        if (!tea_util_1.default.isUnset(request.robotCode)) {
            body["robotCode"] = request.robotCode;
        }
        if (!tea_util_1.default.isUnset(request.userIdType)) {
            body["userIdType"] = request.userIdType;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("InteractiveCardCreateInstance", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interactiveCards/instances`, "json", req, runtime), new InteractiveCardCreateInstanceResponse({}));
    }
    async queryGroupInfoByMemberAuth(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryGroupInfoByMemberAuthHeaders({});
        return await this.queryGroupInfoByMemberAuthWithOptions(request, headers, runtime);
    }
    async queryGroupInfoByMemberAuthWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.coolAppCode)) {
            body["coolAppCode"] = request.coolAppCode;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("QueryGroupInfoByMemberAuth", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/memberAuthorizations/groups/query`, "json", req, runtime), new QueryGroupInfoByMemberAuthResponse({}));
    }
    async queryGroupMember(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryGroupMemberHeaders({});
        return await this.queryGroupMemberWithOptions(request, headers, runtime);
    }
    async queryGroupMemberWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            query["openConversationId"] = request.openConversationId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("QueryGroupMember", "im_1.0", "HTTP", "GET", "AK", `/v1.0/im/interconnections/conversations/members`, "json", req, runtime), new QueryGroupMemberResponse({}));
    }
    async queryGroupMemberByMemberAuth(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryGroupMemberByMemberAuthHeaders({});
        return await this.queryGroupMemberByMemberAuthWithOptions(request, headers, runtime);
    }
    async queryGroupMemberByMemberAuthWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.coolAppCode)) {
            body["coolAppCode"] = request.coolAppCode;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("QueryGroupMemberByMemberAuth", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/memberAuthorizations/groups/members/query`, "json", req, runtime), new QueryGroupMemberByMemberAuthResponse({}));
    }
    async queryGroupMuteStatus(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryGroupMuteStatusHeaders({});
        return await this.queryGroupMuteStatusWithOptions(request, headers, runtime);
    }
    async queryGroupMuteStatusWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            query["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("QueryGroupMuteStatus", "im_1.0", "HTTP", "GET", "AK", `/v1.0/im/sceneGroups/muteSettings`, "json", req, runtime), new QueryGroupMuteStatusResponse({}));
    }
    async queryMembersOfGroupRole(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryMembersOfGroupRoleHeaders({});
        return await this.queryMembersOfGroupRoleWithOptions(request, headers, runtime);
    }
    async queryMembersOfGroupRoleWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.openRoleId)) {
            body["openRoleId"] = request.openRoleId;
        }
        if (!tea_util_1.default.isUnset(request.timestamp)) {
            body["timestamp"] = request.timestamp;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("QueryMembersOfGroupRole", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/sceneGroups/roles/members/query`, "json", req, runtime), new QueryMembersOfGroupRoleResponse({}));
    }
    async querySingleGroup(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QuerySingleGroupHeaders({});
        return await this.querySingleGroupWithOptions(request, headers, runtime);
    }
    async querySingleGroupWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.groupMembers)) {
            body["groupMembers"] = request.groupMembers;
        }
        if (!tea_util_1.default.isUnset(request.groupTemplateId)) {
            body["groupTemplateId"] = request.groupTemplateId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("QuerySingleGroup", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interconnections/doubleGroups/query`, "json", req, runtime), new QuerySingleGroupResponse({}));
    }
    async queryUnReadMessage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryUnReadMessageHeaders({});
        return await this.queryUnReadMessageWithOptions(request, headers, runtime);
    }
    async queryUnReadMessageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.appUserId)) {
            body["appUserId"] = request.appUserId;
        }
        if (!tea_util_1.default.isUnset(request.openConversationIds)) {
            body["openConversationIds"] = request.openConversationIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("QueryUnReadMessage", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interconnections/unReadMsgs/query`, "json", req, runtime), new QueryUnReadMessageResponse({}));
    }
    async sendInteractiveCard(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendInteractiveCardHeaders({});
        return await this.sendInteractiveCardWithOptions(request, headers, runtime);
    }
    async sendInteractiveCardWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.atOpenIds)) {
            body["atOpenIds"] = request.atOpenIds;
        }
        if (!tea_util_1.default.isUnset(request.callbackRouteKey)) {
            body["callbackRouteKey"] = request.callbackRouteKey;
        }
        if (!tea_util_1.default.isUnset($tea.toMap(request.cardData))) {
            body["cardData"] = request.cardData;
        }
        if (!tea_util_1.default.isUnset($tea.toMap(request.cardOptions))) {
            body["cardOptions"] = request.cardOptions;
        }
        if (!tea_util_1.default.isUnset(request.cardTemplateId)) {
            body["cardTemplateId"] = request.cardTemplateId;
        }
        if (!tea_util_1.default.isUnset(request.chatBotId)) {
            body["chatBotId"] = request.chatBotId;
        }
        if (!tea_util_1.default.isUnset(request.conversationType)) {
            body["conversationType"] = request.conversationType;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.outTrackId)) {
            body["outTrackId"] = request.outTrackId;
        }
        if (!tea_util_1.default.isUnset(request.privateData)) {
            body["privateData"] = request.privateData;
        }
        if (!tea_util_1.default.isUnset(request.pullStrategy)) {
            body["pullStrategy"] = request.pullStrategy;
        }
        if (!tea_util_1.default.isUnset(request.receiverUserIdList)) {
            body["receiverUserIdList"] = request.receiverUserIdList;
        }
        if (!tea_util_1.default.isUnset(request.robotCode)) {
            body["robotCode"] = request.robotCode;
        }
        if (!tea_util_1.default.isUnset(request.userIdType)) {
            body["userIdType"] = request.userIdType;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("SendInteractiveCard", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interactiveCards/send`, "json", req, runtime), new SendInteractiveCardResponse({}));
    }
    async sendRobotInteractiveCard(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendRobotInteractiveCardHeaders({});
        return await this.sendRobotInteractiveCardWithOptions(request, headers, runtime);
    }
    async sendRobotInteractiveCardWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.callbackUrl)) {
            body["callbackUrl"] = request.callbackUrl;
        }
        if (!tea_util_1.default.isUnset(request.cardBizId)) {
            body["cardBizId"] = request.cardBizId;
        }
        if (!tea_util_1.default.isUnset(request.cardData)) {
            body["cardData"] = request.cardData;
        }
        if (!tea_util_1.default.isUnset(request.cardTemplateId)) {
            body["cardTemplateId"] = request.cardTemplateId;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.pullStrategy)) {
            body["pullStrategy"] = request.pullStrategy;
        }
        if (!tea_util_1.default.isUnset(request.robotCode)) {
            body["robotCode"] = request.robotCode;
        }
        if (!tea_util_1.default.isUnset($tea.toMap(request.sendOptions))) {
            body["sendOptions"] = request.sendOptions;
        }
        if (!tea_util_1.default.isUnset(request.singleChatReceiver)) {
            body["singleChatReceiver"] = request.singleChatReceiver;
        }
        if (!tea_util_1.default.isUnset(request.unionIdPrivateDataMap)) {
            body["unionIdPrivateDataMap"] = request.unionIdPrivateDataMap;
        }
        if (!tea_util_1.default.isUnset(request.userIdPrivateDataMap)) {
            body["userIdPrivateDataMap"] = request.userIdPrivateDataMap;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("SendRobotInteractiveCard", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/v1.0/robot/interactiveCards/send`, "json", req, runtime), new SendRobotInteractiveCardResponse({}));
    }
    async sendRobotMessage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendRobotMessageHeaders({});
        return await this.sendRobotMessageWithOptions(request, headers, runtime);
    }
    async sendRobotMessageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.atAll)) {
            body["atAll"] = request.atAll;
        }
        if (!tea_util_1.default.isUnset(request.atAppUserId)) {
            body["atAppUserId"] = request.atAppUserId;
        }
        if (!tea_util_1.default.isUnset(request.atDingUserId)) {
            body["atDingUserId"] = request.atDingUserId;
        }
        if (!tea_util_1.default.isUnset(request.msgContent)) {
            body["msgContent"] = request.msgContent;
        }
        if (!tea_util_1.default.isUnset(request.msgType)) {
            body["msgType"] = request.msgType;
        }
        if (!tea_util_1.default.isUnset(request.openConversationIds)) {
            body["openConversationIds"] = request.openConversationIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("SendRobotMessage", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interconnections/robotMessages/send`, "json", req, runtime), new SendRobotMessageResponse({}));
    }
    async sendTemplateInteractiveCard(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendTemplateInteractiveCardHeaders({});
        return await this.sendTemplateInteractiveCardWithOptions(request, headers, runtime);
    }
    async sendTemplateInteractiveCardWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.callbackUrl)) {
            body["callbackUrl"] = request.callbackUrl;
        }
        if (!tea_util_1.default.isUnset(request.cardData)) {
            body["cardData"] = request.cardData;
        }
        if (!tea_util_1.default.isUnset(request.cardTemplateId)) {
            body["cardTemplateId"] = request.cardTemplateId;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.outTrackId)) {
            body["outTrackId"] = request.outTrackId;
        }
        if (!tea_util_1.default.isUnset(request.robotCode)) {
            body["robotCode"] = request.robotCode;
        }
        if (!tea_util_1.default.isUnset($tea.toMap(request.sendOptions))) {
            body["sendOptions"] = request.sendOptions;
        }
        if (!tea_util_1.default.isUnset(request.singleChatReceiver)) {
            body["singleChatReceiver"] = request.singleChatReceiver;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("SendTemplateInteractiveCard", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interactiveCards/templates/send`, "json", req, runtime), new SendTemplateInteractiveCardResponse({}));
    }
    async topboxClose(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new TopboxCloseHeaders({});
        return await this.topboxCloseWithOptions(request, headers, runtime);
    }
    async topboxCloseWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.conversationType)) {
            body["conversationType"] = request.conversationType;
        }
        if (!tea_util_1.default.isUnset(request.coolAppCode)) {
            body["coolAppCode"] = request.coolAppCode;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.outTrackId)) {
            body["outTrackId"] = request.outTrackId;
        }
        if (!tea_util_1.default.isUnset(request.receiverUserIdList)) {
            body["receiverUserIdList"] = request.receiverUserIdList;
        }
        if (!tea_util_1.default.isUnset(request.robotCode)) {
            body["robotCode"] = request.robotCode;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("TopboxClose", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/topBoxes/close`, "none", req, runtime), new TopboxCloseResponse({}));
    }
    async topboxOpen(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new TopboxOpenHeaders({});
        return await this.topboxOpenWithOptions(request, headers, runtime);
    }
    async topboxOpenWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.conversationType)) {
            body["conversationType"] = request.conversationType;
        }
        if (!tea_util_1.default.isUnset(request.coolAppCode)) {
            body["coolAppCode"] = request.coolAppCode;
        }
        if (!tea_util_1.default.isUnset(request.expiredTime)) {
            body["expiredTime"] = request.expiredTime;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.outTrackId)) {
            body["outTrackId"] = request.outTrackId;
        }
        if (!tea_util_1.default.isUnset(request.platforms)) {
            body["platforms"] = request.platforms;
        }
        if (!tea_util_1.default.isUnset(request.receiverUserIdList)) {
            body["receiverUserIdList"] = request.receiverUserIdList;
        }
        if (!tea_util_1.default.isUnset(request.robotCode)) {
            body["robotCode"] = request.robotCode;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("TopboxOpen", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/topBoxes/open`, "none", req, runtime), new TopboxOpenResponse({}));
    }
    async updateGroupAvatar(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateGroupAvatarHeaders({});
        return await this.updateGroupAvatarWithOptions(request, headers, runtime);
    }
    async updateGroupAvatarWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.groupAvatar)) {
            body["groupAvatar"] = request.groupAvatar;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateGroupAvatar", "im_1.0", "HTTP", "PUT", "AK", `/v1.0/im/interconnections/groups/avatars`, "json", req, runtime), new UpdateGroupAvatarResponse({}));
    }
    async updateGroupName(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateGroupNameHeaders({});
        return await this.updateGroupNameWithOptions(request, headers, runtime);
    }
    async updateGroupNameWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.groupName)) {
            body["groupName"] = request.groupName;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateGroupName", "im_1.0", "HTTP", "PUT", "AK", `/v1.0/im/interconnections/groups/names`, "json", req, runtime), new UpdateGroupNameResponse({}));
    }
    async updateGroupPermission(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateGroupPermissionHeaders({});
        return await this.updateGroupPermissionWithOptions(request, headers, runtime);
    }
    async updateGroupPermissionWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.permissionGroup)) {
            body["permissionGroup"] = request.permissionGroup;
        }
        if (!tea_util_1.default.isUnset(request.status)) {
            body["status"] = request.status;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateGroupPermission", "im_1.0", "HTTP", "PUT", "AK", `/v1.0/im/sceneGroups/permissions`, "json", req, runtime), new UpdateGroupPermissionResponse({}));
    }
    async updateGroupSubAdmin(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateGroupSubAdminHeaders({});
        return await this.updateGroupSubAdminWithOptions(request, headers, runtime);
    }
    async updateGroupSubAdminWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.role)) {
            body["role"] = request.role;
        }
        if (!tea_util_1.default.isUnset(request.userIds)) {
            body["userIds"] = request.userIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateGroupSubAdmin", "im_1.0", "HTTP", "PUT", "AK", `/v1.0/im/sceneGroups/subAdmins`, "json", req, runtime), new UpdateGroupSubAdminResponse({}));
    }
    async updateInteractiveCard(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateInteractiveCardHeaders({});
        return await this.updateInteractiveCardWithOptions(request, headers, runtime);
    }
    async updateInteractiveCardWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset($tea.toMap(request.cardData))) {
            body["cardData"] = request.cardData;
        }
        if (!tea_util_1.default.isUnset($tea.toMap(request.cardOptions))) {
            body["cardOptions"] = request.cardOptions;
        }
        if (!tea_util_1.default.isUnset(request.outTrackId)) {
            body["outTrackId"] = request.outTrackId;
        }
        if (!tea_util_1.default.isUnset(request.privateData)) {
            body["privateData"] = request.privateData;
        }
        if (!tea_util_1.default.isUnset(request.userIdType)) {
            body["userIdType"] = request.userIdType;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateInteractiveCard", "im_1.0", "HTTP", "PUT", "AK", `/v1.0/im/interactiveCards`, "json", req, runtime), new UpdateInteractiveCardResponse({}));
    }
    async updateMemberBanWords(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateMemberBanWordsHeaders({});
        return await this.updateMemberBanWordsWithOptions(request, headers, runtime);
    }
    async updateMemberBanWordsWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.muteDuration)) {
            body["muteDuration"] = request.muteDuration;
        }
        if (!tea_util_1.default.isUnset(request.muteStatus)) {
            body["muteStatus"] = request.muteStatus;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.userIdList)) {
            body["userIdList"] = request.userIdList;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateMemberBanWords", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/sceneGroups/muteMembers/set`, "none", req, runtime), new UpdateMemberBanWordsResponse({}));
    }
    async updateMemberGroupNick(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateMemberGroupNickHeaders({});
        return await this.updateMemberGroupNickWithOptions(request, headers, runtime);
    }
    async updateMemberGroupNickWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.groupNick)) {
            body["groupNick"] = request.groupNick;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateMemberGroupNick", "im_1.0", "HTTP", "PUT", "AK", `/v1.0/im/sceneGroups/members/groupNicks`, "json", req, runtime), new UpdateMemberGroupNickResponse({}));
    }
    async updateRobotInteractiveCard(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateRobotInteractiveCardHeaders({});
        return await this.updateRobotInteractiveCardWithOptions(request, headers, runtime);
    }
    async updateRobotInteractiveCardWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.cardBizId)) {
            body["cardBizId"] = request.cardBizId;
        }
        if (!tea_util_1.default.isUnset(request.cardData)) {
            body["cardData"] = request.cardData;
        }
        if (!tea_util_1.default.isUnset(request.unionIdPrivateDataMap)) {
            body["unionIdPrivateDataMap"] = request.unionIdPrivateDataMap;
        }
        if (!tea_util_1.default.isUnset($tea.toMap(request.updateOptions))) {
            body["updateOptions"] = request.updateOptions;
        }
        if (!tea_util_1.default.isUnset(request.userIdPrivateDataMap)) {
            body["userIdPrivateDataMap"] = request.userIdPrivateDataMap;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateRobotInteractiveCard", "im_1.0", "HTTP", "PUT", "AK", `/v1.0/im/robots/interactiveCards`, "json", req, runtime), new UpdateRobotInteractiveCardResponse({}));
    }
    async updateTheGroupRolesOfGroupMember(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateTheGroupRolesOfGroupMemberHeaders({});
        return await this.updateTheGroupRolesOfGroupMemberWithOptions(request, headers, runtime);
    }
    async updateTheGroupRolesOfGroupMemberWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.openRoleIds)) {
            body["openRoleIds"] = request.openRoleIds;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateTheGroupRolesOfGroupMember", "im_1.0", "HTTP", "PUT", "AK", `/v1.0/im/sceneGroups/members/groupRoles`, "json", req, runtime), new UpdateTheGroupRolesOfGroupMemberResponse({}));
    }
    async addGroupMember(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new AddGroupMemberHeaders({});
        return await this.addGroupMemberWithOptions(request, headers, runtime);
    }
    async addGroupMemberWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.appUserIds)) {
            body["appUserIds"] = request.appUserIds;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.operatorId)) {
            body["operatorId"] = request.operatorId;
        }
        if (!tea_util_1.default.isUnset(request.userIds)) {
            body["userIds"] = request.userIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("addGroupMember", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interconnections/groups/members`, "json", req, runtime), new AddGroupMemberResponse({}));
    }
    async removeGroupMember(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new RemoveGroupMemberHeaders({});
        return await this.removeGroupMemberWithOptions(request, headers, runtime);
    }
    async removeGroupMemberWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.appUserIds)) {
            body["appUserIds"] = request.appUserIds;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.operatorId)) {
            body["operatorId"] = request.operatorId;
        }
        if (!tea_util_1.default.isUnset(request.userIds)) {
            body["userIds"] = request.userIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("removeGroupMember", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interconnections/groups/members/remove`, "none", req, runtime), new RemoveGroupMemberResponse({}));
    }
    async sendDingMessage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendDingMessageHeaders({});
        return await this.sendDingMessageWithOptions(request, headers, runtime);
    }
    async sendDingMessageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.code)) {
            body["code"] = request.code;
        }
        if (!tea_util_1.default.isUnset(request.message)) {
            body["message"] = request.message;
        }
        if (!tea_util_1.default.isUnset(request.messageType)) {
            body["messageType"] = request.messageType;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.receiverId)) {
            body["receiverId"] = request.receiverId;
        }
        if (!tea_util_1.default.isUnset(request.senderId)) {
            body["senderId"] = request.senderId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("sendDingMessage", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interconnections/dingMessages/send`, "json", req, runtime), new SendDingMessageResponse({}));
    }
    async sendMessage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SendMessageHeaders({});
        return await this.sendMessageWithOptions(request, headers, runtime);
    }
    async sendMessageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.message)) {
            body["message"] = request.message;
        }
        if (!tea_util_1.default.isUnset(request.messageType)) {
            body["messageType"] = request.messageType;
        }
        if (!tea_util_1.default.isUnset(request.openConversationId)) {
            body["openConversationId"] = request.openConversationId;
        }
        if (!tea_util_1.default.isUnset(request.receiverId)) {
            body["receiverId"] = request.receiverId;
        }
        if (!tea_util_1.default.isUnset(request.senderId)) {
            body["senderId"] = request.senderId;
        }
        if (!tea_util_1.default.isUnset(request.sourceInfos)) {
            body["sourceInfos"] = request.sourceInfos;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("sendMessage", "im_1.0", "HTTP", "POST", "AK", `/v1.0/im/interconnections/messages/send`, "json", req, runtime), new SendMessageResponse({}));
    }
}
exports.default = Client;
