// @alicloud/dingtalk/dist/rcs_call_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class RunCallUserHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.RunCallUserHeaders = RunCallUserHeaders;
class RunCallUserRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            authorizeCorpId: 'authorizeCorpId',
            authorizeUserId: 'authorizeUserId',
            orderId: 'orderId',
            userId: 'userId',
        };
    }
    static types() {
        return {
            authorizeCorpId: 'string',
            authorizeUserId: 'string',
            orderId: 'string',
            userId: 'string',
        };
    }
}
exports.RunCallUserRequest = RunCallUserRequest;
class RunCallUserResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'string',
        };
    }
}
exports.RunCallUserResponseBody = RunCallUserResponseBody;
class RunCallUserResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: RunCallUserResponseBody,
        };
    }
}
exports.RunCallUserResponse = RunCallUserResponse;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async runCallUser(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new RunCallUserHeaders({});
        return await this.runCallUserWithOptions(request, headers, runtime);
    }
    async runCallUserWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.authorizeCorpId)) {
            query["authorizeCorpId"] = request.authorizeCorpId;
        }
        if (!tea_util_1.default.isUnset(request.authorizeUserId)) {
            query["authorizeUserId"] = request.authorizeUserId;
        }
        if (!tea_util_1.default.isUnset(request.orderId)) {
            query["orderId"] = request.orderId;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("RunCallUser", "rcsCall_1.0", "HTTP", "POST", "AK", `/v1.0/rcsCall/users/call`, "json", req, runtime), new RunCallUserResponse({}));
    }
}
exports.default = Client;
