// @alicloud/dingtalk/dist/contact_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class BatchApproveUnionApplyHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.BatchApproveUnionApplyHeaders = BatchApproveUnionApplyHeaders;
class BatchApproveUnionApplyRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            body: 'body',
        };
    }
    static types() {
        return {
            body: { 'type': 'array', 'itemType': BatchApproveUnionApplyRequestBody },
        };
    }
}
exports.BatchApproveUnionApplyRequest = BatchApproveUnionApplyRequest;
class BatchApproveUnionApplyResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
}
exports.BatchApproveUnionApplyResponseBody = BatchApproveUnionApplyResponseBody;
class BatchApproveUnionApplyResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: BatchApproveUnionApplyResponseBody,
        };
    }
}
exports.BatchApproveUnionApplyResponse = BatchApproveUnionApplyResponse;
class ChangeMainAdminHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ChangeMainAdminHeaders = ChangeMainAdminHeaders;
class ChangeMainAdminRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            effectCorpId: 'effectCorpId',
            sourceUserId: 'sourceUserId',
            targetUserId: 'targetUserId',
        };
    }
    static types() {
        return {
            effectCorpId: 'string',
            sourceUserId: 'string',
            targetUserId: 'string',
        };
    }
}
exports.ChangeMainAdminRequest = ChangeMainAdminRequest;
class ChangeMainAdminResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.ChangeMainAdminResponse = ChangeMainAdminResponse;
class CreateCooperateOrgHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CreateCooperateOrgHeaders = CreateCooperateOrgHeaders;
class CreateCooperateOrgRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            industryCode: 'industryCode',
            logoMediaId: 'logoMediaId',
            orgName: 'orgName',
        };
    }
    static types() {
        return {
            industryCode: 'number',
            logoMediaId: 'string',
            orgName: 'string',
        };
    }
}
exports.CreateCooperateOrgRequest = CreateCooperateOrgRequest;
class CreateCooperateOrgResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            cooperateCorpId: 'cooperateCorpId',
        };
    }
    static types() {
        return {
            cooperateCorpId: 'string',
        };
    }
}
exports.CreateCooperateOrgResponseBody = CreateCooperateOrgResponseBody;
class CreateCooperateOrgResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: CreateCooperateOrgResponseBody,
        };
    }
}
exports.CreateCooperateOrgResponse = CreateCooperateOrgResponse;
class CreateManagementGroupHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CreateManagementGroupHeaders = CreateManagementGroupHeaders;
class CreateManagementGroupRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupName: 'groupName',
            members: 'members',
            resourceIds: 'resourceIds',
            scope: 'scope',
        };
    }
    static types() {
        return {
            groupName: 'string',
            members: { 'type': 'array', 'itemType': CreateManagementGroupRequestMembers },
            resourceIds: { 'type': 'array', 'itemType': 'string' },
            scope: CreateManagementGroupRequestScope,
        };
    }
}
exports.CreateManagementGroupRequest = CreateManagementGroupRequest;
class CreateManagementGroupResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupId: 'groupId',
        };
    }
    static types() {
        return {
            groupId: 'string',
        };
    }
}
exports.CreateManagementGroupResponseBody = CreateManagementGroupResponseBody;
class CreateManagementGroupResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: CreateManagementGroupResponseBody,
        };
    }
}
exports.CreateManagementGroupResponse = CreateManagementGroupResponse;
class DeleteContactHideSettingHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.DeleteContactHideSettingHeaders = DeleteContactHideSettingHeaders;
class DeleteContactHideSettingResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.DeleteContactHideSettingResponse = DeleteContactHideSettingResponse;
class DeleteContactRestrictSettingHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.DeleteContactRestrictSettingHeaders = DeleteContactRestrictSettingHeaders;
class DeleteContactRestrictSettingResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
}
exports.DeleteContactRestrictSettingResponseBody = DeleteContactRestrictSettingResponseBody;
class DeleteContactRestrictSettingResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: DeleteContactRestrictSettingResponseBody,
        };
    }
}
exports.DeleteContactRestrictSettingResponse = DeleteContactRestrictSettingResponse;
class DeleteEmpAttributeVisibilityHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.DeleteEmpAttributeVisibilityHeaders = DeleteEmpAttributeVisibilityHeaders;
class DeleteEmpAttributeVisibilityResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.DeleteEmpAttributeVisibilityResponse = DeleteEmpAttributeVisibilityResponse;
class DeleteManagementGroupHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.DeleteManagementGroupHeaders = DeleteManagementGroupHeaders;
class DeleteManagementGroupResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.DeleteManagementGroupResponse = DeleteManagementGroupResponse;
class GetApplyInviteInfoHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetApplyInviteInfoHeaders = GetApplyInviteInfoHeaders;
class GetApplyInviteInfoRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deptId: 'deptId',
            inviterUserId: 'inviterUserId',
        };
    }
    static types() {
        return {
            deptId: 'number',
            inviterUserId: 'string',
        };
    }
}
exports.GetApplyInviteInfoRequest = GetApplyInviteInfoRequest;
class GetApplyInviteInfoResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            auditType: 'auditType',
            empApplyJoinDept: 'empApplyJoinDept',
            inviteSwitch: 'inviteSwitch',
            inviteUrl: 'inviteUrl',
            linkInvite: 'linkInvite',
            orgApplyCodeInvite: 'orgApplyCodeInvite',
            searchNameInvite: 'searchNameInvite',
        };
    }
    static types() {
        return {
            auditType: 'number',
            empApplyJoinDept: 'boolean',
            inviteSwitch: 'boolean',
            inviteUrl: 'string',
            linkInvite: 'boolean',
            orgApplyCodeInvite: 'boolean',
            searchNameInvite: 'boolean',
        };
    }
}
exports.GetApplyInviteInfoResponseBody = GetApplyInviteInfoResponseBody;
class GetApplyInviteInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetApplyInviteInfoResponseBody,
        };
    }
}
exports.GetApplyInviteInfoResponse = GetApplyInviteInfoResponse;
class GetBranchAuthDataHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetBranchAuthDataHeaders = GetBranchAuthDataHeaders;
class GetBranchAuthDataRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            body: 'body',
            branchCorpId: 'branchCorpId',
            code: 'code',
        };
    }
    static types() {
        return {
            body: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            branchCorpId: 'string',
            code: 'string',
        };
    }
}
exports.GetBranchAuthDataRequest = GetBranchAuthDataRequest;
class GetBranchAuthDataResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': GetBranchAuthDataResponseBodyResult },
        };
    }
}
exports.GetBranchAuthDataResponseBody = GetBranchAuthDataResponseBody;
class GetBranchAuthDataResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetBranchAuthDataResponseBody,
        };
    }
}
exports.GetBranchAuthDataResponse = GetBranchAuthDataResponse;
class GetCardInUserHolderHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetCardInUserHolderHeaders = GetCardInUserHolderHeaders;
class GetCardInUserHolderResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            avatarUrl: 'avatarUrl',
            cardId: 'cardId',
            extension: 'extension',
            industryName: 'industryName',
            introduce: 'introduce',
            name: 'name',
            orgName: 'orgName',
            templateId: 'templateId',
            title: 'title',
        };
    }
    static types() {
        return {
            avatarUrl: 'string',
            cardId: 'string',
            extension: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            industryName: 'string',
            introduce: 'string',
            name: 'string',
            orgName: 'string',
            templateId: 'string',
            title: 'string',
        };
    }
}
exports.GetCardInUserHolderResponseBody = GetCardInUserHolderResponseBody;
class GetCardInUserHolderResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetCardInUserHolderResponseBody,
        };
    }
}
exports.GetCardInUserHolderResponse = GetCardInUserHolderResponse;
class GetCardInfoHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetCardInfoHeaders = GetCardInfoHeaders;
class GetCardInfoResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            adminRole: 'adminRole',
            avatarUrl: 'avatarUrl',
            cardId: 'cardId',
            extension: 'extension',
            industryName: 'industryName',
            introduce: 'introduce',
            name: 'name',
            orgName: 'orgName',
            settings: 'settings',
            templateId: 'templateId',
            title: 'title',
        };
    }
    static types() {
        return {
            adminRole: 'number',
            avatarUrl: 'string',
            cardId: 'string',
            extension: GetCardInfoResponseBodyExtension,
            industryName: 'string',
            introduce: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            name: 'string',
            orgName: 'string',
            settings: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            templateId: 'string',
            title: 'string',
        };
    }
}
exports.GetCardInfoResponseBody = GetCardInfoResponseBody;
class GetCardInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetCardInfoResponseBody,
        };
    }
}
exports.GetCardInfoResponse = GetCardInfoResponse;
class GetCooperateOrgInviteInfoHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetCooperateOrgInviteInfoHeaders = GetCooperateOrgInviteInfoHeaders;
class GetCooperateOrgInviteInfoResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            inviteUrl: 'inviteUrl',
        };
    }
    static types() {
        return {
            inviteUrl: 'string',
        };
    }
}
exports.GetCooperateOrgInviteInfoResponseBody = GetCooperateOrgInviteInfoResponseBody;
class GetCooperateOrgInviteInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetCooperateOrgInviteInfoResponseBody,
        };
    }
}
exports.GetCooperateOrgInviteInfoResponse = GetCooperateOrgInviteInfoResponse;
class GetCorpCardStyleListHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetCorpCardStyleListHeaders = GetCorpCardStyleListHeaders;
class GetCorpCardStyleListResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: { 'type': 'array', 'itemType': { 'type': 'map', 'keyType': 'string', 'valueType': 'any' } },
        };
    }
}
exports.GetCorpCardStyleListResponseBody = GetCorpCardStyleListResponseBody;
class GetCorpCardStyleListResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetCorpCardStyleListResponseBody,
        };
    }
}
exports.GetCorpCardStyleListResponse = GetCorpCardStyleListResponse;
class GetDingIdByMigrationDingIdHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetDingIdByMigrationDingIdHeaders = GetDingIdByMigrationDingIdHeaders;
class GetDingIdByMigrationDingIdRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            migrationDingId: 'migrationDingId',
        };
    }
    static types() {
        return {
            migrationDingId: 'string',
        };
    }
}
exports.GetDingIdByMigrationDingIdRequest = GetDingIdByMigrationDingIdRequest;
class GetDingIdByMigrationDingIdResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            dingId: 'dingId',
        };
    }
    static types() {
        return {
            dingId: 'string',
        };
    }
}
exports.GetDingIdByMigrationDingIdResponseBody = GetDingIdByMigrationDingIdResponseBody;
class GetDingIdByMigrationDingIdResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetDingIdByMigrationDingIdResponseBody,
        };
    }
}
exports.GetDingIdByMigrationDingIdResponse = GetDingIdByMigrationDingIdResponse;
class GetLatestDingIndexHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetLatestDingIndexHeaders = GetLatestDingIndexHeaders;
class GetLatestDingIndexResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            idxCarbon: 'idxCarbon',
            idxEfficiency: 'idxEfficiency',
            idxMonthlyAvg: 'idxMonthlyAvg',
            idxTotal: 'idxTotal',
            statDate: 'statDate',
        };
    }
    static types() {
        return {
            idxCarbon: 'number',
            idxEfficiency: 'number',
            idxMonthlyAvg: 'number',
            idxTotal: 'number',
            statDate: 'string',
        };
    }
}
exports.GetLatestDingIndexResponseBody = GetLatestDingIndexResponseBody;
class GetLatestDingIndexResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetLatestDingIndexResponseBody,
        };
    }
}
exports.GetLatestDingIndexResponse = GetLatestDingIndexResponse;
class GetMigrationDingIdByDingIdHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetMigrationDingIdByDingIdHeaders = GetMigrationDingIdByDingIdHeaders;
class GetMigrationDingIdByDingIdRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            dingId: 'dingId',
        };
    }
    static types() {
        return {
            dingId: 'string',
        };
    }
}
exports.GetMigrationDingIdByDingIdRequest = GetMigrationDingIdByDingIdRequest;
class GetMigrationDingIdByDingIdResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            migrationDingIdList: 'migrationDingIdList',
        };
    }
    static types() {
        return {
            migrationDingIdList: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
        };
    }
}
exports.GetMigrationDingIdByDingIdResponseBody = GetMigrationDingIdByDingIdResponseBody;
class GetMigrationDingIdByDingIdResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetMigrationDingIdByDingIdResponseBody,
        };
    }
}
exports.GetMigrationDingIdByDingIdResponse = GetMigrationDingIdByDingIdResponse;
class GetMigrationUnionIdByUnionIdHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetMigrationUnionIdByUnionIdHeaders = GetMigrationUnionIdByUnionIdHeaders;
class GetMigrationUnionIdByUnionIdRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            unionId: 'string',
        };
    }
}
exports.GetMigrationUnionIdByUnionIdRequest = GetMigrationUnionIdByUnionIdRequest;
class GetMigrationUnionIdByUnionIdResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            migrationUnionIdList: 'migrationUnionIdList',
        };
    }
    static types() {
        return {
            migrationUnionIdList: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
        };
    }
}
exports.GetMigrationUnionIdByUnionIdResponseBody = GetMigrationUnionIdByUnionIdResponseBody;
class GetMigrationUnionIdByUnionIdResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetMigrationUnionIdByUnionIdResponseBody,
        };
    }
}
exports.GetMigrationUnionIdByUnionIdResponse = GetMigrationUnionIdByUnionIdResponse;
class GetOrgAuthInfoHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetOrgAuthInfoHeaders = GetOrgAuthInfoHeaders;
class GetOrgAuthInfoRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            targetCorpId: 'targetCorpId',
        };
    }
    static types() {
        return {
            targetCorpId: 'string',
        };
    }
}
exports.GetOrgAuthInfoRequest = GetOrgAuthInfoRequest;
class GetOrgAuthInfoResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            authLevel: 'authLevel',
            legalPerson: 'legalPerson',
            licenseOrgName: 'licenseOrgName',
            licenseUrl: 'licenseUrl',
            orgName: 'orgName',
            organizationCode: 'organizationCode',
            registrationNum: 'registrationNum',
            unifiedSocialCredit: 'unifiedSocialCredit',
        };
    }
    static types() {
        return {
            authLevel: 'number',
            legalPerson: 'string',
            licenseOrgName: 'string',
            licenseUrl: 'string',
            orgName: 'string',
            organizationCode: 'string',
            registrationNum: 'string',
            unifiedSocialCredit: 'string',
        };
    }
}
exports.GetOrgAuthInfoResponseBody = GetOrgAuthInfoResponseBody;
class GetOrgAuthInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetOrgAuthInfoResponseBody,
        };
    }
}
exports.GetOrgAuthInfoResponse = GetOrgAuthInfoResponse;
class GetTranslateFileJobResultHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetTranslateFileJobResultHeaders = GetTranslateFileJobResultHeaders;
class GetTranslateFileJobResultRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            jobId: 'jobId',
        };
    }
    static types() {
        return {
            jobId: 'string',
        };
    }
}
exports.GetTranslateFileJobResultRequest = GetTranslateFileJobResultRequest;
class GetTranslateFileJobResultResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            status: 'status',
            url: 'url',
        };
    }
    static types() {
        return {
            status: 'string',
            url: 'string',
        };
    }
}
exports.GetTranslateFileJobResultResponseBody = GetTranslateFileJobResultResponseBody;
class GetTranslateFileJobResultResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetTranslateFileJobResultResponseBody,
        };
    }
}
exports.GetTranslateFileJobResultResponse = GetTranslateFileJobResultResponse;
class GetUnionIdByMigrationUnionIdHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetUnionIdByMigrationUnionIdHeaders = GetUnionIdByMigrationUnionIdHeaders;
class GetUnionIdByMigrationUnionIdRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            migrationUnionId: 'migrationUnionId',
        };
    }
    static types() {
        return {
            migrationUnionId: 'string',
        };
    }
}
exports.GetUnionIdByMigrationUnionIdRequest = GetUnionIdByMigrationUnionIdRequest;
class GetUnionIdByMigrationUnionIdResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            unionId: 'string',
        };
    }
}
exports.GetUnionIdByMigrationUnionIdResponseBody = GetUnionIdByMigrationUnionIdResponseBody;
class GetUnionIdByMigrationUnionIdResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetUnionIdByMigrationUnionIdResponseBody,
        };
    }
}
exports.GetUnionIdByMigrationUnionIdResponse = GetUnionIdByMigrationUnionIdResponse;
class GetUserHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetUserHeaders = GetUserHeaders;
class GetUserResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            avatarUrl: 'avatarUrl',
            email: 'email',
            mobile: 'mobile',
            nick: 'nick',
            openId: 'openId',
            stateCode: 'stateCode',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            avatarUrl: 'string',
            email: 'string',
            mobile: 'string',
            nick: 'string',
            openId: 'string',
            stateCode: 'string',
            unionId: 'string',
        };
    }
}
exports.GetUserResponseBody = GetUserResponseBody;
class GetUserResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetUserResponseBody,
        };
    }
}
exports.GetUserResponse = GetUserResponse;
class GetUserCardHolderListHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetUserCardHolderListHeaders = GetUserCardHolderListHeaders;
class GetUserCardHolderListRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            maxResults: 'maxResults',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            maxResults: 'number',
            nextToken: 'number',
        };
    }
}
exports.GetUserCardHolderListRequest = GetUserCardHolderListRequest;
class GetUserCardHolderListResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            hasMore: 'hasMore',
            list: 'list',
            nextToken: 'nextToken',
            totalCount: 'totalCount',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            list: { 'type': 'array', 'itemType': GetUserCardHolderListResponseBodyList },
            nextToken: 'number',
            totalCount: 'number',
        };
    }
}
exports.GetUserCardHolderListResponseBody = GetUserCardHolderListResponseBody;
class GetUserCardHolderListResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetUserCardHolderListResponseBody,
        };
    }
}
exports.GetUserCardHolderListResponse = GetUserCardHolderListResponse;
class IsvCardEventPushHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.IsvCardEventPushHeaders = IsvCardEventPushHeaders;
class IsvCardEventPushRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            eventParams: 'eventParams',
            eventType: 'eventType',
            isvCardId: 'isvCardId',
            isvToken: 'isvToken',
            isvUid: 'isvUid',
        };
    }
    static types() {
        return {
            eventParams: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            eventType: 'string',
            isvCardId: 'string',
            isvToken: 'string',
            isvUid: 'string',
        };
    }
}
exports.IsvCardEventPushRequest = IsvCardEventPushRequest;
class IsvCardEventPushResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            success: 'success',
        };
    }
    static types() {
        return {
            success: 'boolean',
        };
    }
}
exports.IsvCardEventPushResponseBody = IsvCardEventPushResponseBody;
class IsvCardEventPushResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: IsvCardEventPushResponseBody,
        };
    }
}
exports.IsvCardEventPushResponse = IsvCardEventPushResponse;
class ListBasicRoleInPageHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ListBasicRoleInPageHeaders = ListBasicRoleInPageHeaders;
class ListBasicRoleInPageRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            agentId: 'agentId',
            maxResults: 'maxResults',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            agentId: 'string',
            maxResults: 'number',
            nextToken: 'number',
        };
    }
}
exports.ListBasicRoleInPageRequest = ListBasicRoleInPageRequest;
class ListBasicRoleInPageResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            hasMore: 'hasMore',
            list: 'list',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            list: { 'type': 'array', 'itemType': ListBasicRoleInPageResponseBodyList },
            nextToken: 'number',
        };
    }
}
exports.ListBasicRoleInPageResponseBody = ListBasicRoleInPageResponseBody;
class ListBasicRoleInPageResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: ListBasicRoleInPageResponseBody,
        };
    }
}
exports.ListBasicRoleInPageResponse = ListBasicRoleInPageResponse;
class ListContactHideSettingsHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ListContactHideSettingsHeaders = ListContactHideSettingsHeaders;
class ListContactHideSettingsRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            maxResults: 'maxResults',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            maxResults: 'number',
            nextToken: 'number',
        };
    }
}
exports.ListContactHideSettingsRequest = ListContactHideSettingsRequest;
class ListContactHideSettingsResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            hasMore: 'hasMore',
            list: 'list',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            list: { 'type': 'array', 'itemType': ListContactHideSettingsResponseBodyList },
            nextToken: 'number',
        };
    }
}
exports.ListContactHideSettingsResponseBody = ListContactHideSettingsResponseBody;
class ListContactHideSettingsResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: ListContactHideSettingsResponseBody,
        };
    }
}
exports.ListContactHideSettingsResponse = ListContactHideSettingsResponse;
class ListContactRestrictSettingHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ListContactRestrictSettingHeaders = ListContactRestrictSettingHeaders;
class ListContactRestrictSettingRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            maxResults: 'maxResults',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            maxResults: 'number',
            nextToken: 'number',
        };
    }
}
exports.ListContactRestrictSettingRequest = ListContactRestrictSettingRequest;
class ListContactRestrictSettingResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            hasMore: 'hasMore',
            list: 'list',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            list: { 'type': 'array', 'itemType': ListContactRestrictSettingResponseBodyList },
            nextToken: 'number',
        };
    }
}
exports.ListContactRestrictSettingResponseBody = ListContactRestrictSettingResponseBody;
class ListContactRestrictSettingResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: ListContactRestrictSettingResponseBody,
        };
    }
}
exports.ListContactRestrictSettingResponse = ListContactRestrictSettingResponse;
class ListEmpAttributeVisibilityHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ListEmpAttributeVisibilityHeaders = ListEmpAttributeVisibilityHeaders;
class ListEmpAttributeVisibilityRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            maxResults: 'maxResults',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            maxResults: 'number',
            nextToken: 'number',
        };
    }
}
exports.ListEmpAttributeVisibilityRequest = ListEmpAttributeVisibilityRequest;
class ListEmpAttributeVisibilityResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            hasMore: 'hasMore',
            list: 'list',
            nextCursor: 'nextCursor',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            list: { 'type': 'array', 'itemType': ListEmpAttributeVisibilityResponseBodyList },
            nextCursor: 'number',
        };
    }
}
exports.ListEmpAttributeVisibilityResponseBody = ListEmpAttributeVisibilityResponseBody;
class ListEmpAttributeVisibilityResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: ListEmpAttributeVisibilityResponseBody,
        };
    }
}
exports.ListEmpAttributeVisibilityResponse = ListEmpAttributeVisibilityResponse;
class ListEmpLeaveRecordsHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ListEmpLeaveRecordsHeaders = ListEmpLeaveRecordsHeaders;
class ListEmpLeaveRecordsRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            endTime: 'endTime',
            maxResults: 'maxResults',
            nextToken: 'nextToken',
            startTime: 'startTime',
        };
    }
    static types() {
        return {
            endTime: 'string',
            maxResults: 'number',
            nextToken: 'string',
            startTime: 'string',
        };
    }
}
exports.ListEmpLeaveRecordsRequest = ListEmpLeaveRecordsRequest;
class ListEmpLeaveRecordsResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            nextToken: 'nextToken',
            records: 'records',
        };
    }
    static types() {
        return {
            nextToken: 'string',
            records: { 'type': 'array', 'itemType': ListEmpLeaveRecordsResponseBodyRecords },
        };
    }
}
exports.ListEmpLeaveRecordsResponseBody = ListEmpLeaveRecordsResponseBody;
class ListEmpLeaveRecordsResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: ListEmpLeaveRecordsResponseBody,
        };
    }
}
exports.ListEmpLeaveRecordsResponse = ListEmpLeaveRecordsResponse;
class ListManagementGroupsHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ListManagementGroupsHeaders = ListManagementGroupsHeaders;
class ListManagementGroupsRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            maxResults: 'maxResults',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            maxResults: 'number',
            nextToken: 'number',
        };
    }
}
exports.ListManagementGroupsRequest = ListManagementGroupsRequest;
class ListManagementGroupsResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groups: 'groups',
            hasMore: 'hasMore',
            nextToken: 'nextToken',
        };
    }
    static types() {
        return {
            groups: { 'type': 'array', 'itemType': ListManagementGroupsResponseBodyGroups },
            hasMore: 'boolean',
            nextToken: 'number',
        };
    }
}
exports.ListManagementGroupsResponseBody = ListManagementGroupsResponseBody;
class ListManagementGroupsResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: ListManagementGroupsResponseBody,
        };
    }
}
exports.ListManagementGroupsResponse = ListManagementGroupsResponse;
class ListOwnedOrgByStaffIdHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ListOwnedOrgByStaffIdHeaders = ListOwnedOrgByStaffIdHeaders;
class ListOwnedOrgByStaffIdRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            userId: 'userId',
        };
    }
    static types() {
        return {
            userId: 'string',
        };
    }
}
exports.ListOwnedOrgByStaffIdRequest = ListOwnedOrgByStaffIdRequest;
class ListOwnedOrgByStaffIdResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            orgList: 'orgList',
        };
    }
    static types() {
        return {
            orgList: { 'type': 'array', 'itemType': ListOwnedOrgByStaffIdResponseBodyOrgList },
        };
    }
}
exports.ListOwnedOrgByStaffIdResponseBody = ListOwnedOrgByStaffIdResponseBody;
class ListOwnedOrgByStaffIdResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: ListOwnedOrgByStaffIdResponseBody,
        };
    }
}
exports.ListOwnedOrgByStaffIdResponse = ListOwnedOrgByStaffIdResponse;
class ListSeniorSettingsHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ListSeniorSettingsHeaders = ListSeniorSettingsHeaders;
class ListSeniorSettingsRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            seniorStaffId: 'seniorStaffId',
        };
    }
    static types() {
        return {
            seniorStaffId: 'string',
        };
    }
}
exports.ListSeniorSettingsRequest = ListSeniorSettingsRequest;
class ListSeniorSettingsResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            protectScenes: 'protectScenes',
            seniorStaffId: 'seniorStaffId',
            seniorWhiteList: 'seniorWhiteList',
        };
    }
    static types() {
        return {
            protectScenes: { 'type': 'array', 'itemType': 'string' },
            seniorStaffId: 'string',
            seniorWhiteList: { 'type': 'array', 'itemType': ListSeniorSettingsResponseBodySeniorWhiteList },
        };
    }
}
exports.ListSeniorSettingsResponseBody = ListSeniorSettingsResponseBody;
class ListSeniorSettingsResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: ListSeniorSettingsResponseBody,
        };
    }
}
exports.ListSeniorSettingsResponse = ListSeniorSettingsResponse;
class MultiOrgPermissionGrantHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.MultiOrgPermissionGrantHeaders = MultiOrgPermissionGrantHeaders;
class MultiOrgPermissionGrantRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            grantDeptIdList: 'grantDeptIdList',
            joinCorpId: 'joinCorpId',
        };
    }
    static types() {
        return {
            grantDeptIdList: { 'type': 'array', 'itemType': 'number' },
            joinCorpId: 'string',
        };
    }
}
exports.MultiOrgPermissionGrantRequest = MultiOrgPermissionGrantRequest;
class MultiOrgPermissionGrantResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.MultiOrgPermissionGrantResponse = MultiOrgPermissionGrantResponse;
class QueryResourceManagementMembersHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryResourceManagementMembersHeaders = QueryResourceManagementMembersHeaders;
class QueryResourceManagementMembersResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            members: 'members',
        };
    }
    static types() {
        return {
            members: { 'type': 'array', 'itemType': QueryResourceManagementMembersResponseBodyMembers },
        };
    }
}
exports.QueryResourceManagementMembersResponseBody = QueryResourceManagementMembersResponseBody;
class QueryResourceManagementMembersResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: QueryResourceManagementMembersResponseBody,
        };
    }
}
exports.QueryResourceManagementMembersResponse = QueryResourceManagementMembersResponse;
class QueryStatusHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryStatusHeaders = QueryStatusHeaders;
class QueryStatusRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            userId: 'userId',
        };
    }
    static types() {
        return {
            userId: 'string',
        };
    }
}
exports.QueryStatusRequest = QueryStatusRequest;
class QueryStatusResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            disable: 'disable',
        };
    }
    static types() {
        return {
            disable: 'boolean',
        };
    }
}
exports.QueryStatusResponseBody = QueryStatusResponseBody;
class QueryStatusResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: QueryStatusResponseBody,
        };
    }
}
exports.QueryStatusResponse = QueryStatusResponse;
class QueryUserManagementResourcesHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryUserManagementResourcesHeaders = QueryUserManagementResourcesHeaders;
class QueryUserManagementResourcesResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            resourceIds: 'resourceIds',
        };
    }
    static types() {
        return {
            resourceIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.QueryUserManagementResourcesResponseBody = QueryUserManagementResourcesResponseBody;
class QueryUserManagementResourcesResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: QueryUserManagementResourcesResponseBody,
        };
    }
}
exports.QueryUserManagementResourcesResponse = QueryUserManagementResourcesResponse;
class SearchDepartmentHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SearchDepartmentHeaders = SearchDepartmentHeaders;
class SearchDepartmentRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            offset: 'offset',
            queryWord: 'queryWord',
            size: 'size',
        };
    }
    static types() {
        return {
            offset: 'number',
            queryWord: 'string',
            size: 'number',
        };
    }
}
exports.SearchDepartmentRequest = SearchDepartmentRequest;
class SearchDepartmentResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            hasMore: 'hasMore',
            list: 'list',
            totalCount: 'totalCount',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            list: { 'type': 'array', 'itemType': 'number' },
            totalCount: 'number',
        };
    }
}
exports.SearchDepartmentResponseBody = SearchDepartmentResponseBody;
class SearchDepartmentResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: SearchDepartmentResponseBody,
        };
    }
}
exports.SearchDepartmentResponse = SearchDepartmentResponse;
class SearchUserHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SearchUserHeaders = SearchUserHeaders;
class SearchUserRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            fullMatchField: 'fullMatchField',
            offset: 'offset',
            queryWord: 'queryWord',
            size: 'size',
        };
    }
    static types() {
        return {
            fullMatchField: 'number',
            offset: 'number',
            queryWord: 'string',
            size: 'number',
        };
    }
}
exports.SearchUserRequest = SearchUserRequest;
class SearchUserResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            hasMore: 'hasMore',
            list: 'list',
            totalCount: 'totalCount',
        };
    }
    static types() {
        return {
            hasMore: 'boolean',
            list: { 'type': 'array', 'itemType': 'string' },
            totalCount: 'number',
        };
    }
}
exports.SearchUserResponseBody = SearchUserResponseBody;
class SearchUserResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: SearchUserResponseBody,
        };
    }
}
exports.SearchUserResponse = SearchUserResponse;
class SeparateBranchOrgHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SeparateBranchOrgHeaders = SeparateBranchOrgHeaders;
class SeparateBranchOrgRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            attachDeptId: 'attachDeptId',
        };
    }
    static types() {
        return {
            attachDeptId: 'number',
        };
    }
}
exports.SeparateBranchOrgRequest = SeparateBranchOrgRequest;
class SeparateBranchOrgResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'boolean',
        };
    }
}
exports.SeparateBranchOrgResponseBody = SeparateBranchOrgResponseBody;
class SeparateBranchOrgResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: SeparateBranchOrgResponseBody,
        };
    }
}
exports.SeparateBranchOrgResponse = SeparateBranchOrgResponse;
class SetDisableHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SetDisableHeaders = SetDisableHeaders;
class SetDisableRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            reason: 'reason',
            userId: 'userId',
        };
    }
    static types() {
        return {
            reason: 'string',
            userId: 'string',
        };
    }
}
exports.SetDisableRequest = SetDisableRequest;
class SetDisableResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.SetDisableResponse = SetDisableResponse;
class SetEnableHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SetEnableHeaders = SetEnableHeaders;
class SetEnableRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            userId: 'userId',
        };
    }
    static types() {
        return {
            userId: 'string',
        };
    }
}
exports.SetEnableRequest = SetEnableRequest;
class SetEnableResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.SetEnableResponse = SetEnableResponse;
class SignOutHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SignOutHeaders = SignOutHeaders;
class SignOutRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            reason: 'reason',
            userId: 'userId',
        };
    }
    static types() {
        return {
            reason: 'string',
            userId: 'string',
        };
    }
}
exports.SignOutRequest = SignOutRequest;
class SignOutResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.SignOutResponse = SignOutResponse;
class SortUserHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.SortUserHeaders = SortUserHeaders;
class SortUserRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            sortType: 'sortType',
            userIdList: 'userIdList',
        };
    }
    static types() {
        return {
            sortType: 'number',
            userIdList: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.SortUserRequest = SortUserRequest;
class SortUserResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            userIdList: 'userIdList',
        };
    }
    static types() {
        return {
            userIdList: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.SortUserResponseBody = SortUserResponseBody;
class SortUserResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: SortUserResponseBody,
        };
    }
}
exports.SortUserResponse = SortUserResponse;
class TransformToExclusiveAccountHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.TransformToExclusiveAccountHeaders = TransformToExclusiveAccountHeaders;
class TransformToExclusiveAccountRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            idpDingTalk: 'idpDingTalk',
            initPassword: 'initPassword',
            loginId: 'loginId',
            transformType: 'transformType',
            userId: 'userId',
        };
    }
    static types() {
        return {
            idpDingTalk: 'boolean',
            initPassword: 'string',
            loginId: 'string',
            transformType: 'string',
            userId: 'string',
        };
    }
}
exports.TransformToExclusiveAccountRequest = TransformToExclusiveAccountRequest;
class TransformToExclusiveAccountResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.TransformToExclusiveAccountResponse = TransformToExclusiveAccountResponse;
class TranslateFileHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.TranslateFileHeaders = TranslateFileHeaders;
class TranslateFileRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            medias: 'medias',
            outputFileName: 'outputFileName',
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            medias: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            outputFileName: 'string',
            unionId: 'string',
        };
    }
}
exports.TranslateFileRequest = TranslateFileRequest;
class TranslateFileResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            jobId: 'jobId',
        };
    }
    static types() {
        return {
            jobId: 'string',
        };
    }
}
exports.TranslateFileResponseBody = TranslateFileResponseBody;
class TranslateFileResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: TranslateFileResponseBody,
        };
    }
}
exports.TranslateFileResponse = TranslateFileResponse;
class UniqueQueryUserCardHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UniqueQueryUserCardHeaders = UniqueQueryUserCardHeaders;
class UniqueQueryUserCardRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            unionId: 'unionId',
        };
    }
    static types() {
        return {
            unionId: 'string',
        };
    }
}
exports.UniqueQueryUserCardRequest = UniqueQueryUserCardRequest;
class UniqueQueryUserCardResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            avatarUrl: 'avatarUrl',
            cardId: 'cardId',
            extension: 'extension',
            industryName: 'industryName',
            introduce: 'introduce',
            name: 'name',
            orgName: 'orgName',
            templateId: 'templateId',
            title: 'title',
        };
    }
    static types() {
        return {
            avatarUrl: 'string',
            cardId: 'string',
            extension: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            industryName: 'string',
            introduce: 'string',
            name: 'string',
            orgName: 'string',
            templateId: 'string',
            title: 'string',
        };
    }
}
exports.UniqueQueryUserCardResponseBody = UniqueQueryUserCardResponseBody;
class UniqueQueryUserCardResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: UniqueQueryUserCardResponseBody,
        };
    }
}
exports.UniqueQueryUserCardResponse = UniqueQueryUserCardResponse;
class UpdateBranchAttributesInCooperateHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateBranchAttributesInCooperateHeaders = UpdateBranchAttributesInCooperateHeaders;
class UpdateBranchAttributesInCooperateRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            body: 'body',
        };
    }
    static types() {
        return {
            body: { 'type': 'array', 'itemType': UpdateBranchAttributesInCooperateRequestBody },
        };
    }
}
exports.UpdateBranchAttributesInCooperateRequest = UpdateBranchAttributesInCooperateRequest;
class UpdateBranchAttributesInCooperateResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.UpdateBranchAttributesInCooperateResponse = UpdateBranchAttributesInCooperateResponse;
class UpdateBranchVisibleSettingInCooperateHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateBranchVisibleSettingInCooperateHeaders = UpdateBranchVisibleSettingInCooperateHeaders;
class UpdateBranchVisibleSettingInCooperateRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            body: 'body',
        };
    }
    static types() {
        return {
            body: { 'type': 'array', 'itemType': UpdateBranchVisibleSettingInCooperateRequestBody },
        };
    }
}
exports.UpdateBranchVisibleSettingInCooperateRequest = UpdateBranchVisibleSettingInCooperateRequest;
class UpdateBranchVisibleSettingInCooperateResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.UpdateBranchVisibleSettingInCooperateResponse = UpdateBranchVisibleSettingInCooperateResponse;
class UpdateContactHideSettingHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateContactHideSettingHeaders = UpdateContactHideSettingHeaders;
class UpdateContactHideSettingRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            active: 'active',
            description: 'description',
            excludeDeptIds: 'excludeDeptIds',
            excludeStaffIds: 'excludeStaffIds',
            excludeTagIds: 'excludeTagIds',
            hideInSearch: 'hideInSearch',
            hideInUserProfile: 'hideInUserProfile',
            id: 'id',
            name: 'name',
            objectDeptIds: 'objectDeptIds',
            objectStaffIds: 'objectStaffIds',
            objectTagIds: 'objectTagIds',
        };
    }
    static types() {
        return {
            active: 'boolean',
            description: 'string',
            excludeDeptIds: { 'type': 'array', 'itemType': 'number' },
            excludeStaffIds: { 'type': 'array', 'itemType': 'string' },
            excludeTagIds: { 'type': 'array', 'itemType': 'number' },
            hideInSearch: 'boolean',
            hideInUserProfile: 'boolean',
            id: 'number',
            name: 'string',
            objectDeptIds: { 'type': 'array', 'itemType': 'number' },
            objectStaffIds: { 'type': 'array', 'itemType': 'string' },
            objectTagIds: { 'type': 'array', 'itemType': 'number' },
        };
    }
}
exports.UpdateContactHideSettingRequest = UpdateContactHideSettingRequest;
class UpdateContactHideSettingResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'number',
        };
    }
}
exports.UpdateContactHideSettingResponseBody = UpdateContactHideSettingResponseBody;
class UpdateContactHideSettingResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: UpdateContactHideSettingResponseBody,
        };
    }
}
exports.UpdateContactHideSettingResponse = UpdateContactHideSettingResponse;
class UpdateContactRestrictSettingHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateContactRestrictSettingHeaders = UpdateContactRestrictSettingHeaders;
class UpdateContactRestrictSettingRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            active: 'active',
            description: 'description',
            excludeDeptIds: 'excludeDeptIds',
            excludeTagIds: 'excludeTagIds',
            excludeUserIds: 'excludeUserIds',
            id: 'id',
            name: 'name',
            restrictInSearch: 'restrictInSearch',
            restrictInUserProfile: 'restrictInUserProfile',
            subjectDeptIds: 'subjectDeptIds',
            subjectTagIds: 'subjectTagIds',
            subjectUserIds: 'subjectUserIds',
            type: 'type',
        };
    }
    static types() {
        return {
            active: 'boolean',
            description: 'string',
            excludeDeptIds: { 'type': 'array', 'itemType': 'number' },
            excludeTagIds: { 'type': 'array', 'itemType': 'number' },
            excludeUserIds: { 'type': 'array', 'itemType': 'string' },
            id: 'number',
            name: 'string',
            restrictInSearch: 'boolean',
            restrictInUserProfile: 'boolean',
            subjectDeptIds: { 'type': 'array', 'itemType': 'number' },
            subjectTagIds: { 'type': 'array', 'itemType': 'number' },
            subjectUserIds: { 'type': 'array', 'itemType': 'string' },
            type: 'string',
        };
    }
}
exports.UpdateContactRestrictSettingRequest = UpdateContactRestrictSettingRequest;
class UpdateContactRestrictSettingResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'number',
        };
    }
}
exports.UpdateContactRestrictSettingResponseBody = UpdateContactRestrictSettingResponseBody;
class UpdateContactRestrictSettingResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: UpdateContactRestrictSettingResponseBody,
        };
    }
}
exports.UpdateContactRestrictSettingResponse = UpdateContactRestrictSettingResponse;
class UpdateDeptSettngTailFirstHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateDeptSettngTailFirstHeaders = UpdateDeptSettngTailFirstHeaders;
class UpdateDeptSettngTailFirstRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            enable: 'enable',
        };
    }
    static types() {
        return {
            enable: 'boolean',
        };
    }
}
exports.UpdateDeptSettngTailFirstRequest = UpdateDeptSettngTailFirstRequest;
class UpdateDeptSettngTailFirstResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.UpdateDeptSettngTailFirstResponse = UpdateDeptSettngTailFirstResponse;
class UpdateEmpAttrbuteVisibilitySettingHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateEmpAttrbuteVisibilitySettingHeaders = UpdateEmpAttrbuteVisibilitySettingHeaders;
class UpdateEmpAttrbuteVisibilitySettingRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            active: 'active',
            description: 'description',
            excludeDeptIds: 'excludeDeptIds',
            excludeStaffIds: 'excludeStaffIds',
            excludeTagIds: 'excludeTagIds',
            hideFields: 'hideFields',
            id: 'id',
            name: 'name',
            objectDeptIds: 'objectDeptIds',
            objectStaffIds: 'objectStaffIds',
            objectTagIds: 'objectTagIds',
        };
    }
    static types() {
        return {
            active: 'boolean',
            description: 'string',
            excludeDeptIds: { 'type': 'array', 'itemType': 'number' },
            excludeStaffIds: { 'type': 'array', 'itemType': 'string' },
            excludeTagIds: { 'type': 'array', 'itemType': 'number' },
            hideFields: { 'type': 'array', 'itemType': 'string' },
            id: 'number',
            name: 'string',
            objectDeptIds: { 'type': 'array', 'itemType': 'number' },
            objectStaffIds: { 'type': 'array', 'itemType': 'string' },
            objectTagIds: { 'type': 'array', 'itemType': 'number' },
        };
    }
}
exports.UpdateEmpAttrbuteVisibilitySettingRequest = UpdateEmpAttrbuteVisibilitySettingRequest;
class UpdateEmpAttrbuteVisibilitySettingResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'number',
        };
    }
}
exports.UpdateEmpAttrbuteVisibilitySettingResponseBody = UpdateEmpAttrbuteVisibilitySettingResponseBody;
class UpdateEmpAttrbuteVisibilitySettingResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: UpdateEmpAttrbuteVisibilitySettingResponseBody,
        };
    }
}
exports.UpdateEmpAttrbuteVisibilitySettingResponse = UpdateEmpAttrbuteVisibilitySettingResponse;
class UpdateManagementGroupHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateManagementGroupHeaders = UpdateManagementGroupHeaders;
class UpdateManagementGroupRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupName: 'groupName',
            members: 'members',
            resourceIds: 'resourceIds',
            scope: 'scope',
        };
    }
    static types() {
        return {
            groupName: 'string',
            members: { 'type': 'array', 'itemType': UpdateManagementGroupRequestMembers },
            resourceIds: { 'type': 'array', 'itemType': 'string' },
            scope: UpdateManagementGroupRequestScope,
        };
    }
}
exports.UpdateManagementGroupRequest = UpdateManagementGroupRequest;
class UpdateManagementGroupResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.UpdateManagementGroupResponse = UpdateManagementGroupResponse;
class UpdateSeniorSettingHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateSeniorSettingHeaders = UpdateSeniorSettingHeaders;
class UpdateSeniorSettingRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            open: 'open',
            permitDeptIds: 'permitDeptIds',
            permitStaffIds: 'permitStaffIds',
            permitTagIds: 'permitTagIds',
            protectScenes: 'protectScenes',
            seniorStaffId: 'seniorStaffId',
        };
    }
    static types() {
        return {
            open: 'boolean',
            permitDeptIds: { 'type': 'array', 'itemType': 'number' },
            permitStaffIds: { 'type': 'array', 'itemType': 'string' },
            permitTagIds: { 'type': 'array', 'itemType': 'number' },
            protectScenes: { 'type': 'array', 'itemType': 'string' },
            seniorStaffId: 'string',
        };
    }
}
exports.UpdateSeniorSettingRequest = UpdateSeniorSettingRequest;
class UpdateSeniorSettingResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.UpdateSeniorSettingResponse = UpdateSeniorSettingResponse;
class UpdateUserOwnnessHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateUserOwnnessHeaders = UpdateUserOwnnessHeaders;
class UpdateUserOwnnessRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deletedFlag: 'deletedFlag',
            endTime: 'endTime',
            id: 'id',
            ownenssType: 'ownenssType',
            startTime: 'startTime',
        };
    }
    static types() {
        return {
            deletedFlag: 'number',
            endTime: 'number',
            id: 'number',
            ownenssType: 'number',
            startTime: 'number',
        };
    }
}
exports.UpdateUserOwnnessRequest = UpdateUserOwnnessRequest;
class UpdateUserOwnnessResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: 'string',
        };
    }
}
exports.UpdateUserOwnnessResponseBody = UpdateUserOwnnessResponseBody;
class UpdateUserOwnnessResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: UpdateUserOwnnessResponseBody,
        };
    }
}
exports.UpdateUserOwnnessResponse = UpdateUserOwnnessResponse;
class BatchApproveUnionApplyRequestBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            branchCorpId: 'branchCorpId',
            linkDeptId: 'linkDeptId',
            unionRootName: 'unionRootName',
        };
    }
    static types() {
        return {
            branchCorpId: 'string',
            linkDeptId: 'number',
            unionRootName: 'string',
        };
    }
}
exports.BatchApproveUnionApplyRequestBody = BatchApproveUnionApplyRequestBody;
class CreateManagementGroupRequestMembers extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            memberId: 'memberId',
            memberType: 'memberType',
        };
    }
    static types() {
        return {
            memberId: 'string',
            memberType: 'string',
        };
    }
}
exports.CreateManagementGroupRequestMembers = CreateManagementGroupRequestMembers;
class CreateManagementGroupRequestScope extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deptIds: 'deptIds',
            scopeType: 'scopeType',
        };
    }
    static types() {
        return {
            deptIds: { 'type': 'array', 'itemType': 'number' },
            scopeType: 'number',
        };
    }
}
exports.CreateManagementGroupRequestScope = CreateManagementGroupRequestScope;
class GetBranchAuthDataResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            fieldCode: 'fieldCode',
            fieldName: 'fieldName',
            fieldValue: 'fieldValue',
        };
    }
    static types() {
        return {
            fieldCode: 'string',
            fieldName: 'string',
            fieldValue: 'string',
        };
    }
}
exports.GetBranchAuthDataResponseBodyResult = GetBranchAuthDataResponseBodyResult;
class GetCardInfoResponseBodyExtensionCardContactInfoAddressArea extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            region: 'region',
            regionFullName: 'regionFullName',
        };
    }
    static types() {
        return {
            region: 'string',
            regionFullName: 'string',
        };
    }
}
exports.GetCardInfoResponseBodyExtensionCardContactInfoAddressArea = GetCardInfoResponseBodyExtensionCardContactInfoAddressArea;
class GetCardInfoResponseBodyExtensionCardContactInfoAddress extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            area: 'area',
            detail: 'detail',
        };
    }
    static types() {
        return {
            area: GetCardInfoResponseBodyExtensionCardContactInfoAddressArea,
            detail: 'string',
        };
    }
}
exports.GetCardInfoResponseBodyExtensionCardContactInfoAddress = GetCardInfoResponseBodyExtensionCardContactInfoAddress;
class GetCardInfoResponseBodyExtensionCardContactInfoEmail extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            label: 'label',
            value: 'value',
        };
    }
    static types() {
        return {
            label: 'string',
            value: 'string',
        };
    }
}
exports.GetCardInfoResponseBodyExtensionCardContactInfoEmail = GetCardInfoResponseBodyExtensionCardContactInfoEmail;
class GetCardInfoResponseBodyExtensionCardContactInfoTelephone extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            label: 'label',
            value: 'value',
        };
    }
    static types() {
        return {
            label: 'string',
            value: 'string',
        };
    }
}
exports.GetCardInfoResponseBodyExtensionCardContactInfoTelephone = GetCardInfoResponseBodyExtensionCardContactInfoTelephone;
class GetCardInfoResponseBodyExtensionCardContactInfoWechat extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            label: 'label',
            value: 'value',
        };
    }
    static types() {
        return {
            label: 'string',
            value: 'string',
        };
    }
}
exports.GetCardInfoResponseBodyExtensionCardContactInfoWechat = GetCardInfoResponseBodyExtensionCardContactInfoWechat;
class GetCardInfoResponseBodyExtensionCardContactInfo extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            address: 'address',
            email: 'email',
            telephone: 'telephone',
            wechat: 'wechat',
        };
    }
    static types() {
        return {
            address: { 'type': 'array', 'itemType': GetCardInfoResponseBodyExtensionCardContactInfoAddress },
            email: { 'type': 'array', 'itemType': GetCardInfoResponseBodyExtensionCardContactInfoEmail },
            telephone: { 'type': 'array', 'itemType': GetCardInfoResponseBodyExtensionCardContactInfoTelephone },
            wechat: { 'type': 'array', 'itemType': GetCardInfoResponseBodyExtensionCardContactInfoWechat },
        };
    }
}
exports.GetCardInfoResponseBodyExtensionCardContactInfo = GetCardInfoResponseBodyExtensionCardContactInfo;
class GetCardInfoResponseBodyExtension extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            cardContactInfo: 'cardContactInfo',
            corpId: 'corpId',
            department: 'department',
            orgAuthLevel: 'orgAuthLevel',
            orgAuthed: 'orgAuthed',
            orgLogo: 'orgLogo',
            originCardUrl: 'originCardUrl',
            shareContent: 'shareContent',
            thumbnailUrl: 'thumbnailUrl',
            videoFileName: 'videoFileName',
            videoTitle: 'videoTitle',
            videoUrl: 'videoUrl',
        };
    }
    static types() {
        return {
            cardContactInfo: GetCardInfoResponseBodyExtensionCardContactInfo,
            corpId: 'string',
            department: 'string',
            orgAuthLevel: 'number',
            orgAuthed: 'boolean',
            orgLogo: 'string',
            originCardUrl: 'string',
            shareContent: 'string',
            thumbnailUrl: 'string',
            videoFileName: 'string',
            videoTitle: 'string',
            videoUrl: 'string',
        };
    }
}
exports.GetCardInfoResponseBodyExtension = GetCardInfoResponseBodyExtension;
class GetUserCardHolderListResponseBodyList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            avatarUrl: 'avatarUrl',
            cardId: 'cardId',
            extension: 'extension',
            industryName: 'industryName',
            introduce: 'introduce',
            name: 'name',
            orgName: 'orgName',
            templateId: 'templateId',
            title: 'title',
        };
    }
    static types() {
        return {
            avatarUrl: 'string',
            cardId: 'string',
            extension: { 'type': 'map', 'keyType': 'string', 'valueType': 'any' },
            industryName: 'string',
            introduce: 'string',
            name: 'string',
            orgName: 'string',
            templateId: 'string',
            title: 'string',
        };
    }
}
exports.GetUserCardHolderListResponseBodyList = GetUserCardHolderListResponseBodyList;
class ListBasicRoleInPageResponseBodyListOpenActionOpenConditionOpenContactScope extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deptIds: 'deptIds',
            includeMemberDepts: 'includeMemberDepts',
            includeSelfManageDepts: 'includeSelfManageDepts',
            userIds: 'userIds',
        };
    }
    static types() {
        return {
            deptIds: { 'type': 'array', 'itemType': 'number' },
            includeMemberDepts: 'boolean',
            includeSelfManageDepts: 'boolean',
            userIds: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.ListBasicRoleInPageResponseBodyListOpenActionOpenConditionOpenContactScope = ListBasicRoleInPageResponseBodyListOpenActionOpenConditionOpenContactScope;
class ListBasicRoleInPageResponseBodyListOpenActionOpenCondition extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            openContactScope: 'openContactScope',
        };
    }
    static types() {
        return {
            openContactScope: ListBasicRoleInPageResponseBodyListOpenActionOpenConditionOpenContactScope,
        };
    }
}
exports.ListBasicRoleInPageResponseBodyListOpenActionOpenCondition = ListBasicRoleInPageResponseBodyListOpenActionOpenCondition;
class ListBasicRoleInPageResponseBodyListOpenAction extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            actionIds: 'actionIds',
            openCondition: 'openCondition',
        };
    }
    static types() {
        return {
            actionIds: { 'type': 'array', 'itemType': 'string' },
            openCondition: ListBasicRoleInPageResponseBodyListOpenActionOpenCondition,
        };
    }
}
exports.ListBasicRoleInPageResponseBodyListOpenAction = ListBasicRoleInPageResponseBodyListOpenAction;
class ListBasicRoleInPageResponseBodyListOpenMembers extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            belongCorpId: 'belongCorpId',
            memberId: 'memberId',
            memberType: 'memberType',
            operateUserId: 'operateUserId',
        };
    }
    static types() {
        return {
            belongCorpId: 'string',
            memberId: 'string',
            memberType: 'string',
            operateUserId: 'string',
        };
    }
}
exports.ListBasicRoleInPageResponseBodyListOpenMembers = ListBasicRoleInPageResponseBodyListOpenMembers;
class ListBasicRoleInPageResponseBodyList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            openAction: 'openAction',
            openMembers: 'openMembers',
            openResources: 'openResources',
            openRoleId: 'openRoleId',
            openRoleName: 'openRoleName',
        };
    }
    static types() {
        return {
            openAction: ListBasicRoleInPageResponseBodyListOpenAction,
            openMembers: { 'type': 'array', 'itemType': ListBasicRoleInPageResponseBodyListOpenMembers },
            openResources: { 'type': 'array', 'itemType': 'string' },
            openRoleId: 'string',
            openRoleName: 'string',
        };
    }
}
exports.ListBasicRoleInPageResponseBodyList = ListBasicRoleInPageResponseBodyList;
class ListContactHideSettingsResponseBodyList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            active: 'active',
            description: 'description',
            excludeDeptIds: 'excludeDeptIds',
            excludeStaffIds: 'excludeStaffIds',
            excludeTagIds: 'excludeTagIds',
            id: 'id',
            name: 'name',
            objectDeptIds: 'objectDeptIds',
            objectStaffIds: 'objectStaffIds',
            objectTagIds: 'objectTagIds',
        };
    }
    static types() {
        return {
            active: 'boolean',
            description: 'string',
            excludeDeptIds: { 'type': 'array', 'itemType': 'number' },
            excludeStaffIds: { 'type': 'array', 'itemType': 'string' },
            excludeTagIds: { 'type': 'array', 'itemType': 'number' },
            id: 'number',
            name: 'string',
            objectDeptIds: { 'type': 'array', 'itemType': 'number' },
            objectStaffIds: { 'type': 'array', 'itemType': 'string' },
            objectTagIds: { 'type': 'array', 'itemType': 'number' },
        };
    }
}
exports.ListContactHideSettingsResponseBodyList = ListContactHideSettingsResponseBodyList;
class ListContactRestrictSettingResponseBodyList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            active: 'active',
            description: 'description',
            excludeDeptIds: 'excludeDeptIds',
            excludeTagIds: 'excludeTagIds',
            excludeUserIds: 'excludeUserIds',
            id: 'id',
            name: 'name',
            subjectDeptIds: 'subjectDeptIds',
            subjectTagIds: 'subjectTagIds',
            subjectUserIds: 'subjectUserIds',
            type: 'type',
        };
    }
    static types() {
        return {
            active: 'boolean',
            description: 'string',
            excludeDeptIds: { 'type': 'array', 'itemType': 'number' },
            excludeTagIds: { 'type': 'array', 'itemType': 'number' },
            excludeUserIds: { 'type': 'array', 'itemType': 'string' },
            id: 'number',
            name: 'string',
            subjectDeptIds: { 'type': 'array', 'itemType': 'number' },
            subjectTagIds: { 'type': 'array', 'itemType': 'number' },
            subjectUserIds: { 'type': 'array', 'itemType': 'string' },
            type: 'string',
        };
    }
}
exports.ListContactRestrictSettingResponseBodyList = ListContactRestrictSettingResponseBodyList;
class ListEmpAttributeVisibilityResponseBodyList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            active: 'active',
            description: 'description',
            excludeDeptIds: 'excludeDeptIds',
            excludeStaffIds: 'excludeStaffIds',
            excludeTagIds: 'excludeTagIds',
            gmtCreate: 'gmtCreate',
            gmtModified: 'gmtModified',
            hideFields: 'hideFields',
            id: 'id',
            name: 'name',
            objectDeptIds: 'objectDeptIds',
            objectStaffIds: 'objectStaffIds',
            objectTagIds: 'objectTagIds',
        };
    }
    static types() {
        return {
            active: 'boolean',
            description: 'string',
            excludeDeptIds: { 'type': 'array', 'itemType': 'number' },
            excludeStaffIds: { 'type': 'array', 'itemType': 'string' },
            excludeTagIds: { 'type': 'array', 'itemType': 'number' },
            gmtCreate: 'string',
            gmtModified: 'string',
            hideFields: { 'type': 'array', 'itemType': 'string' },
            id: 'number',
            name: 'string',
            objectDeptIds: { 'type': 'array', 'itemType': 'number' },
            objectStaffIds: { 'type': 'array', 'itemType': 'string' },
            objectTagIds: { 'type': 'array', 'itemType': 'number' },
        };
    }
}
exports.ListEmpAttributeVisibilityResponseBodyList = ListEmpAttributeVisibilityResponseBodyList;
class ListEmpLeaveRecordsResponseBodyRecords extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            leaveReason: 'leaveReason',
            leaveTime: 'leaveTime',
            mobile: 'mobile',
            name: 'name',
            stateCode: 'stateCode',
            userId: 'userId',
        };
    }
    static types() {
        return {
            leaveReason: 'string',
            leaveTime: 'string',
            mobile: 'string',
            name: 'string',
            stateCode: 'string',
            userId: 'string',
        };
    }
}
exports.ListEmpLeaveRecordsResponseBodyRecords = ListEmpLeaveRecordsResponseBodyRecords;
class ListManagementGroupsResponseBodyGroupsMembers extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            memberId: 'memberId',
            memberType: 'memberType',
        };
    }
    static types() {
        return {
            memberId: 'string',
            memberType: 'string',
        };
    }
}
exports.ListManagementGroupsResponseBodyGroupsMembers = ListManagementGroupsResponseBodyGroupsMembers;
class ListManagementGroupsResponseBodyGroupsScope extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deptIds: 'deptIds',
            scopeType: 'scopeType',
        };
    }
    static types() {
        return {
            deptIds: { 'type': 'array', 'itemType': 'number' },
            scopeType: 'number',
        };
    }
}
exports.ListManagementGroupsResponseBodyGroupsScope = ListManagementGroupsResponseBodyGroupsScope;
class ListManagementGroupsResponseBodyGroups extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupId: 'groupId',
            groupName: 'groupName',
            members: 'members',
            resourceIds: 'resourceIds',
            scope: 'scope',
        };
    }
    static types() {
        return {
            groupId: 'string',
            groupName: 'string',
            members: { 'type': 'array', 'itemType': ListManagementGroupsResponseBodyGroupsMembers },
            resourceIds: { 'type': 'array', 'itemType': 'string' },
            scope: ListManagementGroupsResponseBodyGroupsScope,
        };
    }
}
exports.ListManagementGroupsResponseBodyGroups = ListManagementGroupsResponseBodyGroups;
class ListOwnedOrgByStaffIdResponseBodyOrgList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            corpId: 'corpId',
            corpName: 'corpName',
        };
    }
    static types() {
        return {
            corpId: 'string',
            corpName: 'string',
        };
    }
}
exports.ListOwnedOrgByStaffIdResponseBodyOrgList = ListOwnedOrgByStaffIdResponseBodyOrgList;
class ListSeniorSettingsResponseBodySeniorWhiteList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            id: 'id',
            name: 'name',
            type: 'type',
        };
    }
    static types() {
        return {
            id: 'string',
            name: 'string',
            type: 'number',
        };
    }
}
exports.ListSeniorSettingsResponseBodySeniorWhiteList = ListSeniorSettingsResponseBodySeniorWhiteList;
class QueryResourceManagementMembersResponseBodyMembers extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            memberId: 'memberId',
            memberType: 'memberType',
        };
    }
    static types() {
        return {
            memberId: 'string',
            memberType: 'string',
        };
    }
}
exports.QueryResourceManagementMembersResponseBodyMembers = QueryResourceManagementMembersResponseBodyMembers;
class UpdateBranchAttributesInCooperateRequestBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            branchCorpId: 'branchCorpId',
            linkDeptId: 'linkDeptId',
            unionRootName: 'unionRootName',
        };
    }
    static types() {
        return {
            branchCorpId: 'string',
            linkDeptId: 'number',
            unionRootName: 'string',
        };
    }
}
exports.UpdateBranchAttributesInCooperateRequestBody = UpdateBranchAttributesInCooperateRequestBody;
class UpdateBranchVisibleSettingInCooperateRequestBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            branchCorpId: 'branchCorpId',
            open: 'open',
            type: 'type',
            visibleBranchCorpIds: 'visibleBranchCorpIds',
            visibleDeptIds: 'visibleDeptIds',
        };
    }
    static types() {
        return {
            branchCorpId: 'string',
            open: 'boolean',
            type: 'number',
            visibleBranchCorpIds: { 'type': 'array', 'itemType': 'string' },
            visibleDeptIds: { 'type': 'array', 'itemType': 'number' },
        };
    }
}
exports.UpdateBranchVisibleSettingInCooperateRequestBody = UpdateBranchVisibleSettingInCooperateRequestBody;
class UpdateManagementGroupRequestMembers extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            memberId: 'memberId',
            memberType: 'memberType',
        };
    }
    static types() {
        return {
            memberId: 'string',
            memberType: 'string',
        };
    }
}
exports.UpdateManagementGroupRequestMembers = UpdateManagementGroupRequestMembers;
class UpdateManagementGroupRequestScope extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deptIds: 'deptIds',
            scopeType: 'scopeType',
        };
    }
    static types() {
        return {
            deptIds: { 'type': 'array', 'itemType': 'number' },
            scopeType: 'number',
        };
    }
}
exports.UpdateManagementGroupRequestScope = UpdateManagementGroupRequestScope;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async batchApproveUnionApply(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new BatchApproveUnionApplyHeaders({});
        return await this.batchApproveUnionApplyWithOptions(request, headers, runtime);
    }
    async batchApproveUnionApplyWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: tea_util_1.default.toArray(request.body),
        });
        return $tea.cast(await this.doROARequest("BatchApproveUnionApply", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/cooperateCorps/unionApplications/approve`, "json", req, runtime), new BatchApproveUnionApplyResponse({}));
    }
    async changeMainAdmin(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ChangeMainAdminHeaders({});
        return await this.changeMainAdminWithOptions(request, headers, runtime);
    }
    async changeMainAdminWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.effectCorpId)) {
            body["effectCorpId"] = request.effectCorpId;
        }
        if (!tea_util_1.default.isUnset(request.sourceUserId)) {
            body["sourceUserId"] = request.sourceUserId;
        }
        if (!tea_util_1.default.isUnset(request.targetUserId)) {
            body["targetUserId"] = request.targetUserId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("ChangeMainAdmin", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/orgAccounts/mainAdministrators/change`, "none", req, runtime), new ChangeMainAdminResponse({}));
    }
    async createCooperateOrg(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateCooperateOrgHeaders({});
        return await this.createCooperateOrgWithOptions(request, headers, runtime);
    }
    async createCooperateOrgWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.industryCode)) {
            body["industryCode"] = request.industryCode;
        }
        if (!tea_util_1.default.isUnset(request.logoMediaId)) {
            body["logoMediaId"] = request.logoMediaId;
        }
        if (!tea_util_1.default.isUnset(request.orgName)) {
            body["orgName"] = request.orgName;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("CreateCooperateOrg", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/cooperateCorps`, "json", req, runtime), new CreateCooperateOrgResponse({}));
    }
    async createManagementGroup(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateManagementGroupHeaders({});
        return await this.createManagementGroupWithOptions(request, headers, runtime);
    }
    async createManagementGroupWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.groupName)) {
            body["groupName"] = request.groupName;
        }
        if (!tea_util_1.default.isUnset(request.members)) {
            body["members"] = request.members;
        }
        if (!tea_util_1.default.isUnset(request.resourceIds)) {
            body["resourceIds"] = request.resourceIds;
        }
        if (!tea_util_1.default.isUnset($tea.toMap(request.scope))) {
            body["scope"] = request.scope;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("CreateManagementGroup", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/managementGroups`, "json", req, runtime), new CreateManagementGroupResponse({}));
    }
    async deleteContactHideSetting(settingId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeleteContactHideSettingHeaders({});
        return await this.deleteContactHideSettingWithOptions(settingId, headers, runtime);
    }
    async deleteContactHideSettingWithOptions(settingId, headers, runtime) {
        settingId = openapi_util_1.default.getEncodeParam(settingId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("DeleteContactHideSetting", "contact_1.0", "HTTP", "DELETE", "AK", `/v1.0/contact/contactHideSettings/${settingId}`, "none", req, runtime), new DeleteContactHideSettingResponse({}));
    }
    async deleteContactRestrictSetting(settingId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeleteContactRestrictSettingHeaders({});
        return await this.deleteContactRestrictSettingWithOptions(settingId, headers, runtime);
    }
    async deleteContactRestrictSettingWithOptions(settingId, headers, runtime) {
        settingId = openapi_util_1.default.getEncodeParam(settingId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("DeleteContactRestrictSetting", "contact_1.0", "HTTP", "DELETE", "AK", `/v1.0/contact/restrictions/settings/${settingId}`, "json", req, runtime), new DeleteContactRestrictSettingResponse({}));
    }
    async deleteEmpAttributeVisibility(settingId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeleteEmpAttributeVisibilityHeaders({});
        return await this.deleteEmpAttributeVisibilityWithOptions(settingId, headers, runtime);
    }
    async deleteEmpAttributeVisibilityWithOptions(settingId, headers, runtime) {
        settingId = openapi_util_1.default.getEncodeParam(settingId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("DeleteEmpAttributeVisibility", "contact_1.0", "HTTP", "DELETE", "AK", `/v1.0/contact/staffAttributes/visibilitySettings/${settingId}`, "none", req, runtime), new DeleteEmpAttributeVisibilityResponse({}));
    }
    async deleteManagementGroup(groupId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new DeleteManagementGroupHeaders({});
        return await this.deleteManagementGroupWithOptions(groupId, headers, runtime);
    }
    async deleteManagementGroupWithOptions(groupId, headers, runtime) {
        groupId = openapi_util_1.default.getEncodeParam(groupId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("DeleteManagementGroup", "contact_1.0", "HTTP", "DELETE", "AK", `/v1.0/contact/managementGroups/${groupId}`, "none", req, runtime), new DeleteManagementGroupResponse({}));
    }
    async getApplyInviteInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetApplyInviteInfoHeaders({});
        return await this.getApplyInviteInfoWithOptions(request, headers, runtime);
    }
    async getApplyInviteInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.deptId)) {
            query["deptId"] = request.deptId;
        }
        if (!tea_util_1.default.isUnset(request.inviterUserId)) {
            query["inviterUserId"] = request.inviterUserId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("GetApplyInviteInfo", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/invites/infos`, "json", req, runtime), new GetApplyInviteInfoResponse({}));
    }
    async getBranchAuthData(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetBranchAuthDataHeaders({});
        return await this.getBranchAuthDataWithOptions(request, headers, runtime);
    }
    async getBranchAuthDataWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.branchCorpId)) {
            query["branchCorpId"] = request.branchCorpId;
        }
        if (!tea_util_1.default.isUnset(request.code)) {
            query["code"] = request.code;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.body)) {
            body = request.body;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("GetBranchAuthData", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/branchAuthDatas/search`, "json", req, runtime), new GetBranchAuthDataResponse({}));
    }
    async getCardInUserHolder(cardId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetCardInUserHolderHeaders({});
        return await this.getCardInUserHolderWithOptions(cardId, headers, runtime);
    }
    async getCardInUserHolderWithOptions(cardId, headers, runtime) {
        cardId = openapi_util_1.default.getEncodeParam(cardId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("GetCardInUserHolder", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/cards/holders/infos/${cardId}`, "json", req, runtime), new GetCardInUserHolderResponse({}));
    }
    async getCardInfo(cardId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetCardInfoHeaders({});
        return await this.getCardInfoWithOptions(cardId, headers, runtime);
    }
    async getCardInfoWithOptions(cardId, headers, runtime) {
        cardId = openapi_util_1.default.getEncodeParam(cardId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("GetCardInfo", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/cards/infos/${cardId}`, "json", req, runtime), new GetCardInfoResponse({}));
    }
    async getCooperateOrgInviteInfo(cooperateCorpId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetCooperateOrgInviteInfoHeaders({});
        return await this.getCooperateOrgInviteInfoWithOptions(cooperateCorpId, headers, runtime);
    }
    async getCooperateOrgInviteInfoWithOptions(cooperateCorpId, headers, runtime) {
        cooperateCorpId = openapi_util_1.default.getEncodeParam(cooperateCorpId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("GetCooperateOrgInviteInfo", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/cooperateCorps/${cooperateCorpId}/inviteInfos`, "json", req, runtime), new GetCooperateOrgInviteInfoResponse({}));
    }
    async getCorpCardStyleList() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetCorpCardStyleListHeaders({});
        return await this.getCorpCardStyleListWithOptions(headers, runtime);
    }
    async getCorpCardStyleListWithOptions(headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("GetCorpCardStyleList", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/cards/styles/lists`, "json", req, runtime), new GetCorpCardStyleListResponse({}));
    }
    async getDingIdByMigrationDingId(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetDingIdByMigrationDingIdHeaders({});
        return await this.getDingIdByMigrationDingIdWithOptions(request, headers, runtime);
    }
    async getDingIdByMigrationDingIdWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.migrationDingId)) {
            query["migrationDingId"] = request.migrationDingId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("GetDingIdByMigrationDingId", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/orgAccount/getDingIdByMigrationDingIds`, "json", req, runtime), new GetDingIdByMigrationDingIdResponse({}));
    }
    async getLatestDingIndex() {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetLatestDingIndexHeaders({});
        return await this.getLatestDingIndexWithOptions(headers, runtime);
    }
    async getLatestDingIndexWithOptions(headers, runtime) {
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("GetLatestDingIndex", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/dingIndexs`, "json", req, runtime), new GetLatestDingIndexResponse({}));
    }
    async getMigrationDingIdByDingId(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetMigrationDingIdByDingIdHeaders({});
        return await this.getMigrationDingIdByDingIdWithOptions(request, headers, runtime);
    }
    async getMigrationDingIdByDingIdWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.dingId)) {
            query["dingId"] = request.dingId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("GetMigrationDingIdByDingId", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/orgAccount/getMigrationDingIdByDingIds`, "json", req, runtime), new GetMigrationDingIdByDingIdResponse({}));
    }
    async getMigrationUnionIdByUnionId(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetMigrationUnionIdByUnionIdHeaders({});
        return await this.getMigrationUnionIdByUnionIdWithOptions(request, headers, runtime);
    }
    async getMigrationUnionIdByUnionIdWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("GetMigrationUnionIdByUnionId", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/orgAccount/getMigrationUnionIdByUnionIds`, "json", req, runtime), new GetMigrationUnionIdByUnionIdResponse({}));
    }
    async getOrgAuthInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetOrgAuthInfoHeaders({});
        return await this.getOrgAuthInfoWithOptions(request, headers, runtime);
    }
    async getOrgAuthInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.targetCorpId)) {
            query["targetCorpId"] = request.targetCorpId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("GetOrgAuthInfo", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/organizations/authInfos`, "json", req, runtime), new GetOrgAuthInfoResponse({}));
    }
    async getTranslateFileJobResult(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetTranslateFileJobResultHeaders({});
        return await this.getTranslateFileJobResultWithOptions(request, headers, runtime);
    }
    async getTranslateFileJobResultWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.jobId)) {
            query["jobId"] = request.jobId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("GetTranslateFileJobResult", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/files/translateResults`, "json", req, runtime), new GetTranslateFileJobResultResponse({}));
    }
    async getUnionIdByMigrationUnionId(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetUnionIdByMigrationUnionIdHeaders({});
        return await this.getUnionIdByMigrationUnionIdWithOptions(request, headers, runtime);
    }
    async getUnionIdByMigrationUnionIdWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.migrationUnionId)) {
            query["migrationUnionId"] = request.migrationUnionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("GetUnionIdByMigrationUnionId", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/orgAccount/getUnionIdByMigrationUnionIds`, "json", req, runtime), new GetUnionIdByMigrationUnionIdResponse({}));
    }
    async getUser(unionId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetUserHeaders({});
        return await this.getUserWithOptions(unionId, headers, runtime);
    }
    async getUserWithOptions(unionId, headers, runtime) {
        unionId = openapi_util_1.default.getEncodeParam(unionId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("GetUser", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/users/${unionId}`, "json", req, runtime), new GetUserResponse({}));
    }
    async getUserCardHolderList(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetUserCardHolderListHeaders({});
        return await this.getUserCardHolderListWithOptions(request, headers, runtime);
    }
    async getUserCardHolderListWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("GetUserCardHolderList", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/cards/holders/lists`, "json", req, runtime), new GetUserCardHolderListResponse({}));
    }
    async isvCardEventPush(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new IsvCardEventPushHeaders({});
        return await this.isvCardEventPushWithOptions(request, headers, runtime);
    }
    async isvCardEventPushWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.isvCardId)) {
            query["isvCardId"] = request.isvCardId;
        }
        if (!tea_util_1.default.isUnset(request.isvToken)) {
            query["isvToken"] = request.isvToken;
        }
        if (!tea_util_1.default.isUnset(request.isvUid)) {
            query["isvUid"] = request.isvUid;
        }
        let body = {};
        if (!tea_util_1.default.isUnset(request.eventParams)) {
            body["eventParams"] = request.eventParams;
        }
        if (!tea_util_1.default.isUnset(request.eventType)) {
            body["eventType"] = request.eventType;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("IsvCardEventPush", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/cards/events/push`, "json", req, runtime), new IsvCardEventPushResponse({}));
    }
    async listBasicRoleInPage(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListBasicRoleInPageHeaders({});
        return await this.listBasicRoleInPageWithOptions(request, headers, runtime);
    }
    async listBasicRoleInPageWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.agentId)) {
            query["agentId"] = request.agentId;
        }
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("ListBasicRoleInPage", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/rbac/administrativeGroups/baseInfos`, "json", req, runtime), new ListBasicRoleInPageResponse({}));
    }
    async listContactHideSettings(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListContactHideSettingsHeaders({});
        return await this.listContactHideSettingsWithOptions(request, headers, runtime);
    }
    async listContactHideSettingsWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("ListContactHideSettings", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/contactHideSettings`, "json", req, runtime), new ListContactHideSettingsResponse({}));
    }
    async listContactRestrictSetting(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListContactRestrictSettingHeaders({});
        return await this.listContactRestrictSettingWithOptions(request, headers, runtime);
    }
    async listContactRestrictSettingWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("ListContactRestrictSetting", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/restrictions/settings`, "json", req, runtime), new ListContactRestrictSettingResponse({}));
    }
    async listEmpAttributeVisibility(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListEmpAttributeVisibilityHeaders({});
        return await this.listEmpAttributeVisibilityWithOptions(request, headers, runtime);
    }
    async listEmpAttributeVisibilityWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("ListEmpAttributeVisibility", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/staffAttributes/visibilitySettings`, "json", req, runtime), new ListEmpAttributeVisibilityResponse({}));
    }
    async listEmpLeaveRecords(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListEmpLeaveRecordsHeaders({});
        return await this.listEmpLeaveRecordsWithOptions(request, headers, runtime);
    }
    async listEmpLeaveRecordsWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.endTime)) {
            query["endTime"] = request.endTime;
        }
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        if (!tea_util_1.default.isUnset(request.startTime)) {
            query["startTime"] = request.startTime;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("ListEmpLeaveRecords", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/empLeaveRecords`, "json", req, runtime), new ListEmpLeaveRecordsResponse({}));
    }
    async listManagementGroups(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListManagementGroupsHeaders({});
        return await this.listManagementGroupsWithOptions(request, headers, runtime);
    }
    async listManagementGroupsWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.maxResults)) {
            query["maxResults"] = request.maxResults;
        }
        if (!tea_util_1.default.isUnset(request.nextToken)) {
            query["nextToken"] = request.nextToken;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("ListManagementGroups", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/managementGroups`, "json", req, runtime), new ListManagementGroupsResponse({}));
    }
    async listOwnedOrgByStaffId(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListOwnedOrgByStaffIdHeaders({});
        return await this.listOwnedOrgByStaffIdWithOptions(request, headers, runtime);
    }
    async listOwnedOrgByStaffIdWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("ListOwnedOrgByStaffId", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/orgAccounts/ownedOrganizations`, "json", req, runtime), new ListOwnedOrgByStaffIdResponse({}));
    }
    async listSeniorSettings(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListSeniorSettingsHeaders({});
        return await this.listSeniorSettingsWithOptions(request, headers, runtime);
    }
    async listSeniorSettingsWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.seniorStaffId)) {
            query["seniorStaffId"] = request.seniorStaffId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("ListSeniorSettings", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/seniorSettings`, "json", req, runtime), new ListSeniorSettingsResponse({}));
    }
    async multiOrgPermissionGrant(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new MultiOrgPermissionGrantHeaders({});
        return await this.multiOrgPermissionGrantWithOptions(request, headers, runtime);
    }
    async multiOrgPermissionGrantWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.grantDeptIdList)) {
            body["grantDeptIdList"] = request.grantDeptIdList;
        }
        if (!tea_util_1.default.isUnset(request.joinCorpId)) {
            body["joinCorpId"] = request.joinCorpId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("MultiOrgPermissionGrant", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/orgAccounts/multiOrgPermissions/auth`, "none", req, runtime), new MultiOrgPermissionGrantResponse({}));
    }
    async queryResourceManagementMembers(resourceId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryResourceManagementMembersHeaders({});
        return await this.queryResourceManagementMembersWithOptions(resourceId, headers, runtime);
    }
    async queryResourceManagementMembersWithOptions(resourceId, headers, runtime) {
        resourceId = openapi_util_1.default.getEncodeParam(resourceId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("QueryResourceManagementMembers", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/resources/${resourceId}/managementMembers`, "json", req, runtime), new QueryResourceManagementMembersResponse({}));
    }
    async queryStatus(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryStatusHeaders({});
        return await this.queryStatusWithOptions(request, headers, runtime);
    }
    async queryStatusWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("QueryStatus", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/orgAccounts/status`, "json", req, runtime), new QueryStatusResponse({}));
    }
    async queryUserManagementResources(userId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryUserManagementResourcesHeaders({});
        return await this.queryUserManagementResourcesWithOptions(userId, headers, runtime);
    }
    async queryUserManagementResourcesWithOptions(userId, headers, runtime) {
        userId = openapi_util_1.default.getEncodeParam(userId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("QueryUserManagementResources", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/users/${userId}/managemementResources`, "json", req, runtime), new QueryUserManagementResourcesResponse({}));
    }
    async searchDepartment(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SearchDepartmentHeaders({});
        return await this.searchDepartmentWithOptions(request, headers, runtime);
    }
    async searchDepartmentWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.offset)) {
            body["offset"] = request.offset;
        }
        if (!tea_util_1.default.isUnset(request.queryWord)) {
            body["queryWord"] = request.queryWord;
        }
        if (!tea_util_1.default.isUnset(request.size)) {
            body["size"] = request.size;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("SearchDepartment", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/departments/search`, "json", req, runtime), new SearchDepartmentResponse({}));
    }
    async searchUser(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SearchUserHeaders({});
        return await this.searchUserWithOptions(request, headers, runtime);
    }
    async searchUserWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.fullMatchField)) {
            body["fullMatchField"] = request.fullMatchField;
        }
        if (!tea_util_1.default.isUnset(request.offset)) {
            body["offset"] = request.offset;
        }
        if (!tea_util_1.default.isUnset(request.queryWord)) {
            body["queryWord"] = request.queryWord;
        }
        if (!tea_util_1.default.isUnset(request.size)) {
            body["size"] = request.size;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("SearchUser", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/users/search`, "json", req, runtime), new SearchUserResponse({}));
    }
    async separateBranchOrg(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SeparateBranchOrgHeaders({});
        return await this.separateBranchOrgWithOptions(request, headers, runtime);
    }
    async separateBranchOrgWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.attachDeptId)) {
            body["attachDeptId"] = request.attachDeptId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("SeparateBranchOrg", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/cooperateCorps/separate`, "json", req, runtime), new SeparateBranchOrgResponse({}));
    }
    async setDisable(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SetDisableHeaders({});
        return await this.setDisableWithOptions(request, headers, runtime);
    }
    async setDisableWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.reason)) {
            body["reason"] = request.reason;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("SetDisable", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/orgAccounts/disable`, "none", req, runtime), new SetDisableResponse({}));
    }
    async setEnable(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SetEnableHeaders({});
        return await this.setEnableWithOptions(request, headers, runtime);
    }
    async setEnableWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("SetEnable", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/orgAccounts/enable`, "none", req, runtime), new SetEnableResponse({}));
    }
    async signOut(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SignOutHeaders({});
        return await this.signOutWithOptions(request, headers, runtime);
    }
    async signOutWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.reason)) {
            body["reason"] = request.reason;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("SignOut", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/orgAccounts/signOut`, "none", req, runtime), new SignOutResponse({}));
    }
    async sortUser(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new SortUserHeaders({});
        return await this.sortUserWithOptions(request, headers, runtime);
    }
    async sortUserWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.sortType)) {
            body["sortType"] = request.sortType;
        }
        if (!tea_util_1.default.isUnset(request.userIdList)) {
            body["userIdList"] = request.userIdList;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("SortUser", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/users/sort`, "json", req, runtime), new SortUserResponse({}));
    }
    async transformToExclusiveAccount(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new TransformToExclusiveAccountHeaders({});
        return await this.transformToExclusiveAccountWithOptions(request, headers, runtime);
    }
    async transformToExclusiveAccountWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.idpDingTalk)) {
            body["idpDingTalk"] = request.idpDingTalk;
        }
        if (!tea_util_1.default.isUnset(request.initPassword)) {
            body["initPassword"] = request.initPassword;
        }
        if (!tea_util_1.default.isUnset(request.loginId)) {
            body["loginId"] = request.loginId;
        }
        if (!tea_util_1.default.isUnset(request.transformType)) {
            body["transformType"] = request.transformType;
        }
        if (!tea_util_1.default.isUnset(request.userId)) {
            body["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("TransformToExclusiveAccount", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/orgAccount/transformToExclusiveAccounts`, "none", req, runtime), new TransformToExclusiveAccountResponse({}));
    }
    async translateFile(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new TranslateFileHeaders({});
        return await this.translateFileWithOptions(request, headers, runtime);
    }
    async translateFileWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.medias)) {
            body["medias"] = request.medias;
        }
        if (!tea_util_1.default.isUnset(request.outputFileName)) {
            body["outputFileName"] = request.outputFileName;
        }
        if (!tea_util_1.default.isUnset(request.unionId)) {
            body["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("TranslateFile", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/files/translate`, "json", req, runtime), new TranslateFileResponse({}));
    }
    async uniqueQueryUserCard(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UniqueQueryUserCardHeaders({});
        return await this.uniqueQueryUserCardWithOptions(request, headers, runtime);
    }
    async uniqueQueryUserCardWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.unionId)) {
            query["unionId"] = request.unionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("UniqueQueryUserCard", "contact_1.0", "HTTP", "GET", "AK", `/v1.0/contact/uniques/cards`, "json", req, runtime), new UniqueQueryUserCardResponse({}));
    }
    async updateBranchAttributesInCooperate(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateBranchAttributesInCooperateHeaders({});
        return await this.updateBranchAttributesInCooperateWithOptions(request, headers, runtime);
    }
    async updateBranchAttributesInCooperateWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: tea_util_1.default.toArray(request.body),
        });
        return $tea.cast(await this.doROARequest("UpdateBranchAttributesInCooperate", "contact_1.0", "HTTP", "PUT", "AK", `/v1.0/contact/cooperateCorps/branchAttributes`, "none", req, runtime), new UpdateBranchAttributesInCooperateResponse({}));
    }
    async updateBranchVisibleSettingInCooperate(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateBranchVisibleSettingInCooperateHeaders({});
        return await this.updateBranchVisibleSettingInCooperateWithOptions(request, headers, runtime);
    }
    async updateBranchVisibleSettingInCooperateWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: tea_util_1.default.toArray(request.body),
        });
        return $tea.cast(await this.doROARequest("UpdateBranchVisibleSettingInCooperate", "contact_1.0", "HTTP", "PUT", "AK", `/v1.0/contact/cooperateCorps/branchVisibleSettings`, "none", req, runtime), new UpdateBranchVisibleSettingInCooperateResponse({}));
    }
    async updateContactHideSetting(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateContactHideSettingHeaders({});
        return await this.updateContactHideSettingWithOptions(request, headers, runtime);
    }
    async updateContactHideSettingWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.active)) {
            body["active"] = request.active;
        }
        if (!tea_util_1.default.isUnset(request.description)) {
            body["description"] = request.description;
        }
        if (!tea_util_1.default.isUnset(request.excludeDeptIds)) {
            body["excludeDeptIds"] = request.excludeDeptIds;
        }
        if (!tea_util_1.default.isUnset(request.excludeStaffIds)) {
            body["excludeStaffIds"] = request.excludeStaffIds;
        }
        if (!tea_util_1.default.isUnset(request.excludeTagIds)) {
            body["excludeTagIds"] = request.excludeTagIds;
        }
        if (!tea_util_1.default.isUnset(request.hideInSearch)) {
            body["hideInSearch"] = request.hideInSearch;
        }
        if (!tea_util_1.default.isUnset(request.hideInUserProfile)) {
            body["hideInUserProfile"] = request.hideInUserProfile;
        }
        if (!tea_util_1.default.isUnset(request.id)) {
            body["id"] = request.id;
        }
        if (!tea_util_1.default.isUnset(request.name)) {
            body["name"] = request.name;
        }
        if (!tea_util_1.default.isUnset(request.objectDeptIds)) {
            body["objectDeptIds"] = request.objectDeptIds;
        }
        if (!tea_util_1.default.isUnset(request.objectStaffIds)) {
            body["objectStaffIds"] = request.objectStaffIds;
        }
        if (!tea_util_1.default.isUnset(request.objectTagIds)) {
            body["objectTagIds"] = request.objectTagIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateContactHideSetting", "contact_1.0", "HTTP", "PUT", "AK", `/v1.0/contact/contactHideSettings`, "json", req, runtime), new UpdateContactHideSettingResponse({}));
    }
    async updateContactRestrictSetting(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateContactRestrictSettingHeaders({});
        return await this.updateContactRestrictSettingWithOptions(request, headers, runtime);
    }
    async updateContactRestrictSettingWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.active)) {
            body["active"] = request.active;
        }
        if (!tea_util_1.default.isUnset(request.description)) {
            body["description"] = request.description;
        }
        if (!tea_util_1.default.isUnset(request.excludeDeptIds)) {
            body["excludeDeptIds"] = request.excludeDeptIds;
        }
        if (!tea_util_1.default.isUnset(request.excludeTagIds)) {
            body["excludeTagIds"] = request.excludeTagIds;
        }
        if (!tea_util_1.default.isUnset(request.excludeUserIds)) {
            body["excludeUserIds"] = request.excludeUserIds;
        }
        if (!tea_util_1.default.isUnset(request.id)) {
            body["id"] = request.id;
        }
        if (!tea_util_1.default.isUnset(request.name)) {
            body["name"] = request.name;
        }
        if (!tea_util_1.default.isUnset(request.restrictInSearch)) {
            body["restrictInSearch"] = request.restrictInSearch;
        }
        if (!tea_util_1.default.isUnset(request.restrictInUserProfile)) {
            body["restrictInUserProfile"] = request.restrictInUserProfile;
        }
        if (!tea_util_1.default.isUnset(request.subjectDeptIds)) {
            body["subjectDeptIds"] = request.subjectDeptIds;
        }
        if (!tea_util_1.default.isUnset(request.subjectTagIds)) {
            body["subjectTagIds"] = request.subjectTagIds;
        }
        if (!tea_util_1.default.isUnset(request.subjectUserIds)) {
            body["subjectUserIds"] = request.subjectUserIds;
        }
        if (!tea_util_1.default.isUnset(request.type)) {
            body["type"] = request.type;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateContactRestrictSetting", "contact_1.0", "HTTP", "PUT", "AK", `/v1.0/contact/restrictions/settings`, "json", req, runtime), new UpdateContactRestrictSettingResponse({}));
    }
    async updateDeptSettngTailFirst(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateDeptSettngTailFirstHeaders({});
        return await this.updateDeptSettngTailFirstWithOptions(request, headers, runtime);
    }
    async updateDeptSettngTailFirstWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.enable)) {
            body["enable"] = request.enable;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateDeptSettngTailFirst", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/depts/settings/priorities`, "none", req, runtime), new UpdateDeptSettngTailFirstResponse({}));
    }
    async updateEmpAttrbuteVisibilitySetting(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateEmpAttrbuteVisibilitySettingHeaders({});
        return await this.updateEmpAttrbuteVisibilitySettingWithOptions(request, headers, runtime);
    }
    async updateEmpAttrbuteVisibilitySettingWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.active)) {
            body["active"] = request.active;
        }
        if (!tea_util_1.default.isUnset(request.description)) {
            body["description"] = request.description;
        }
        if (!tea_util_1.default.isUnset(request.excludeDeptIds)) {
            body["excludeDeptIds"] = request.excludeDeptIds;
        }
        if (!tea_util_1.default.isUnset(request.excludeStaffIds)) {
            body["excludeStaffIds"] = request.excludeStaffIds;
        }
        if (!tea_util_1.default.isUnset(request.excludeTagIds)) {
            body["excludeTagIds"] = request.excludeTagIds;
        }
        if (!tea_util_1.default.isUnset(request.hideFields)) {
            body["hideFields"] = request.hideFields;
        }
        if (!tea_util_1.default.isUnset(request.id)) {
            body["id"] = request.id;
        }
        if (!tea_util_1.default.isUnset(request.name)) {
            body["name"] = request.name;
        }
        if (!tea_util_1.default.isUnset(request.objectDeptIds)) {
            body["objectDeptIds"] = request.objectDeptIds;
        }
        if (!tea_util_1.default.isUnset(request.objectStaffIds)) {
            body["objectStaffIds"] = request.objectStaffIds;
        }
        if (!tea_util_1.default.isUnset(request.objectTagIds)) {
            body["objectTagIds"] = request.objectTagIds;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateEmpAttrbuteVisibilitySetting", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/staffAttributes/visibilitySettings`, "json", req, runtime), new UpdateEmpAttrbuteVisibilitySettingResponse({}));
    }
    async updateManagementGroup(groupId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateManagementGroupHeaders({});
        return await this.updateManagementGroupWithOptions(groupId, request, headers, runtime);
    }
    async updateManagementGroupWithOptions(groupId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        groupId = openapi_util_1.default.getEncodeParam(groupId);
        let body = {};
        if (!tea_util_1.default.isUnset(request.groupName)) {
            body["groupName"] = request.groupName;
        }
        if (!tea_util_1.default.isUnset(request.members)) {
            body["members"] = request.members;
        }
        if (!tea_util_1.default.isUnset(request.resourceIds)) {
            body["resourceIds"] = request.resourceIds;
        }
        if (!tea_util_1.default.isUnset($tea.toMap(request.scope))) {
            body["scope"] = request.scope;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateManagementGroup", "contact_1.0", "HTTP", "PUT", "AK", `/v1.0/contact/managementGroups/${groupId}`, "none", req, runtime), new UpdateManagementGroupResponse({}));
    }
    async updateSeniorSetting(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateSeniorSettingHeaders({});
        return await this.updateSeniorSettingWithOptions(request, headers, runtime);
    }
    async updateSeniorSettingWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.open)) {
            body["open"] = request.open;
        }
        if (!tea_util_1.default.isUnset(request.permitDeptIds)) {
            body["permitDeptIds"] = request.permitDeptIds;
        }
        if (!tea_util_1.default.isUnset(request.permitStaffIds)) {
            body["permitStaffIds"] = request.permitStaffIds;
        }
        if (!tea_util_1.default.isUnset(request.permitTagIds)) {
            body["permitTagIds"] = request.permitTagIds;
        }
        if (!tea_util_1.default.isUnset(request.protectScenes)) {
            body["protectScenes"] = request.protectScenes;
        }
        if (!tea_util_1.default.isUnset(request.seniorStaffId)) {
            body["seniorStaffId"] = request.seniorStaffId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateSeniorSetting", "contact_1.0", "HTTP", "POST", "AK", `/v1.0/contact/seniorSettings`, "none", req, runtime), new UpdateSeniorSettingResponse({}));
    }
    async updateUserOwnness(userId, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateUserOwnnessHeaders({});
        return await this.updateUserOwnnessWithOptions(userId, request, headers, runtime);
    }
    async updateUserOwnnessWithOptions(userId, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        userId = openapi_util_1.default.getEncodeParam(userId);
        let body = {};
        if (!tea_util_1.default.isUnset(request.deletedFlag)) {
            body["deletedFlag"] = request.deletedFlag;
        }
        if (!tea_util_1.default.isUnset(request.endTime)) {
            body["endTime"] = request.endTime;
        }
        if (!tea_util_1.default.isUnset(request.id)) {
            body["id"] = request.id;
        }
        if (!tea_util_1.default.isUnset(request.ownenssType)) {
            body["ownenssType"] = request.ownenssType;
        }
        if (!tea_util_1.default.isUnset(request.startTime)) {
            body["startTime"] = request.startTime;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateUserOwnness", "contact_1.0", "HTTP", "PUT", "AK", `/v1.0/contact/user/${userId}/ownness`, "json", req, runtime), new UpdateUserOwnnessResponse({}));
    }
}
exports.default = Client;
