// @alicloud/dingtalk/dist/trade_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class CheckOpportunityResultHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CheckOpportunityResultHeaders = CheckOpportunityResultHeaders;
class CheckOpportunityResultRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            belongToPhoneNum: 'belongToPhoneNum',
            contactPhoneNum: 'contactPhoneNum',
            corpId: 'corpId',
            deptId: 'deptId',
            marketCode: 'marketCode',
        };
    }
    static types() {
        return {
            belongToPhoneNum: 'string',
            contactPhoneNum: 'string',
            corpId: 'string',
            deptId: 'number',
            marketCode: 'string',
        };
    }
}
exports.CheckOpportunityResultRequest = CheckOpportunityResultRequest;
class CheckOpportunityResultResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            bizSuccess: 'bizSuccess',
        };
    }
    static types() {
        return {
            bizSuccess: 'boolean',
        };
    }
}
exports.CheckOpportunityResultResponseBody = CheckOpportunityResultResponseBody;
class CheckOpportunityResultResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: CheckOpportunityResultResponseBody,
        };
    }
}
exports.CheckOpportunityResultResponse = CheckOpportunityResultResponse;
class CreateOpportunityHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CreateOpportunityHeaders = CreateOpportunityHeaders;
class CreateOpportunityRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            belongToPhoneNum: 'belongToPhoneNum',
            contactPhoneNum: 'contactPhoneNum',
            corpId: 'corpId',
            deptId: 'deptId',
            marketCode: 'marketCode',
        };
    }
    static types() {
        return {
            belongToPhoneNum: 'string',
            contactPhoneNum: 'string',
            corpId: 'string',
            deptId: 'number',
            marketCode: 'string',
        };
    }
}
exports.CreateOpportunityRequest = CreateOpportunityRequest;
class CreateOpportunityResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.CreateOpportunityResponse = CreateOpportunityResponse;
class QueryTradeOrderHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryTradeOrderHeaders = QueryTradeOrderHeaders;
class QueryTradeOrderRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            orderId: 'orderId',
            outerOrderId: 'outerOrderId',
        };
    }
    static types() {
        return {
            orderId: 'number',
            outerOrderId: 'string',
        };
    }
}
exports.QueryTradeOrderRequest = QueryTradeOrderRequest;
class QueryTradeOrderResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            articleCode: 'articleCode',
            articleItemCode: 'articleItemCode',
            articleItemName: 'articleItemName',
            articleName: 'articleName',
            closeTime: 'closeTime',
            createTime: 'createTime',
            fee: 'fee',
            isvCropId: 'isvCropId',
            orderId: 'orderId',
            outerOrderId: 'outerOrderId',
            payFee: 'payFee',
            payTime: 'payTime',
            quantity: 'quantity',
            refundTime: 'refundTime',
            status: 'status',
        };
    }
    static types() {
        return {
            articleCode: 'string',
            articleItemCode: 'string',
            articleItemName: 'string',
            articleName: 'string',
            closeTime: 'number',
            createTime: 'number',
            fee: 'number',
            isvCropId: 'string',
            orderId: 'number',
            outerOrderId: 'string',
            payFee: 'number',
            payTime: 'number',
            quantity: 'number',
            refundTime: 'number',
            status: 'number',
        };
    }
}
exports.QueryTradeOrderResponseBody = QueryTradeOrderResponseBody;
class QueryTradeOrderResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: QueryTradeOrderResponseBody,
        };
    }
}
exports.QueryTradeOrderResponse = QueryTradeOrderResponse;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async checkOpportunityResult(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CheckOpportunityResultHeaders({});
        return await this.checkOpportunityResultWithOptions(request, headers, runtime);
    }
    async checkOpportunityResultWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.belongToPhoneNum)) {
            query["belongToPhoneNum"] = request.belongToPhoneNum;
        }
        if (!tea_util_1.default.isUnset(request.contactPhoneNum)) {
            query["contactPhoneNum"] = request.contactPhoneNum;
        }
        if (!tea_util_1.default.isUnset(request.corpId)) {
            query["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.deptId)) {
            query["deptId"] = request.deptId;
        }
        if (!tea_util_1.default.isUnset(request.marketCode)) {
            query["marketCode"] = request.marketCode;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("CheckOpportunityResult", "trade_1.0", "HTTP", "GET", "AK", `/v1.0/trade/opportunity/check`, "json", req, runtime), new CheckOpportunityResultResponse({}));
    }
    async createOpportunity(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateOpportunityHeaders({});
        return await this.createOpportunityWithOptions(request, headers, runtime);
    }
    async createOpportunityWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.belongToPhoneNum)) {
            body["belongToPhoneNum"] = request.belongToPhoneNum;
        }
        if (!tea_util_1.default.isUnset(request.contactPhoneNum)) {
            body["contactPhoneNum"] = request.contactPhoneNum;
        }
        if (!tea_util_1.default.isUnset(request.corpId)) {
            body["corpId"] = request.corpId;
        }
        if (!tea_util_1.default.isUnset(request.deptId)) {
            body["deptId"] = request.deptId;
        }
        if (!tea_util_1.default.isUnset(request.marketCode)) {
            body["marketCode"] = request.marketCode;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("CreateOpportunity", "trade_1.0", "HTTP", "POST", "AK", `/v1.0/trade/opportunities`, "none", req, runtime), new CreateOpportunityResponse({}));
    }
    async queryTradeOrder(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryTradeOrderHeaders({});
        return await this.queryTradeOrderWithOptions(request, headers, runtime);
    }
    async queryTradeOrderWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.orderId)) {
            body["orderId"] = request.orderId;
        }
        if (!tea_util_1.default.isUnset(request.outerOrderId)) {
            body["outerOrderId"] = request.outerOrderId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("QueryTradeOrder", "trade_1.0", "HTTP", "POST", "AK", `/v1.0/trade/orders/query`, "json", req, runtime), new QueryTradeOrderResponse({}));
    }
}
exports.default = Client;
