// @alicloud/dingtalk/dist/app_market_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class CreateAppGoodsServiceConversationHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.CreateAppGoodsServiceConversationHeaders = CreateAppGoodsServiceConversationHeaders;
class CreateAppGoodsServiceConversationRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            isvUserId: 'isvUserId',
            orderId: 'orderId',
        };
    }
    static types() {
        return {
            isvUserId: 'string',
            orderId: 'number',
        };
    }
}
exports.CreateAppGoodsServiceConversationRequest = CreateAppGoodsServiceConversationRequest;
class CreateAppGoodsServiceConversationResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            conversationName: 'conversationName',
            newConversation: 'newConversation',
        };
    }
    static types() {
        return {
            conversationName: 'string',
            newConversation: 'boolean',
        };
    }
}
exports.CreateAppGoodsServiceConversationResponseBody = CreateAppGoodsServiceConversationResponseBody;
class CreateAppGoodsServiceConversationResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: CreateAppGoodsServiceConversationResponseBody,
        };
    }
}
exports.CreateAppGoodsServiceConversationResponse = CreateAppGoodsServiceConversationResponse;
class GetCoolAppAccessStatusHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetCoolAppAccessStatusHeaders = GetCoolAppAccessStatusHeaders;
class GetCoolAppAccessStatusRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            authCode: 'authCode',
            coolAppCode: 'coolAppCode',
            encFieldBizCode: 'encFieldBizCode',
        };
    }
    static types() {
        return {
            authCode: 'string',
            coolAppCode: 'string',
            encFieldBizCode: 'string',
        };
    }
}
exports.GetCoolAppAccessStatusRequest = GetCoolAppAccessStatusRequest;
class GetCoolAppAccessStatusResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            status: 'status',
        };
    }
    static types() {
        return {
            status: 'string',
        };
    }
}
exports.GetCoolAppAccessStatusResponseBody = GetCoolAppAccessStatusResponseBody;
class GetCoolAppAccessStatusResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetCoolAppAccessStatusResponseBody,
        };
    }
}
exports.GetCoolAppAccessStatusResponse = GetCoolAppAccessStatusResponse;
class GetPersonalExperienceInfoHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetPersonalExperienceInfoHeaders = GetPersonalExperienceInfoHeaders;
class GetPersonalExperienceInfoRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            userId: 'userId',
        };
    }
    static types() {
        return {
            userId: 'string',
        };
    }
}
exports.GetPersonalExperienceInfoRequest = GetPersonalExperienceInfoRequest;
class GetPersonalExperienceInfoResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            result: 'result',
        };
    }
    static types() {
        return {
            result: GetPersonalExperienceInfoResponseBodyResult,
        };
    }
}
exports.GetPersonalExperienceInfoResponseBody = GetPersonalExperienceInfoResponseBody;
class GetPersonalExperienceInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetPersonalExperienceInfoResponseBody,
        };
    }
}
exports.GetPersonalExperienceInfoResponse = GetPersonalExperienceInfoResponse;
class QueryMarketOrderHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryMarketOrderHeaders = QueryMarketOrderHeaders;
class QueryMarketOrderResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            bizOrderId: 'bizOrderId',
            corpId: 'corpId',
            createTimestamp: 'createTimestamp',
            endTimestamp: 'endTimestamp',
            goodsCode: 'goodsCode',
            goodsName: 'goodsName',
            inAppOrder: 'inAppOrder',
            itemCode: 'itemCode',
            itemName: 'itemName',
            paidTimestamp: 'paidTimestamp',
            quantity: 'quantity',
            startTimestamp: 'startTimestamp',
            status: 'status',
            totalActualPayFee: 'totalActualPayFee',
        };
    }
    static types() {
        return {
            bizOrderId: 'number',
            corpId: 'string',
            createTimestamp: 'number',
            endTimestamp: 'number',
            goodsCode: 'string',
            goodsName: 'string',
            inAppOrder: 'boolean',
            itemCode: 'string',
            itemName: 'string',
            paidTimestamp: 'number',
            quantity: 'number',
            startTimestamp: 'number',
            status: 'number',
            totalActualPayFee: 'number',
        };
    }
}
exports.QueryMarketOrderResponseBody = QueryMarketOrderResponseBody;
class QueryMarketOrderResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: QueryMarketOrderResponseBody,
        };
    }
}
exports.QueryMarketOrderResponse = QueryMarketOrderResponse;
class UserTaskReportHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UserTaskReportHeaders = UserTaskReportHeaders;
class UserTaskReportRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            bizNo: 'bizNo',
            operateDate: 'operateDate',
            taskTag: 'taskTag',
            userid: 'userid',
        };
    }
    static types() {
        return {
            bizNo: 'string',
            operateDate: 'string',
            taskTag: 'string',
            userid: 'string',
        };
    }
}
exports.UserTaskReportRequest = UserTaskReportRequest;
class UserTaskReportResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: 'boolean',
        };
    }
}
exports.UserTaskReportResponse = UserTaskReportResponse;
class GetPersonalExperienceInfoResponseBodyResult extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            mainCorpId: 'mainCorpId',
        };
    }
    static types() {
        return {
            mainCorpId: 'string',
        };
    }
}
exports.GetPersonalExperienceInfoResponseBodyResult = GetPersonalExperienceInfoResponseBodyResult;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async createAppGoodsServiceConversation(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new CreateAppGoodsServiceConversationHeaders({});
        return await this.createAppGoodsServiceConversationWithOptions(request, headers, runtime);
    }
    async createAppGoodsServiceConversationWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.isvUserId)) {
            body["isvUserId"] = request.isvUserId;
        }
        if (!tea_util_1.default.isUnset(request.orderId)) {
            body["orderId"] = request.orderId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("CreateAppGoodsServiceConversation", "appMarket_1.0", "HTTP", "POST", "AK", `/v1.0/appMarket/orders/serviceGroups`, "json", req, runtime), new CreateAppGoodsServiceConversationResponse({}));
    }
    async getCoolAppAccessStatus(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetCoolAppAccessStatusHeaders({});
        return await this.getCoolAppAccessStatusWithOptions(request, headers, runtime);
    }
    async getCoolAppAccessStatusWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.authCode)) {
            body["authCode"] = request.authCode;
        }
        if (!tea_util_1.default.isUnset(request.coolAppCode)) {
            body["coolAppCode"] = request.coolAppCode;
        }
        if (!tea_util_1.default.isUnset(request.encFieldBizCode)) {
            body["encFieldBizCode"] = request.encFieldBizCode;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("GetCoolAppAccessStatus", "appMarket_1.0", "HTTP", "POST", "AK", `/v1.0/appMarket/coolApps/accessions/statuses/query`, "json", req, runtime), new GetCoolAppAccessStatusResponse({}));
    }
    async getPersonalExperienceInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetPersonalExperienceInfoHeaders({});
        return await this.getPersonalExperienceInfoWithOptions(request, headers, runtime);
    }
    async getPersonalExperienceInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.userId)) {
            query["userId"] = request.userId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("GetPersonalExperienceInfo", "appMarket_1.0", "HTTP", "GET", "AK", `/v1.0/appMarket/personalExperiences`, "json", req, runtime), new GetPersonalExperienceInfoResponse({}));
    }
    async queryMarketOrder(orderId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryMarketOrderHeaders({});
        return await this.queryMarketOrderWithOptions(orderId, headers, runtime);
    }
    async queryMarketOrderWithOptions(orderId, headers, runtime) {
        orderId = openapi_util_1.default.getEncodeParam(orderId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("QueryMarketOrder", "appMarket_1.0", "HTTP", "GET", "AK", `/v1.0/appMarket/orders/${orderId}`, "json", req, runtime), new QueryMarketOrderResponse({}));
    }
    async userTaskReport(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UserTaskReportHeaders({});
        return await this.userTaskReportWithOptions(request, headers, runtime);
    }
    async userTaskReportWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let body = {};
        if (!tea_util_1.default.isUnset(request.bizNo)) {
            body["bizNo"] = request.bizNo;
        }
        if (!tea_util_1.default.isUnset(request.operateDate)) {
            body["operateDate"] = request.operateDate;
        }
        if (!tea_util_1.default.isUnset(request.taskTag)) {
            body["taskTag"] = request.taskTag;
        }
        if (!tea_util_1.default.isUnset(request.userid)) {
            body["userid"] = request.userid;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UserTaskReport", "appMarket_1.0", "HTTP", "POST", "AK", `/v1.0/appMarket/tasks`, "boolean", req, runtime), new UserTaskReportResponse({}));
    }
}
exports.default = Client;
