// @alicloud/dingtalk/dist/workbench_1_0/client.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// This file is auto-generated, don't edit it
/**
 *
 */
const tea_util_1 = __importStar(require("@alicloud/tea-util")), $Util = tea_util_1;
const openapi_client_1 = __importStar(require("@alicloud/openapi-client")), $OpenApi = openapi_client_1;
const openapi_util_1 = __importDefault(require("@alicloud/openapi-util"));
const $tea = __importStar(require("@alicloud/tea-typescript"));
class GetDingPortalDetailHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetDingPortalDetailHeaders = GetDingPortalDetailHeaders;
class GetDingPortalDetailResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            appUuid: 'appUuid',
            dingPortalName: 'dingPortalName',
            pages: 'pages',
        };
    }
    static types() {
        return {
            appUuid: 'string',
            dingPortalName: 'string',
            pages: { 'type': 'array', 'itemType': GetDingPortalDetailResponseBodyPages },
        };
    }
}
exports.GetDingPortalDetailResponseBody = GetDingPortalDetailResponseBody;
class GetDingPortalDetailResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetDingPortalDetailResponseBody,
        };
    }
}
exports.GetDingPortalDetailResponse = GetDingPortalDetailResponse;
class GetPluginPermissionPointHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetPluginPermissionPointHeaders = GetPluginPermissionPointHeaders;
class GetPluginPermissionPointRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            miniAppId: 'miniAppId',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
        };
    }
}
exports.GetPluginPermissionPointRequest = GetPluginPermissionPointRequest;
class GetPluginPermissionPointResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            permissionPointList: 'permissionPointList',
        };
    }
    static types() {
        return {
            permissionPointList: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.GetPluginPermissionPointResponseBody = GetPluginPermissionPointResponseBody;
class GetPluginPermissionPointResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetPluginPermissionPointResponseBody,
        };
    }
}
exports.GetPluginPermissionPointResponse = GetPluginPermissionPointResponse;
class GetPluginRuleCheckInfoHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.GetPluginRuleCheckInfoHeaders = GetPluginRuleCheckInfoHeaders;
class GetPluginRuleCheckInfoRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            miniAppId: 'miniAppId',
        };
    }
    static types() {
        return {
            miniAppId: 'string',
        };
    }
}
exports.GetPluginRuleCheckInfoRequest = GetPluginRuleCheckInfoRequest;
class GetPluginRuleCheckInfoResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            packCode: 'packCode',
            pluginRuleCheckDetail: 'pluginRuleCheckDetail',
        };
    }
    static types() {
        return {
            packCode: 'string',
            pluginRuleCheckDetail: 'string',
        };
    }
}
exports.GetPluginRuleCheckInfoResponseBody = GetPluginRuleCheckInfoResponseBody;
class GetPluginRuleCheckInfoResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: GetPluginRuleCheckInfoResponseBody,
        };
    }
}
exports.GetPluginRuleCheckInfoResponse = GetPluginRuleCheckInfoResponse;
class ListWorkBenchGroupHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.ListWorkBenchGroupHeaders = ListWorkBenchGroupHeaders;
class ListWorkBenchGroupRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            ecologicalCorpId: 'ecologicalCorpId',
            groupType: 'groupType',
            opUnionId: 'opUnionId',
        };
    }
    static types() {
        return {
            ecologicalCorpId: 'string',
            groupType: 'string',
            opUnionId: 'string',
        };
    }
}
exports.ListWorkBenchGroupRequest = ListWorkBenchGroupRequest;
class ListWorkBenchGroupResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            groupList: 'groupList',
        };
    }
    static types() {
        return {
            groupList: { 'type': 'array', 'itemType': ListWorkBenchGroupResponseBodyGroupList },
        };
    }
}
exports.ListWorkBenchGroupResponseBody = ListWorkBenchGroupResponseBody;
class ListWorkBenchGroupResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: ListWorkBenchGroupResponseBody,
        };
    }
}
exports.ListWorkBenchGroupResponse = ListWorkBenchGroupResponse;
class QueryComponentScopesHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryComponentScopesHeaders = QueryComponentScopesHeaders;
class QueryComponentScopesResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deptVisibleScopes: 'deptVisibleScopes',
            userVisibleScopes: 'userVisibleScopes',
        };
    }
    static types() {
        return {
            deptVisibleScopes: { 'type': 'array', 'itemType': 'number' },
            userVisibleScopes: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.QueryComponentScopesResponseBody = QueryComponentScopesResponseBody;
class QueryComponentScopesResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: QueryComponentScopesResponseBody,
        };
    }
}
exports.QueryComponentScopesResponse = QueryComponentScopesResponse;
class QueryShortcutScopesHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.QueryShortcutScopesHeaders = QueryShortcutScopesHeaders;
class QueryShortcutScopesResponseBody extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            deptVisibleScopes: 'deptVisibleScopes',
            userVisibleScopes: 'userVisibleScopes',
        };
    }
    static types() {
        return {
            deptVisibleScopes: { 'type': 'array', 'itemType': 'number' },
            userVisibleScopes: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.QueryShortcutScopesResponseBody = QueryShortcutScopesResponseBody;
class QueryShortcutScopesResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
            body: 'body',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            body: QueryShortcutScopesResponseBody,
        };
    }
}
exports.QueryShortcutScopesResponse = QueryShortcutScopesResponse;
class UpdateDingPortalPageScopeHeaders extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            commonHeaders: 'commonHeaders',
            xAcsDingtalkAccessToken: 'x-acs-dingtalk-access-token',
        };
    }
    static types() {
        return {
            commonHeaders: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
            xAcsDingtalkAccessToken: 'string',
        };
    }
}
exports.UpdateDingPortalPageScopeHeaders = UpdateDingPortalPageScopeHeaders;
class UpdateDingPortalPageScopeRequest extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            allVisible: 'allVisible',
            deptIds: 'deptIds',
            roleIds: 'roleIds',
            userids: 'userids',
        };
    }
    static types() {
        return {
            allVisible: 'boolean',
            deptIds: { 'type': 'array', 'itemType': 'number' },
            roleIds: { 'type': 'array', 'itemType': 'number' },
            userids: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.UpdateDingPortalPageScopeRequest = UpdateDingPortalPageScopeRequest;
class UpdateDingPortalPageScopeResponse extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            headers: 'headers',
        };
    }
    static types() {
        return {
            headers: { 'type': 'map', 'keyType': 'string', 'valueType': 'string' },
        };
    }
}
exports.UpdateDingPortalPageScopeResponse = UpdateDingPortalPageScopeResponse;
class GetDingPortalDetailResponseBodyPages extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            allVisible: 'allVisible',
            deptIds: 'deptIds',
            pageName: 'pageName',
            pageUuid: 'pageUuid',
            roleIds: 'roleIds',
            userids: 'userids',
        };
    }
    static types() {
        return {
            allVisible: 'boolean',
            deptIds: { 'type': 'array', 'itemType': 'number' },
            pageName: 'string',
            pageUuid: 'string',
            roleIds: { 'type': 'array', 'itemType': 'number' },
            userids: { 'type': 'array', 'itemType': 'string' },
        };
    }
}
exports.GetDingPortalDetailResponseBodyPages = GetDingPortalDetailResponseBodyPages;
class ListWorkBenchGroupResponseBodyGroupList extends $tea.Model {
    constructor(map) {
        super(map);
    }
    static names() {
        return {
            componentId: 'componentId',
            name: 'name',
        };
    }
    static types() {
        return {
            componentId: 'string',
            name: 'string',
        };
    }
}
exports.ListWorkBenchGroupResponseBodyGroupList = ListWorkBenchGroupResponseBodyGroupList;
class Client extends openapi_client_1.default {
    constructor(config) {
        super(config);
        this._endpointRule = "";
        if (tea_util_1.default.empty(this._endpoint)) {
            this._endpoint = "api.dingtalk.com";
        }
    }
    async getDingPortalDetail(appUuid) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetDingPortalDetailHeaders({});
        return await this.getDingPortalDetailWithOptions(appUuid, headers, runtime);
    }
    async getDingPortalDetailWithOptions(appUuid, headers, runtime) {
        appUuid = openapi_util_1.default.getEncodeParam(appUuid);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("GetDingPortalDetail", "workbench_1.0", "HTTP", "GET", "AK", `/v1.0/workbench/dingPortals/${appUuid}`, "json", req, runtime), new GetDingPortalDetailResponse({}));
    }
    async getPluginPermissionPoint(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetPluginPermissionPointHeaders({});
        return await this.getPluginPermissionPointWithOptions(request, headers, runtime);
    }
    async getPluginPermissionPointWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            query["miniAppId"] = request.miniAppId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("GetPluginPermissionPoint", "workbench_1.0", "HTTP", "GET", "AK", `/v1.0/workbench/plugins/permissions`, "json", req, runtime), new GetPluginPermissionPointResponse({}));
    }
    async getPluginRuleCheckInfo(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new GetPluginRuleCheckInfoHeaders({});
        return await this.getPluginRuleCheckInfoWithOptions(request, headers, runtime);
    }
    async getPluginRuleCheckInfoWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.miniAppId)) {
            query["miniAppId"] = request.miniAppId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("GetPluginRuleCheckInfo", "workbench_1.0", "HTTP", "GET", "AK", `/v1.0/workbench/plugins/validationRules`, "json", req, runtime), new GetPluginRuleCheckInfoResponse({}));
    }
    async listWorkBenchGroup(request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new ListWorkBenchGroupHeaders({});
        return await this.listWorkBenchGroupWithOptions(request, headers, runtime);
    }
    async listWorkBenchGroupWithOptions(request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        let query = {};
        if (!tea_util_1.default.isUnset(request.ecologicalCorpId)) {
            query["ecologicalCorpId"] = request.ecologicalCorpId;
        }
        if (!tea_util_1.default.isUnset(request.groupType)) {
            query["groupType"] = request.groupType;
        }
        if (!tea_util_1.default.isUnset(request.opUnionId)) {
            query["opUnionId"] = request.opUnionId;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            query: openapi_util_1.default.query(query),
        });
        return $tea.cast(await this.doROARequest("ListWorkBenchGroup", "workbench_1.0", "HTTP", "GET", "AK", `/v1.0/workbench/groups`, "json", req, runtime), new ListWorkBenchGroupResponse({}));
    }
    async queryComponentScopes(componentId) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryComponentScopesHeaders({});
        return await this.queryComponentScopesWithOptions(componentId, headers, runtime);
    }
    async queryComponentScopesWithOptions(componentId, headers, runtime) {
        componentId = openapi_util_1.default.getEncodeParam(componentId);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("QueryComponentScopes", "workbench_1.0", "HTTP", "GET", "AK", `/v1.0/workbench/components/${componentId}/scopes`, "json", req, runtime), new QueryComponentScopesResponse({}));
    }
    async queryShortcutScopes(shortcutKey) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new QueryShortcutScopesHeaders({});
        return await this.queryShortcutScopesWithOptions(shortcutKey, headers, runtime);
    }
    async queryShortcutScopesWithOptions(shortcutKey, headers, runtime) {
        shortcutKey = openapi_util_1.default.getEncodeParam(shortcutKey);
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
        });
        return $tea.cast(await this.doROARequest("QueryShortcutScopes", "workbench_1.0", "HTTP", "GET", "AK", `/v1.0/workbench/shortcuts/${shortcutKey}/scopes`, "json", req, runtime), new QueryShortcutScopesResponse({}));
    }
    async updateDingPortalPageScope(pageUuid, appUuid, request) {
        let runtime = new $Util.RuntimeOptions({});
        let headers = new UpdateDingPortalPageScopeHeaders({});
        return await this.updateDingPortalPageScopeWithOptions(pageUuid, appUuid, request, headers, runtime);
    }
    async updateDingPortalPageScopeWithOptions(pageUuid, appUuid, request, headers, runtime) {
        tea_util_1.default.validateModel(request);
        pageUuid = openapi_util_1.default.getEncodeParam(pageUuid);
        appUuid = openapi_util_1.default.getEncodeParam(appUuid);
        let body = {};
        if (!tea_util_1.default.isUnset(request.allVisible)) {
            body["allVisible"] = request.allVisible;
        }
        if (!tea_util_1.default.isUnset(request.deptIds)) {
            body["deptIds"] = request.deptIds;
        }
        if (!tea_util_1.default.isUnset(request.roleIds)) {
            body["roleIds"] = request.roleIds;
        }
        if (!tea_util_1.default.isUnset(request.userids)) {
            body["userids"] = request.userids;
        }
        let realHeaders = {};
        if (!tea_util_1.default.isUnset(headers.commonHeaders)) {
            realHeaders = headers.commonHeaders;
        }
        if (!tea_util_1.default.isUnset(headers.xAcsDingtalkAccessToken)) {
            realHeaders["x-acs-dingtalk-access-token"] = tea_util_1.default.toJSONString(headers.xAcsDingtalkAccessToken);
        }
        let req = new $OpenApi.OpenApiRequest({
            headers: realHeaders,
            body: openapi_util_1.default.parseToMap(body),
        });
        return $tea.cast(await this.doROARequest("UpdateDingPortalPageScope", "workbench_1.0", "HTTP", "PUT", "AK", `/v1.0/workbench/dingPortals/${appUuid}/pageScopes/${pageUuid}`, "none", req, runtime), new UpdateDingPortalPageScopeResponse({}));
    }
}
exports.default = Client;
